From 5fe18bad83a7c718f7619d29b74ad419f5e233dc Mon Sep 17 00:00:00 2001
From: Xabier Marquiegui <xmarquiegui@ainguraiiot.com>
Date: Mon, 15 Jul 2024 13:16:57 +0200
Subject: [PATCH] Fix install errors on cross-compile environments

---
 src/extrae-uncore/Makefile.am | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/src/extrae-uncore/Makefile.am b/src/extrae-uncore/Makefile.am
index 4d975770..1a61b3ac 100644
--- a/src/extrae-uncore/Makefile.am
+++ b/src/extrae-uncore/Makefile.am
@@ -3,12 +3,12 @@ SUBDIRS = uncore-service
 dist_bin_SCRIPTS = extrae-uncore.py
 
 install-data-hook:
-	$(top_srcdir)/substitute $(SED) "@sub_PREFIXDIR@" "${prefix}"    ${prefix}/bin/extrae-uncore.py
-	$(top_srcdir)/substitute $(SED) "@sub_LIBDIR@"    "${libdir}"    ${prefix}/bin/extrae-uncore.py
-	$(top_srcdir)/substitute $(SED) "@sub_PAPI_HOME@" "${PAPI_HOME}" ${prefix}/bin/extrae-uncore.py
-	ln -fs ${prefix}/bin/extrae-uncore.py ${prefix}/bin/extrae-uncore
-	chmod u+x ${prefix}/bin/extrae-uncore
+	$(top_srcdir)/substitute $(SED) "@sub_PREFIXDIR@" "${prefix}"    $(DESTDIR)${prefix}/bin/extrae-uncore.py
+	$(top_srcdir)/substitute $(SED) "@sub_LIBDIR@"    "${libdir}"    $(DESTDIR)${prefix}/bin/extrae-uncore.py
+	$(top_srcdir)/substitute $(SED) "@sub_PAPI_HOME@" "${PAPI_HOME}" $(DESTDIR)${prefix}/bin/extrae-uncore.py
+	chmod u+x $(DESTDIR)${prefix}/bin/extrae-uncore.py
+	ln -fs ${prefix}/bin/extrae-uncore.py $(DESTDIR)${prefix}/bin/extrae-uncore
 	$(MKDIR_P) $(DESTDIR)$(sysconfdir)
 	cp extrae_uncore_template.xml $(DESTDIR)$(sysconfdir)/extrae_uncore_template.xml
-	$(top_srcdir)/substitute $(SED) "@sub_PREFIXDIR@" "${prefix}" ${prefix}/etc/extrae_uncore_template.xml
+	$(top_srcdir)/substitute $(SED) "@sub_PREFIXDIR@" "${prefix}" $(DESTDIR)${prefix}/etc/extrae_uncore_template.xml
 
