--- ndoutils-2.1.2/configure	2016-11-14 11:23:05.000000000 -0800
+++ ndoutils-2.1.2.patch/configure	2017-04-04 13:42:10.000000000 -0700
@@ -6277,7 +6277,7 @@
 
 
 
-INSTALL_OPTS="-o $ndo2db_user -g $ndo2db_group"
+INSTALL_OPTS=""
 
 
 
--- ndoutils-2.1.2/src/Makefile.in	2016-11-14 11:23:05.000000000 -0800
+++ ndoutils-2.1.2.patch/src/Makefile.in	2017-04-04 13:43:51.000000000 -0700
@@ -126,9 +126,9 @@
 devclean: distclean
 
 install: install-4x
-	$(INSTALL) -m 774 $(INSTALL_OPTS) file2sock $(BINDIR)
-	$(INSTALL) -m 774 $(INSTALL_OPTS) log2ndo $(BINDIR)
-	$(INSTALL) -m 774 $(INSTALL_OPTS) sockdebug $(BINDIR)
+	$(INSTALL) -m 755 $(INSTALL_OPTS) file2sock $(BINDIR)
+	$(INSTALL) -m 755 $(INSTALL_OPTS) log2ndo $(BINDIR)
+	$(INSTALL) -m 755 $(INSTALL_OPTS) sockdebug $(BINDIR)
 	@echo ""
 	@echo "  Hint: NDOUtils Installation against Nagios v4.x"
 	@echo "  completed."
