diff -Naur ndoutils-2.0.0.old/configure ndoutils-2.0.0/configure
--- ndoutils-2.0.0.old/configure	2014-02-28 19:13:22.000000000 +0100
+++ ndoutils-2.0.0/configure	2014-04-18 10:49:52.749318424 +0200
@@ -7004,7 +7004,7 @@
 
 
 
-INSTALL_OPTS="-o $ndo2db_user -g $ndo2db_group"
+INSTALL_OPTS=""
 
 
 
diff -Naur ndoutils-2.0.0.old/configure.in ndoutils-2.0.0/configure.in
--- ndoutils-2.0.0.old/configure.in	2014-02-28 19:13:22.000000000 +0100
+++ ndoutils-2.0.0/configure.in	2014-04-18 10:49:44.353381035 +0200
@@ -263,7 +263,7 @@
 AC_ARG_WITH(ndo2db_group,AC_HELP_STRING([--with-ndo2db-group=<group>],[sets group name to run NDO2DB]),ndo2db_group=$withval,ndo2db_group=nagios)
 AC_SUBST(ndo2db_user)
 AC_SUBST(ndo2db_group)
-INSTALL_OPTS="-o $ndo2db_user -g $ndo2db_group"
+INSTALL_OPTS=""
 AC_SUBST(INSTALL_OPTS)
 
 
diff -Naur ndoutils-2.0.0.old/src/Makefile.in ndoutils-2.0.0/src/Makefile.in
--- ndoutils-2.0.0.old/src/Makefile.in	2014-02-28 19:13:22.000000000 +0100
+++ ndoutils-2.0.0/src/Makefile.in	2014-04-18 10:49:23.510534175 +0200
@@ -118,9 +118,9 @@
 devclean: distclean
 
 install: install-4x
-	$(INSTALL) -m 774 $(INSTALL_OPTS) file2sock $(DESTDIR)$(BINDIR)
-	$(INSTALL) -m 774 $(INSTALL_OPTS) log2ndo $(DESTDIR)$(BINDIR)
-	$(INSTALL) -m 774 $(INSTALL_OPTS) sockdebug $(DESTDIR)$(BINDIR)
+	$(INSTALL) -m 755 $(INSTALL_OPTS) file2sock $(DESTDIR)$(BINDIR)
+	$(INSTALL) -m 755 $(INSTALL_OPTS) log2ndo $(DESTDIR)$(BINDIR)
+	$(INSTALL) -m 755 $(INSTALL_OPTS) sockdebug $(DESTDIR)$(BINDIR)
 	@echo ""
 	@echo "  Hint: NDOUtils Installation against Nagios v4.x"
 	@echo "  completed."
