diff -Naur ndoutils-2.1.2.old/configure ndoutils-2.1.2/configure
--- ndoutils-2.1.2.old/configure	2016-12-11 21:56:02.759137228 +0100
+++ ndoutils-2.1.2/configure	2016-12-11 21:57:31.047462050 +0100
@@ -6277,7 +6277,7 @@
 
 
 
-INSTALL_OPTS="-o $ndo2db_user -g $ndo2db_group"
+INSTALL_OPTS=""
 
 
 
diff -Naur ndoutils-2.1.2.old/configure.ac ndoutils-2.1.2/configure.ac
--- ndoutils-2.1.2.old/configure.ac	2016-12-11 21:56:02.759137228 +0100
+++ ndoutils-2.1.2/configure.ac	2016-12-11 21:56:29.099532482 +0100
@@ -317,7 +317,7 @@
 AC_ARG_WITH(ndo2db_group,AC_HELP_STRING([--with-ndo2db-group=<group>],[sets group name to run NDO2DB]),ndo2db_group=$withval,ndo2db_group=nagios)
 AC_SUBST(ndo2db_user)
 AC_SUBST(ndo2db_group)
-INSTALL_OPTS="-o $ndo2db_user -g $ndo2db_group"
+INSTALL_OPTS=""
 AC_SUBST(INSTALL_OPTS)
 
 
diff -Naur ndoutils-2.1.2.old/Makefile.in ndoutils-2.1.2/Makefile.in
--- ndoutils-2.1.2.old/Makefile.in	2016-12-11 21:56:02.757137198 +0100
+++ ndoutils-2.1.2/Makefile.in	2016-12-11 22:08:50.063651109 +0100
@@ -98,10 +98,10 @@
 	@echo ""
 
 install-config:
-	$(INSTALL) -m 775 $(INSTALL_OPTS) -d $(CFGDIR)
-	$(INSTALL) -m 775 $(INSTALL_OPTS) -d $(PIPEDIR)
-	$(INSTALL) -m 644 $(INSTALL_OPTS) config/ndo2db.cfg-sample $(CFGDIR)
-	$(INSTALL) -m 644 $(INSTALL_OPTS) config/ndomod.cfg-sample $(CFGDIR)
+	$(INSTALL) -m 775 $(INSTALL_OPTS) -d $(DESTDIR)$(CFGDIR)
+	$(INSTALL) -m 775 $(INSTALL_OPTS) -d $(DESTDIR)$(PIPEDIR)
+	$(INSTALL) -m 644 $(INSTALL_OPTS) config/ndo2db.cfg-sample $(DESTDIR)$(CFGDIR)
+	$(INSTALL) -m 644 $(INSTALL_OPTS) config/ndomod.cfg-sample $(DESTDIR)$(CFGDIR)
 	@echo ""
 	@echo "*** Config files installed ***"
 	@echo ""
@@ -140,7 +140,7 @@
 		if test $(INIT_TYPE) = newbsd; then\
 			if test $(DIST) = openbsd; then\
 				echo "# ndo2db@bsd_enable@=NO" >> /etc/rc.conf;\
-				echo "ndo2db@bsd_enable@=\"-d -c $(CFGDIR)/ndo2db.cfg\"" >> /etc/rc.conf;\
+				echo "ndo2db@bsd_enable@=\"-d -c $(DESTDIR)$(CFGDIR)/ndo2db.cfg\"" >> /etc/rc.conf;\
 				echo "Make sure to enable the ndo2db daemon";\
 			else\
 				echo "ndo2db@bsd_enable@=YES" >> /etc/rc.conf;\
diff -Naur ndoutils-2.1.2.old/src/Makefile.in ndoutils-2.1.2/src/Makefile.in
--- ndoutils-2.1.2.old/src/Makefile.in	2016-12-11 21:56:02.766137333 +0100
+++ ndoutils-2.1.2/src/Makefile.in	2016-12-11 22:06:16.270343345 +0100
@@ -126,9 +126,9 @@
 devclean: distclean
 
 install: install-4x
-	$(INSTALL) -m 774 $(INSTALL_OPTS) file2sock $(BINDIR)
-	$(INSTALL) -m 774 $(INSTALL_OPTS) log2ndo $(BINDIR)
-	$(INSTALL) -m 774 $(INSTALL_OPTS) sockdebug $(BINDIR)
+	$(INSTALL) -m 755 $(INSTALL_OPTS) file2sock $(DESTDIR)$(BINDIR)
+	$(INSTALL) -m 755 $(INSTALL_OPTS) log2ndo $(DESTDIR)$(BINDIR)
+	$(INSTALL) -m 755 $(INSTALL_OPTS) sockdebug $(DESTDIR)$(BINDIR)
 	@echo ""
 	@echo "  Hint: NDOUtils Installation against Nagios v4.x"
 	@echo "  completed."
@@ -147,20 +147,20 @@
 	@echo ""
 
 install-2x:
-	$(INSTALL) -m 775 $(INSTALL_OPTS) -d $(PIPEDIR)
-	$(INSTALL) -m 775 $(INSTALL_OPTS) -d $(BINDIR)
-	$(INSTALL) -m 755 $(INSTALL_OPTS) ndo2db-2x $(BINDIR)/ndo2db
-	$(INSTALL) -m 755 $(INSTALL_OPTS) ndomod-2x.o $(BINDIR)/ndomod.o
+	$(INSTALL) -m 775 $(INSTALL_OPTS) -d $(DESTDIR)$(PIPEDIR)
+	$(INSTALL) -m 775 $(INSTALL_OPTS) -d $(DESTDIR)$(BINDIR)
+	$(INSTALL) -m 755 $(INSTALL_OPTS) ndo2db-2x $(DESTDIR)$(BINDIR)/ndo2db
+	$(INSTALL) -m 755 $(INSTALL_OPTS) ndomod-2x.o $(DESTDIR)$(BINDIR)/ndomod.o
 
 install-3x:
-	$(INSTALL) -m 775 $(INSTALL_OPTS) -d $(PIPEDIR)
-	$(INSTALL) -m 775 $(INSTALL_OPTS) -d $(BINDIR)
-	$(INSTALL) -m 755 $(INSTALL_OPTS) ndo2db-3x $(BINDIR)/ndo2db
-	$(INSTALL) -m 755 $(INSTALL_OPTS) ndomod-3x.o $(BINDIR)/ndomod.o
+	$(INSTALL) -m 775 $(INSTALL_OPTS) -d $(DESTDIR)$(PIPEDIR)
+	$(INSTALL) -m 775 $(INSTALL_OPTS) -d $(DESTDIR)$(BINDIR)
+	$(INSTALL) -m 755 $(INSTALL_OPTS) ndo2db-3x $(DESTDIR)$(BINDIR)/ndo2db
+	$(INSTALL) -m 755 $(INSTALL_OPTS) ndomod-3x.o $(DESTDIR)$(BINDIR)/ndomod.o
 
 install-4x:
-	$(INSTALL) -m 775 $(INSTALL_OPTS) -d $(PIPEDIR)
-	$(INSTALL) -m 775 $(INSTALL_OPTS) -d $(BINDIR)
-	$(INSTALL) -m 755 $(INSTALL_OPTS) ndo2db-4x $(BINDIR)/ndo2db
-	$(INSTALL) -m 755 $(INSTALL_OPTS) ndomod-4x.o $(BINDIR)/ndomod.o
+	$(INSTALL) -m 775 $(INSTALL_OPTS) -d $(DESTDIR)$(PIPEDIR)
+	$(INSTALL) -m 775 $(INSTALL_OPTS) -d $(DESTDIR)$(BINDIR)
+	$(INSTALL) -m 755 $(INSTALL_OPTS) ndo2db-4x $(DESTDIR)$(BINDIR)/ndo2db
+	$(INSTALL) -m 755 $(INSTALL_OPTS) ndomod-4x.o $(DESTDIR)$(BINDIR)/ndomod.o
 
