diff -Naur ndoutils-2.1.2.old/src/queue.c ndoutils-2.1.2/src/queue.c
--- ndoutils-2.1.2.old/src/queue.c	2016-11-14 20:23:05.000000000 +0100
+++ ndoutils-2.1.2/src/queue.c	2016-12-11 21:47:35.434613721 +0100
@@ -99,7 +99,7 @@
 		if(msgctl(queue_id, IPC_STAT, &queue_stats)) {
 			sprintf(curstats, "Unable to determine current message queue usage: error reading IPC_STAT: %d", errno);
 			sprintf(logmsg, logfmt, curstats);
-			syslog(LOG_ERR, logmsg);
+			syslog(LOG_ERR, "%s", logmsg);
 			}
 		else {
 #if defined( __linux__)
@@ -108,18 +108,18 @@
 			if( msgmni < 0) {
 				sprintf(curstats, "Unable to determine current message queue usage: error reading IPC_INFO: %d", errno);
 				sprintf(logmsg, logfmt, curstats);
-				syslog(LOG_ERR, logmsg);
+				syslog(LOG_ERR, "%s", logmsg);
 				}
 			else {
 				sprintf(curstats, statsfmt, queue_stats.msg_qnum,
 						(unsigned long)msgmni, queue_stats.__msg_cbytes,
 						queue_stats.msg_qbytes);
 				sprintf(logmsg, logfmt, curstats);
-				syslog(LOG_ERR, logmsg);
+				syslog(LOG_ERR, "%s", logmsg);
 				}
 #else
 			sprintf(logmsg, logfmt, "");
-			syslog(LOG_ERR, logmsg);
+			syslog(LOG_ERR, "%s", logmsg);
 #endif
 			}
 		last_retry_log_time = now;
