--- boost_1_71_0/tools/build/src/engine/build.sh~       2019-10-09 13:19:04.837484245 -0700
+++ boost_1_71_0/tools/build/src/engine/build.sh        2019-10-09 13:19:30.843484245 -0700
@@ -234,7 +234,8 @@
             export LD_RUN_PATH
             . ${B2_TOOLSET_ROOT}bin/iccvars.sh $ARCH
         fi
-        B2_CXX="${CXX} -xc++"
+       # https://github.com/boostorg/build/issues/475
+        B2_CXX="${CXX} -xc++  -std=c++11"
         B2_CXXFLAGS_RELEASE="-O3 -s"
         B2_CXXFLAGS_DEBUG="-O0 -g -p"
     ;;

