--- lib/spack/spack/__init__.py.orig	2016-01-21 15:05:47.191197890 -0800
+++ lib/spack/spack/__init__.py	2016-01-21 15:06:32.112006915 -0800
@@ -33,7 +33,7 @@
 spack_file = join_path(prefix, "bin", "spack")
 
 # spack directory hierarchy
-etc_path       = join_path(prefix, "etc")
+etc_path       = join_path("/", "etc")
 lib_path       = join_path(prefix, "lib", "spack")
 build_env_path = join_path(lib_path, "env")
 module_path    = join_path(lib_path, "spack")
@@ -42,8 +42,8 @@
 hooks_path     = join_path(module_path, "hooks")
 var_path       = join_path(prefix, "var", "spack")
 stage_path     = join_path(var_path, "stage")
-install_path   = join_path(prefix, "opt")
-share_path     = join_path(prefix, "share", "spack")
+install_path   = join_path("/", "opt", "ohpc", "pub")
+share_path     = join_path("/", "opt", "ohpc", "pub")
 
 #
 # Set up the packages database.
--- lib/spack/spack/architecture.py.orig	2016-01-21 16:21:55.000000000 -0800
+++ lib/spack/spack/architecture.py	2016-01-21 16:22:24.000000000 -0800
@@ -84,7 +84,7 @@
 
     # Couldn't determine the sys_type for this machine.
     if sys_type is None:
-        return "unknown_arch"
+        return ""
 
     if not isinstance(sys_type, basestring):
         raise InvalidSysTypeError(sys_type)
--- lib/spack/spack/modules.py.orig	2016-01-21 16:22:07.000000000 -0800
+++ lib/spack/spack/modules.py	2016-01-21 16:22:40.000000000 -0800
@@ -220,7 +220,7 @@
 
 class TclModule(EnvModule):
     name = 'tcl'
-    path = join_path(spack.share_path, "modules")
+    path = join_path(spack.share_path, "modulefiles")
 
     @property
     def file_name(self):
