--- setuptools-18.8/setuptools/command/easy_install.py	2015-12-01 19:00:35.000000000 -0800
+++ setuptools-18.8.patch/setuptools/command/easy_install.py	2016-02-18 10:36:17.433544800 -0800
@@ -425,6 +425,13 @@
         instdir = normalize_path(self.install_dir)
         pth_file = os.path.join(instdir, 'easy-install.pth')
 
+        # make sure that the directory exists
+        if not os.path.exists(instdir):
+            try:
+                os.makedirs(instdir)
+            except (OSError,IOError):
+                self.cant_write_to_target()
+
         # Is it a configured, PYTHONPATH, implicit, or explicit site dir?
         is_site_dir = instdir in self.all_site_dirs
 
