--- plasma-installer_2.8.0/script/framework.py	2015-12-05 03:59:11.000000000 -0800
+++ plasma-installer_2.8.0.patch/script/framework.py	2017-08-25 10:35:25.000000000 -0700
@@ -53,7 +53,7 @@
                     "2.0.0",
                     "1.0.1", "1.0.0")
     oldversions  = ("2.1.0", "2.0.0", "1.0.1", "1.0.0") # Releases from version which need the internal cblas and lapack
-    lapversion   = "lapack-3.6.0"
+    lapversion   = "lapack-3.8.0"
     ranlib       = ""                        # the ranlib command
     clean        = 0
     nbcores      = 2
--- plasma-installer_2.8.0/script/lapcwrapper.py	2016-03-08 09:26:24.000000000 -0800
+++ plasma-installer_2.8.0.patch/script/lapcwrapper.py	2017-08-25 13:06:30.000000000 -0700
@@ -279,7 +279,6 @@
         shutil.copy('LAPACKE/include/lapacke_config.h', os.path.join(self.prefix,'include/lapacke_config.h'))
         shutil.copy('LAPACKE/include/lapacke_utils.h', os.path.join(self.prefix,'include/lapacke_utils.h'))
         shutil.copy('LAPACKE/include/lapacke_mangling.h', os.path.join(self.prefix,'include/lapacke_mangling.h'))
-        shutil.copy('LAPACKE/include/lapacke_mangling_with_flags.h', os.path.join(self.prefix,'include/lapacke_mangling_with_flags.h'))
 
         # set framework variables to point to the freshly installed BLAS library
         self.config.lapclib   = '-L'+os.path.join(self.prefix,'lib')+' -llapacke'
 
--- plasma-installer_2.8.0/script/lapcwrapper.py.org	2018-07-11 08:12:56.098166020 -0400
+++ plasma-installer_2.8.0/script/lapcwrapper.py	2018-07-11 08:13:43.144487294 -0400
@@ -36,7 +36,7 @@
         self.prefix   = plasma.prefix
         self.plasma   = plasma
         self.lapversion = plasma.lapversion
-        self.lapackurl  = "http://www.netlib.org/lapack/"+self.lapversion+".tgz"
+        self.lapackurl  = "http://www.netlib.org/lapack/"+self.lapversion+".tar.gz"
 
         if self.downlapc == 2:
             self.down_install_lapc()
@@ -145,10 +145,10 @@
 
             # unzip and untar
             print 'Unzip and untar reference LAPACK C interface...',
-            comm = 'gunzip -f '+self.lapversion+'.tgz'
+            comm = 'gunzip -f '+self.lapversion+'.tar.gz'
             (output, error, retz) = runShellCommand(comm)
             if retz:
-                print '\n\nLAPCWRAPPER: cannot unzip '+self.lapversion+'.tgz'
+                print '\n\nLAPCWRAPPER: cannot unzip '+self.lapversion+'.tar.gz'
                 print 'stderr:\n','*'*40,'\n',comm,'\n',error,'\n','*'*40
                 sys.exit()
 
