--- trilinos-11.14.3-Source/cmake/tribits/devtools_install/install-gcc.py	2015-04-17 12:05:04.000000000 -0700
+++ trilinos-11.14.3-Source.patch/cmake/tribits/devtools_install/install-gcc.py	2015-05-01 10:31:32.000000000 -0700
@@ -258,7 +258,7 @@
     print "Writing to GCC specs file "+specsfile
     gccSpecs = getCmndOutput(gccExec+" -dumpspecs", True)
     #print "gccSpecs:\n", gccSpecs
-    gccSpecsMod = addRpathToLink(gccSpecs, rpath)
+    gccSpecsMod = addRpathToLink(gccSpecs, "")
     #print "gccSpecsMod:\n", gccSpecsMod
     writeStrToFile(specsfile, gccSpecsMod)
 
