--- ./test/fw/bin/pbs_snapshot.org	2020-01-17 18:59:27.184706913 +0000
+++ ./test/fw/bin/pbs_snapshot	2020-01-17 18:59:58.563954236 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # coding: utf-8
 
 # Copyright (C) 1994-2018 Altair Engineering, Inc.
--- ./test/fw/bin/pbs_stat.org	2020-01-17 18:59:43.315834053 +0000
+++ ./test/fw/bin/pbs_stat	2020-01-17 19:00:04.913004272 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # coding: utf-8
 
 # Copyright (C) 1994-2018 Altair Engineering, Inc.
--- ./test/fw/bin/pbs_loganalyzer.org	2020-01-17 18:59:54.003918291 +0000
+++ ./test/fw/bin/pbs_loganalyzer	2020-01-17 19:00:12.339062801 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python2
 # coding: utf-8
 
 # Copyright (C) 1994-2018 Altair Engineering, Inc.
