diff --git a/Makefile.in b/Makefile.in
index c3e934c..7fe5afc 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -3692,7 +3692,7 @@ doc/aclocal.1 doc/automake.1:
 doc/aclocal-$(APIVERSION).1: $(aclocal_script) lib/Automake/Config.pm
 	$(update_mans) aclocal-$(APIVERSION)
 doc/automake-$(APIVERSION).1: $(automake_script) lib/Automake/Config.pm
-	$(update_mans) automake-$(APIVERSION)
+	$(update_mans) automake-$(APIVERSION) --no-discard-stderr
 
 # We depend on configure.ac so that we regenerate the tarball
 # whenever the Automake version changes.
