Index: boost_1_56_0/boost/config/compiler/gcc.hpp
===================================================================
--- boost_1_56_0.orig/boost/config/compiler/gcc.hpp
+++ boost_1_56_0/boost/config/compiler/gcc.hpp
@@ -68,7 +68,7 @@
 // those platforms where we can know for sure). It will get turned off again
 // later if no threading API is detected.
 //
-#if !defined(__MINGW32__) && !defined(linux) && !defined(__linux) && !defined(__linux__)
+#if !defined(__MINGW32__)
 # define BOOST_HAS_THREADS
 #endif
 
