--- warewulf-common-3.5/lib/Warewulf/SystemFactory.pm	2014-02-06 13:37:25.000000000 -0800
+++ warewulf-common-3.5.patch/lib/Warewulf/SystemFactory.pm	2014-09-10 11:40:34.586450600 -0700
@@ -49,6 +49,8 @@
     if (! $type) {
         if (-f "/etc/redhat-release") {
             $type = "rhel";
+        } elsif (-f "/etc/SuSE-release") {
+            $type = "Suse";
         }
 		else {
             $type = "Deb";
--- warewulf-common-3.5/lib/Warewulf/System/Makefile.am	2014-02-06 13:23:05.000000000 -0800
+++ warewulf-common-3.5.patch/lib/Warewulf/System/Makefile.am	2014-09-10 11:42:47.873778000 -0700
@@ -1,6 +1,6 @@
 perlmodsdir = ${PERL_VENDORLIB}/Warewulf/System
 
-dist_perlmods_SCRIPTS = Rhel.pm Deb.pm
+dist_perlmods_SCRIPTS = Rhel.pm Deb.pm Suse.pm
 
 MAINTAINERCLEANFILES = Makefile.in
 
--- warewulf-common-3.5/lib/Warewulf/System/Makefile.in	2014-02-07 13:14:27.000000000 -0800
+++ warewulf-common-3.5.patch/lib/Warewulf/System/Makefile.in	2014-09-10 11:43:04.555446000 -0700
@@ -143,7 +143,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 perlmodsdir = ${PERL_VENDORLIB}/Warewulf/System
-dist_perlmods_SCRIPTS = Rhel.pm Deb.pm
+dist_perlmods_SCRIPTS = Rhel.pm Deb.pm Suse.pm
 MAINTAINERCLEANFILES = Makefile.in
 all: all-am
 
--- /dev/null	2014-09-10 14:23:30.000000000 -0700
+++ warewulf-common-3.5/lib/Warewulf/System/Suse.pm	2014-09-10 11:37:12.620256000 -0700
@@ -0,0 +1,168 @@
+# Copyright (c) 2001-2003 Gregory M. Kurtzer
+#
+# Copyright (c) 2003-2011, The Regents of the University of California,
+# through Lawrence Berkeley National Laboratory (subject to receipt of any
+# required approvals from the U.S. Dept. of Energy).  All rights reserved.
+#
+#
+# $Id$
+#
+
+package Warewulf::System::Suse;
+
+use Warewulf::System;
+use Warewulf::Logger;
+
+our @ISA = ('Warewulf::System');
+
+=head1 NAME
+
+Warewulf::Suse - Warewulf's general object instance object interface.
+
+=head1 ABOUT
+
+
+=head1 SYNOPSIS
+
+    use Warewulf::System::Suse;
+
+    my $obj = Warewulf::System::Suse->new();
+
+
+=head1 METHODS
+
+=over 12
+
+=cut
+
+=item new()
+
+The new constructor will create the object that references configuration the
+stores.
+
+=cut
+
+sub
+new($$)
+{
+    my $proto = shift;
+    my $class = ref($proto) || $proto;
+    my $self = {};
+
+    bless($self, $class);
+
+    return $self;
+}
+
+
+=item service($name, $command)
+
+Run a command on a service script (e.g. /etc/init.d/service restart).
+
+=cut
+
+sub
+service($$$)
+{
+    my ($self, $service, $command) = @_;
+
+    &dprint("Running service command: $service, $command\n");
+
+    if ( -x "/bin/systemctl" ) {
+        system("/bin/systemctl $command $service.service");
+    } elsif (-x "/etc/init.d/$service") {
+        $self->{"OUTPUT"} = ();
+        open(SERVICE, "/etc/init.d/$service $command 2>&1|");
+        while(<SERVICE>) {
+            $self->{"OUTPUT"} .= $_;
+        }
+        chomp($self->{"OUTPUT"});
+        if (close SERVICE) {
+            &dprint("Service command ran successfully\n");
+            return(1);
+        } else {
+            &dprint("Error running: /etc/init.d/$service $command\n");
+        }
+    }
+        if ($self->{"OUTPUT"}) {
+        chomp($self->{"OUTPUT"});
+        if (close SERVICE) {
+            &dprint("Service command ran successfully\n");
+            return(1);
+        } else {
+            &dprint("Error running: /usr/bin/systemctl $command $service\n");
+        }
+    }
+    return(1);
+}
+
+
+
+
+=item chkconfig($name, $command)
+
+Enable a service script to be enabled or disabled at boot (e.g.
+/sbin/chkconfig service on).
+
+=cut
+
+sub
+chkconfig($$$)
+{
+    my ($self, $service, $command) = @_;
+
+    if ( -x "/bin/systemctl" ) {
+        system("/bin/systemctl enable $service.service");
+    } elsif (-x "/sbin/chkconfig") {
+        open(CHKCONFIG, "/sbin/chkconfig $service $command 2>&1|");
+        while(<CHKCONFIG>) {
+            $self->{"OUTPUT"} .= $_;
+        }
+        if (defined($self->{"OUTPUT"})) {
+            chomp($self->{"OUTPUT"});
+        }
+        if (close CHKCONFIG) {
+            &dprint("Chkconfig command ran successfully\n");
+            return(1);
+        } else {
+            &dprint("Error running: /sbin/chkconfig $service $command\n");
+        }
+    }
+    return(1);
+}
+
+
+=item output()
+
+return the output cache on a command
+
+=cut
+
+sub
+output($)
+{
+    my $self = shift;
+
+    return(defined($self->{"OUTPUT"}) ? $self->{"OUTPUT"} : "");
+}
+
+
+
+=back
+
+=head1 SEE ALSO
+
+Warewulf::Object
+
+=head1 COPYRIGHT
+
+Copyright (c) 2001-2003 Gregory M. Kurtzer
+
+Copyright (c) 2003-2011, The Regents of the University of California,
+through Lawrence Berkeley National Laboratory (subject to receipt of any
+required approvals from the U.S. Dept. of Energy).  All rights reserved.
+
+=cut
+
+
+1;
--- warewulf-common-3.6/lib/Warewulf/System/Rhel.pm	2014-05-21 11:14:43.000000000 -0700
+++ warewulf-common-3.6.patch/lib/Warewulf/System/Rhel.pm	2016-04-14 17:57:41.000000000 -0700
@@ -5,7 +5,7 @@
 # required approvals from the U.S. Dept. of Energy).  All rights reserved.
 #
 #
-# $Id: Rhel.pm 1680 2014-05-21 18:14:43Z gmk $
+# $Id$
 #
 
 package Warewulf::System::Rhel;
@@ -68,7 +68,9 @@
 
     &dprint("Running service command: $service, $command\n");
 
-    if (-x "/etc/init.d/$service") {
+    if ( -x "/bin/systemctl" ) {
+        system("/bin/systemctl $command $service.service");
+    } elsif (-x "/etc/init.d/$service") {
         $self->{"OUTPUT"} = ();
         open(SERVICE, "/etc/init.d/$service $command 2>&1|");
         while(<SERVICE>) {
@@ -82,10 +84,21 @@
             &dprint("Error running: /etc/init.d/$service $command\n");
         }
     }
-    return();
+        if ($self->{"OUTPUT"}) {
+        chomp($self->{"OUTPUT"});
+        if (close SERVICE) {
+            &dprint("Service command ran successfully\n");
+            return(1);
+        } else {
+            &dprint("Error running: /usr/bin/systemctl $command $service\n");
+        }
+    }
+    return(1);
 }
 
 
+
+
 =item chkconfig($name, $command)
 
 Enable a service script to be enabled or disabled at boot (e.g.
@@ -98,7 +111,9 @@
 {
     my ($self, $service, $command) = @_;
 
-    if (-x "/sbin/chkconfig") {
+    if ( -x "/bin/systemctl" ) {
+        system("/bin/systemctl enable $service.service");
+    } elsif (-x "/sbin/chkconfig") {
         open(CHKCONFIG, "/sbin/chkconfig $service $command 2>&1|");
         while(<CHKCONFIG>) {
             $self->{"OUTPUT"} .= $_;
@@ -113,7 +128,7 @@
             &dprint("Error running: /sbin/chkconfig $service $command\n");
         }
     }
-    return();
+    return(1);
 }
 
 
@@ -128,7 +143,7 @@
 {
     my $self = shift;
 
-    return(defined($self->{"OUTPUT"}) ? $self->{"OUTPUT"} : undef);
+    return(defined($self->{"OUTPUT"}) ? $self->{"OUTPUT"} : "");
 }
 
 
