--- a/lib/Warewulf/File.pm	2018-03-12 15:18:19.000000000 -0700
+++ b/lib/Warewulf/File.pm	2018-07-11 10:23:25.000000000 -0700
@@ -502,6 +502,8 @@
                     $format = "perl";
                 } elsif ($buffer =~ /^#!\/[a-zA-Z0-9\/_\.]+\/python/) {
                     $format = "python";
+                } elsif (-B _) {
+                    $format = "binary";
                 } else {
                     $format = "data";
                 }
