--- warewulf-common-3.6/libexec/wwinit/10-database.init	2014-05-21 14:12:17.000000000 -0400
+++ warewulf-common-3.6.patch/libexec/wwinit/10-database.init	2014-11-04 16:54:57.853838575 -0500
@@ -71,6 +71,21 @@
             fi
         fi
     fi
+elif [ "$DATASTORE" = "mariadb" ]; then
+    wwprint "Checking /usr/lib/systemd/system/mariadb.service is installed"
+    if ! wwtest test -f /usr/lib/systemd/system/mariadb.service; then
+        exit 255
+    fi
+
+    wwprint "Confirming mysqld is configured to start at boot:\n"
+    if ! wwtest test -f /etc/systemd/system/multi-user.target.wants/mariadb.service; then
+        exit 255
+    fi
+
+    wwprint "Checking to see if MySQL needs to be started:\n"
+    if ! wwrun service mariadb start; then
+        exit 255
+    fi
 else
     wwprint "Skipping configuration of datastore driver: $DATABASE\n"
 fi
--- warewulf-common-3.6/lib/Warewulf/DataStore/SQL.pm	2014-05-21 14:14:43.000000000 -0400
+++ warewulf-common-3.6.patch/lib/Warewulf/DataStore/SQL.pm	2014-11-04 17:04:31.415844023 -0500
@@ -44,6 +44,8 @@
 
     if ($db_engine eq "mysql") {
         return(Warewulf::DataStore::SQL::MySQL->new(@_));
+    } elsif ($db_engine eq "mariadb") {
+        return(Warewulf::DataStore::SQL::MySQL->new(@_));
     } else {
         &eprint("Could not load DB type: $db_engine\n");
         exit 1;
