--- a/libexec/wwinit/10-database.init	2018-03-12 15:18:19.000000000 -0700
+++ b/libexec/wwinit/10-database.init	2018-07-11 10:22:20.000000000 -0700
@@ -145,6 +145,8 @@
         wwservice_activate mysqld mysql
     elif wwpackage_check mariadb-server; then
         wwservice_activate mariadb
+    elif wwpackage_check mariadb; then
+        wwservice_activate mysql
     else
         wwprint "No database package seems to exist for mysql!\n" error
         exit 255
