--- a/lib/Warewulf/DataStore/SQL/BaseClass.pm	2017-10-16 09:49:37.000000000 -0700
+++ b/lib/Warewulf/DataStore/SQL/BaseClass.pm	2017-10-16 09:50:34.000000000 -0700
@@ -1099,10 +1099,10 @@
     #
     # Scrub the binstore:
     #
-    if ( $self->{"BINSTORE_KIND"} eq $BINSTORE_KIND_DATABASE && ! $self->del_object_binstore_db_impl($buffer) ) {
+    if ( $self->{"BINSTORE_KIND"} eq $BINSTORE_KIND_DATABASE && ! $self->del_object_binstore_db_impl($object_id) ) {
         goto EARLY_EXIT;
     }
-    elsif ( $self->{"BINSTORE_KIND"} eq $BINSTORE_KIND_FILESYSTEM && ! $self->del_object_binstore_fs_impl($buffer) ) {
+    elsif ( $self->{"BINSTORE_KIND"} eq $BINSTORE_KIND_FILESYSTEM && ! $self->del_object_binstore_fs_impl($object_id) ) {
         goto EARLY_EXIT;
     }
 
