--- a/utils/Makefile	2017-12-04 12:58:56.000000000 -0800
+++ b/utils/Makefile	2017-12-04 13:02:47.000000000 -0800
@@ -84,7 +84,7 @@
 tau_events.o: 	tau_events.cpp
 		$(CXX) $(CFLAGS)  -c tau_events.cpp
 
-tau_instrumentor: tau_instrumentor.o tau_instrument.o tau_selective.o $(PDTDIR)/$(PDTARCHDIR)/lib/libpdb.a 
+tau_instrumentor: tau_instrumentor.o tau_instrument.o tau_selective.o $(PDTDIR)/$(PDTARCHDIR)/lib/libpdb.so 
 		$(PDT_CXX_COMP) $(PDT_CFLAGS) tau_instrumentor.o tau_instrument.o tau_selective.o -o $@ $(PDTOPTS) -lm $(LEXTRAUTIL) 
 
 tau_instrumentor.o: tau_instrumentor.cpp 
@@ -123,13 +123,13 @@
 tau_trace2json.o: tau_trace2json.cpp
 		$(CXX) $(CFLAGS) -std=c++11  $< -c 
 
-tau_wrap: tau_wrap.o tau_selective.o tau_instrument.o $(PDTDIR)/$(PDTARCHDIR)/lib/libpdb.a 
+tau_wrap: tau_wrap.o tau_selective.o tau_instrument.o $(PDTDIR)/$(PDTARCHDIR)/lib/libpdb.so 
 		$(PDT_CXX_COMP) $(PDT_CFLAGS) tau_wrap.o tau_selective.o tau_instrument.o -o $@ $(PDTOPTS) -lm $(LEXTRAUTIL) 
 
 tau_wrap.o: tau_wrap.cpp 
 		$(PDT_CXX_COMP) $(PDT_CFLAGS) -I$(PDTDIR)/include -c tau_wrap.cpp $(WRAPOPTS)
 
-tau_ompcheck: tau_ompcheck.o $(PDTDIR)/$(PDTARCHDIR)/lib/libpdb.a 
+tau_ompcheck: tau_ompcheck.o $(PDTDIR)/$(PDTARCHDIR)/lib/libpdb.so 
 		$(PDT_CXX_COMP) $(PDT_CFLAGS) tau_ompcheck.o -o $@ $(PDTOPTS) -lm $(LEXTRAUTIL) 
 
 tau_ompcheck.o: tau_ompcheck.cpp 
@@ -142,10 +142,10 @@
 		gcc -c perf2tau.c -I$(PERFINCDIR)
 
 
-$(PDTDIR)/$(PDTARCHDIR)/lib/libpdb.a:
+$(PDTDIR)/$(PDTARCHDIR)/lib/libpdb.so:
 		@echo 
 		@echo "******************************************************"
-		@echo "PDT Configuration Error. $(PDTDIR)/$(PDTARCHDIR)/lib/libpdb.a not found."
+		@echo "PDT Configuration Error. $(PDTDIR)/$(PDTARCHDIR)/lib/libpdb.so not found."
 	        @echo "Please configure PDT again and then install TAU."
 		@echo 
 		@echo "******************************************************"
