diff -ruN tau-2.25.2.old/include/Makefile.skel tau-2.25.2/include/Makefile.skel
--- tau-2.25.2.old/include/Makefile.skel	2016-08-21 07:41:18.000000000 +0100
+++ tau-2.25.2/include/Makefile.skel	2016-12-07 19:02:25.000000000 +0000
@@ -1397,7 +1397,7 @@
 	-optTauCC="$(TAU_CCOMPILER)" \
 	-optTauIncludes="$(OPARIINCDIR) $(TAU_INCLUDE) $(TAU_MPI_COMPILE_INCLUDE) $(TAU_SHMEM_INCLUDE) " \
 	-optTauDefs="$(TAU_DEFS)" $(TAU_IBM_NOCOMPINST) \
-        -optLinking="$(TAU_SCOREP_EXTRA_FLAGS) $(TAU_MPI_FLIBS) $(TAU_LIBS) $(TAU_CXXLIBS) $(TAU_SHMEM_LIBS) $(TAU_STATIC_WRAPPERS) $(TAU_LDFLAGS) "\
+        -optLinking="-lTAU $(TAU_SCOREP_EXTRA_FLAGS) $(TAU_MPI_FLIBS) $(TAU_LIBS) $(TAU_CXXLIBS) $(TAU_SHMEM_LIBS) $(TAU_STATIC_WRAPPERS) $(TAU_LDFLAGS) "\
         -optSharedLinking="-L$(TAU_LIB_DIR)/shared$(TAU_CONFIG) -lTAU $(TAU_MPI_FLIBS)  $(TAU_SHMEM_LIBS) $(TAU_EXLIBS) $(TAU_CXXLIBS) $(TAU_STATIC_WRAPPERS) $(TAU_LINKER_RPATH_OPT) \
 				 $(TAU_LDFLAGS)"\
 				-optMICOffloadLinking="$(MIC_OFFLOAD_LINKING)"\
