--- tau-2.24/src/Profile/PapiLayer.cpp	2014-11-10 14:07:45.000000000 -0800
+++ tau-2.24.patch/src/Profile/PapiLayer.cpp	2015-03-31 09:16:37.000000000 -0700
@@ -692,6 +692,7 @@
     printf("WARNING: TAU is already using PAPI counters. Please unset the TAU_METRICS environment variable so PAPI events do no appear in it if you plan to use TAU_TRACK_POWER API. Currently, TAU does not support both at the same time due to the higer overhead of power events.\n");
     return -1;
   }
+  return 0;
 }
 /////////////////////////////////////////////////
 int PapiLayer::initializePerfRAPL(int tid) {
