diff -ruN mvapich2-2.1.old/src/mpid/ch3/channels/common/include/mv2_clock.h mvapich2-2.1/src/mpid/ch3/channels/common/include/mv2_clock.h
--- mvapich2-2.1.old/src/mpid/ch3/channels/common/include/mv2_clock.h	2015-04-02 22:00:25.000000000 +0100
+++ mvapich2-2.1/src/mpid/ch3/channels/common/include/mv2_clock.h	2016-09-01 17:08:26.216406038 +0100
@@ -81,6 +81,19 @@
     asm volatile ("mov %0=ar.itc" : "=r" (ret));
     return ret;
 }
+#elif defined(__aarch64__)
+typedef unsigned long long cycles_t;
+static inline cycles_t get_cycles()
+{
+    cycles_t ret;
+
+    asm volatile("    isb                \n"
+                 "    mrs %0, cntvct_el0 \n"
+                 : "=r" (ret)
+                 :
+                 : "memory");
+    return ret;
+}
 
 #else
 #warning get_cycles not implemented for this architecture: attempt asm/timex.h
