diff -aruN SuperLU_DIST_3.2.orig/EXAMPLE/pddrive1_ABglobal.c SuperLU_DIST_3.2/EXAMPLE/pddrive1_ABglobal.c
--- SuperLU_DIST_3.2.orig/EXAMPLE/pddrive1_ABglobal.c	2012-10-24 21:46:18.000000000 +0200
+++ SuperLU_DIST_3.2/EXAMPLE/pddrive1_ABglobal.c	2012-12-09 12:39:11.577668100 +0100
@@ -34,7 +34,7 @@
  * </pre>
  */
 
-int main(int argc, char *argv[])
+void main(int argc, char *argv[])
 {
     superlu_options_t options;
     SuperLUStat_t stat;
diff -aruN SuperLU_DIST_3.2.orig/EXAMPLE/pddrive1.c SuperLU_DIST_3.2/EXAMPLE/pddrive1.c
--- SuperLU_DIST_3.2.orig/EXAMPLE/pddrive1.c	2012-10-24 21:46:18.000000000 +0200
+++ SuperLU_DIST_3.2/EXAMPLE/pddrive1.c	2012-12-09 12:38:39.581667369 +0100
@@ -14,7 +14,7 @@
 #include "superlu_ddefs.h"
 
 
-int main(int argc, char *argv[])
+void main(int argc, char *argv[])
 /*
  * Purpose
  * =======
diff -aruN SuperLU_DIST_3.2.orig/EXAMPLE/pddrive2_ABglobal.c SuperLU_DIST_3.2/EXAMPLE/pddrive2_ABglobal.c
--- SuperLU_DIST_3.2.orig/EXAMPLE/pddrive2_ABglobal.c	2012-10-24 21:46:18.000000000 +0200
+++ SuperLU_DIST_3.2/EXAMPLE/pddrive2_ABglobal.c	2012-12-09 12:39:38.781668747 +0100
@@ -34,7 +34,7 @@
  * </pre>
  */
 
-int main(int argc, char *argv[])
+void main(int argc, char *argv[])
 {
     superlu_options_t options;
     SuperLUStat_t stat;
diff -aruN SuperLU_DIST_3.2.orig/EXAMPLE/pddrive2.c SuperLU_DIST_3.2/EXAMPLE/pddrive2.c
--- SuperLU_DIST_3.2.orig/EXAMPLE/pddrive2.c	2012-10-24 21:46:18.000000000 +0200
+++ SuperLU_DIST_3.2/EXAMPLE/pddrive2.c	2012-12-09 12:39:27.705668484 +0100
@@ -34,7 +34,7 @@
  * </pre>
  */
 
-int main(int argc, char *argv[])
+void main(int argc, char *argv[])
 {
     superlu_options_t options;
     SuperLUStat_t stat;
diff -aruN SuperLU_DIST_3.2.orig/EXAMPLE/pddrive3_ABglobal.c SuperLU_DIST_3.2/EXAMPLE/pddrive3_ABglobal.c
--- SuperLU_DIST_3.2.orig/EXAMPLE/pddrive3_ABglobal.c	2012-10-24 21:46:18.000000000 +0200
+++ SuperLU_DIST_3.2/EXAMPLE/pddrive3_ABglobal.c	2012-12-09 12:40:04.473669357 +0100
@@ -40,7 +40,7 @@
  * </pre>
  */
 
-int main(int argc, char *argv[])
+void main(int argc, char *argv[])
 {
     superlu_options_t options;
     SuperLUStat_t stat;
diff -aruN SuperLU_DIST_3.2.orig/EXAMPLE/pddrive3.c SuperLU_DIST_3.2/EXAMPLE/pddrive3.c
--- SuperLU_DIST_3.2.orig/EXAMPLE/pddrive3.c	2012-10-24 21:46:18.000000000 +0200
+++ SuperLU_DIST_3.2/EXAMPLE/pddrive3.c	2012-12-09 12:39:54.217669106 +0100
@@ -40,7 +40,7 @@
  * </pre>
  */
 
-int main(int argc, char *argv[])
+void main(int argc, char *argv[])
 {
     superlu_options_t options;
     SuperLUStat_t stat;
diff -aruN SuperLU_DIST_3.2.orig/EXAMPLE/pddrive4_ABglobal.c SuperLU_DIST_3.2/EXAMPLE/pddrive4_ABglobal.c
--- SuperLU_DIST_3.2.orig/EXAMPLE/pddrive4_ABglobal.c	2012-10-24 21:46:18.000000000 +0200
+++ SuperLU_DIST_3.2/EXAMPLE/pddrive4_ABglobal.c	2012-12-09 12:40:31.545669999 +0100
@@ -36,7 +36,7 @@
  * </pre>
  */
 
-int main(int argc, char *argv[])
+void main(int argc, char *argv[])
 {
     superlu_options_t options;
     SuperLUStat_t stat;
diff -aruN SuperLU_DIST_3.2.orig/EXAMPLE/pddrive4.c SuperLU_DIST_3.2/EXAMPLE/pddrive4.c
--- SuperLU_DIST_3.2.orig/EXAMPLE/pddrive4.c	2012-10-24 21:46:19.000000000 +0200
+++ SuperLU_DIST_3.2/EXAMPLE/pddrive4.c	2012-12-09 12:40:17.085669655 +0100
@@ -36,7 +36,7 @@
  * </pre>
  */
 
-int main(int argc, char *argv[])
+void main(int argc, char *argv[])
 {
     superlu_options_t options;
     SuperLUStat_t stat;
diff -aruN SuperLU_DIST_3.2.orig/EXAMPLE/pddrive_ABglobal.c SuperLU_DIST_3.2/EXAMPLE/pddrive_ABglobal.c
--- SuperLU_DIST_3.2.orig/EXAMPLE/pddrive_ABglobal.c	2012-10-24 21:46:18.000000000 +0200
+++ SuperLU_DIST_3.2/EXAMPLE/pddrive_ABglobal.c	2012-12-09 12:44:55.637676245 +0100
@@ -36,7 +36,7 @@
  * </pre>
  */
 
-int main(int argc, char *argv[])
+void main(int argc, char *argv[])
 {
     superlu_options_t options;
     SuperLUStat_t stat;
diff -aruN SuperLU_DIST_3.2.orig/EXAMPLE/pddrive.c SuperLU_DIST_3.2/EXAMPLE/pddrive.c
--- SuperLU_DIST_3.2.orig/EXAMPLE/pddrive.c	2012-10-24 21:46:19.000000000 +0200
+++ SuperLU_DIST_3.2/EXAMPLE/pddrive.c	2012-12-09 12:38:23.569666962 +0100
@@ -36,7 +36,7 @@
  * </pre>
  */
 
-int main(int argc, char *argv[])
+void main(int argc, char *argv[])
 {
     superlu_options_t options;
     SuperLUStat_t stat;
diff -aruN SuperLU_DIST_3.2.orig/EXAMPLE/pzdrive1_ABglobal.c SuperLU_DIST_3.2/EXAMPLE/pzdrive1_ABglobal.c
--- SuperLU_DIST_3.2.orig/EXAMPLE/pzdrive1_ABglobal.c	2012-10-24 21:46:19.000000000 +0200
+++ SuperLU_DIST_3.2/EXAMPLE/pzdrive1_ABglobal.c	2012-12-09 12:41:00.165670680 +0100
@@ -33,7 +33,7 @@
  * </pre>
  */
 
-int main(int argc, char *argv[])
+void main(int argc, char *argv[])
 {
     superlu_options_t options;
     SuperLUStat_t stat;
diff -aruN SuperLU_DIST_3.2.orig/EXAMPLE/pzdrive1.c SuperLU_DIST_3.2/EXAMPLE/pzdrive1.c
--- SuperLU_DIST_3.2.orig/EXAMPLE/pzdrive1.c	2012-10-24 21:46:18.000000000 +0200
+++ SuperLU_DIST_3.2/EXAMPLE/pzdrive1.c	2012-12-09 12:40:51.713670477 +0100
@@ -13,7 +13,7 @@
 #include "superlu_zdefs.h"
 
 
-int main(int argc, char *argv[])
+void main(int argc, char *argv[])
 /*
  * Purpose
  * =======
diff -aruN SuperLU_DIST_3.2.orig/EXAMPLE/pzdrive2_ABglobal.c SuperLU_DIST_3.2/EXAMPLE/pzdrive2_ABglobal.c
--- SuperLU_DIST_3.2.orig/EXAMPLE/pzdrive2_ABglobal.c	2012-10-24 21:46:18.000000000 +0200
+++ SuperLU_DIST_3.2/EXAMPLE/pzdrive2_ABglobal.c	2012-12-09 12:41:14.805671025 +0100
@@ -33,7 +33,7 @@
  * </pre>
  */
 
-int main(int argc, char *argv[])
+void main(int argc, char *argv[])
 {
     superlu_options_t options;
     SuperLUStat_t stat;
diff -aruN SuperLU_DIST_3.2.orig/EXAMPLE/pzdrive2.c SuperLU_DIST_3.2/EXAMPLE/pzdrive2.c
--- SuperLU_DIST_3.2.orig/EXAMPLE/pzdrive2.c	2012-10-24 21:46:18.000000000 +0200
+++ SuperLU_DIST_3.2/EXAMPLE/pzdrive2.c	2012-12-09 12:41:06.929670840 +0100
@@ -33,7 +33,7 @@
  * </pre>
  */
 
-int main(int argc, char *argv[])
+void main(int argc, char *argv[])
 {
     superlu_options_t options;
     SuperLUStat_t stat;
diff -aruN SuperLU_DIST_3.2.orig/EXAMPLE/pzdrive3_ABglobal.c SuperLU_DIST_3.2/EXAMPLE/pzdrive3_ABglobal.c
--- SuperLU_DIST_3.2.orig/EXAMPLE/pzdrive3_ABglobal.c	2012-10-24 21:46:19.000000000 +0200
+++ SuperLU_DIST_3.2/EXAMPLE/pzdrive3_ABglobal.c	2012-12-09 12:41:29.565671428 +0100
@@ -39,7 +39,7 @@
  * </pre>
  */
 
-int main(int argc, char *argv[])
+void main(int argc, char *argv[])
 {
     superlu_options_t options;
     SuperLUStat_t stat;
diff -aruN SuperLU_DIST_3.2.orig/EXAMPLE/pzdrive3.c SuperLU_DIST_3.2/EXAMPLE/pzdrive3.c
--- SuperLU_DIST_3.2.orig/EXAMPLE/pzdrive3.c	2012-10-24 21:46:18.000000000 +0200
+++ SuperLU_DIST_3.2/EXAMPLE/pzdrive3.c	2012-12-09 12:41:21.097671183 +0100
@@ -39,7 +39,7 @@
  * </pre>
  */
 
-int main(int argc, char *argv[])
+void main(int argc, char *argv[])
 {
     superlu_options_t options;
     SuperLUStat_t stat;
diff -aruN SuperLU_DIST_3.2.orig/EXAMPLE/pzdrive4_ABglobal.c SuperLU_DIST_3.2/EXAMPLE/pzdrive4_ABglobal.c
--- SuperLU_DIST_3.2.orig/EXAMPLE/pzdrive4_ABglobal.c	2012-10-24 21:46:18.000000000 +0200
+++ SuperLU_DIST_3.2/EXAMPLE/pzdrive4_ABglobal.c	2012-12-09 12:45:06.477676574 +0100
@@ -35,7 +35,7 @@
  * </pre>
  */
 
-int main(int argc, char *argv[])
+void main(int argc, char *argv[])
 {
     superlu_options_t options;
     SuperLUStat_t stat;
diff -aruN SuperLU_DIST_3.2.orig/EXAMPLE/pzdrive4.c SuperLU_DIST_3.2/EXAMPLE/pzdrive4.c
--- SuperLU_DIST_3.2.orig/EXAMPLE/pzdrive4.c	2012-10-24 21:46:18.000000000 +0200
+++ SuperLU_DIST_3.2/EXAMPLE/pzdrive4.c	2012-12-09 12:41:37.705671569 +0100
@@ -35,7 +35,7 @@
  * </pre>
  */
 
-int main(int argc, char *argv[])
+void main(int argc, char *argv[])
 {
     superlu_options_t options;
     SuperLUStat_t stat;
diff -aruN SuperLU_DIST_3.2.orig/EXAMPLE/pzdrive_ABglobal.c SuperLU_DIST_3.2/EXAMPLE/pzdrive_ABglobal.c
--- SuperLU_DIST_3.2.orig/EXAMPLE/pzdrive_ABglobal.c	2012-10-24 21:46:18.000000000 +0200
+++ SuperLU_DIST_3.2/EXAMPLE/pzdrive_ABglobal.c	2012-12-09 12:45:18.765676814 +0100
@@ -35,7 +35,7 @@
  * </pre>
  */
 
-int main(int argc, char *argv[])
+void main(int argc, char *argv[])
 {
     superlu_options_t options;
     SuperLUStat_t stat;
diff -aruN SuperLU_DIST_3.2.orig/EXAMPLE/pzdrive.c SuperLU_DIST_3.2/EXAMPLE/pzdrive.c
--- SuperLU_DIST_3.2.orig/EXAMPLE/pzdrive.c	2012-10-24 21:46:18.000000000 +0200
+++ SuperLU_DIST_3.2/EXAMPLE/pzdrive.c	2012-12-09 12:40:45.297670325 +0100
@@ -35,7 +35,7 @@
  * </pre>
  */
 
-int main(int argc, char *argv[])
+void main(int argc, char *argv[])
 {
     superlu_options_t options;
     SuperLUStat_t stat;
