--- superlu_dist-4.0/make.inc	2014-10-11 12:01:16.000000000 -0700
+++ superlu_dist-4.0.patch/make.inc	2015-08-06 09:31:43.000000000 -0700
@@ -30,13 +30,13 @@
 #
 ############################################################################
 ## parmetis 4.x.x, 32-bit integer
-PARMETIS_DIR	:= ${HOME}/Carver/lib/parmetis-4.0.3
+#PARMETIS_DIR	:= ${HOME}/Carver/lib/parmetis-4.0.3
 ## parmetis 4.x.x, 64-bit integer
 # PARMETIS_DIR	:= ${HOME}/Carver/lib/parmetis-4.0.3_64
 
-METISLIB := -L${PARMETIS_DIR}/build/Linux-x86_64/libmetis -lmetis
-PARMETISLIB := -L${PARMETIS_DIR}/build/Linux-x86_64/libparmetis -lparmetis
-I_PARMETIS := -I${PARMETIS_DIR}/include -I${PARMETIS_DIR}/metis/include
+METISLIB := -L${METIS_LIB} -lmetis
+#PARMETISLIB := -L${PARMETIS_DIR}/build/Linux-x86_64/libparmetis -lparmetis
+#I_PARMETIS := -I${PARMETIS_DIR}/include -I${PARMETIS_DIR}/metis/include
 ############################################################################
 
 # Define the required Fortran libraries, if you use C compiler to link
@@ -67,13 +67,13 @@
 # C compiler setup
 CC           	= mpicc
 # CFLAGS should be set to be the C flags that include optimization
-CFLAGS          = ${CUDA_FLAGS} ${INCS} -std=c99 -O3 -Wall -w2 -mkl -openmp \
+CFLAGS          = $(RPM_OPT_FLAGS) -fPIC ${CUDA_FLAGS} ${INCS} -std=c99 -O3 -Wall \
 		-DDEBUGlevel=0 -DPRNTlevel=1 -DPROFlevel=0 \
 #		-D_LONGINT 
 #	-Wunused-variable 
 #
 # NOOPTS should be set to be the C flags that turn off any optimization
-NOOPTS		= -O0
+NOOPTS		= $(RPM_OPT_FLAGS) -Os -fPIC
 ############################################################################
 # FORTRAN compiler setup
 FORTRAN         = mpif90
