--- superlu_dist-4.1/SRC/Makefile	2015-07-17 19:54:17.000000000 -0700
+++ superlu_dist-4.1.patch/SRC/Makefile	2015-08-19 15:05:47.000000000 -0700
@@ -32,7 +32,7 @@
 ALLAUX 	= sp_ienv.o etree.o sp_colorder.o get_perm_c.o \
 	  mmd.o comm.o memory.o util.o superlu_grid.o \
 	  pxerbla.o superlu_timer.o GetDiagU.o symbfact.o \
-	  psymbfact.o psymbfact_util.o get_perm_c_parmetis.o mc64ad_dist.o \
+	  psymbfact.o psymbfact_util.o mc64ad_dist.o \
 	  static_schedule.o
 
 ifeq "${ACC}" "GPU"
--- superlu_dist-4.1/SRC/pdgssvx.c	2015-07-17 19:54:17.000000000 -0700
+++ superlu_dist-4.1.patch/SRC/pdgssvx.c	2015-08-19 15:05:47.000000000 -0700
@@ -931,9 +931,9 @@
 	       * and does not modify it.  It also allocates memory for       *
 	       * sizes[] and fstVtxSep[] arrays, that contain information    *
 	       * on the separator tree computed by ParMETIS.                 */
-	      flinfo = get_perm_c_parmetis(A, perm_r, perm_c, nprocs_num,
+	      flinfo = 0;/*get_perm_c_parmetis(A, perm_r, perm_c, nprocs_num,
                                   	   noDomains, &sizes, &fstVtxSep,
-                                           grid, &symb_comm);
+                                           grid, &symb_comm);*/
 	      if (flinfo > 0)
 	          ABORT("ERROR in get perm_c parmetis.");
 	  } else {
--- superlu_dist-4.1/SRC/pzgssvx.c	2015-07-17 19:54:17.000000000 -0700
+++ superlu_dist-4.1.patch/SRC/pzgssvx.c	2015-08-19 15:05:47.000000000 -0700
@@ -932,9 +932,9 @@
 	       * and does not modify it.  It also allocates memory for       *
 	       * sizes[] and fstVtxSep[] arrays, that contain information    *
 	       * on the separator tree computed by ParMETIS.                 */
-	      flinfo = get_perm_c_parmetis(A, perm_r, perm_c, nprocs_num,
+	      flinfo = 0;/*get_perm_c_parmetis(A, perm_r, perm_c, nprocs_num,
                                   	   noDomains, &sizes, &fstVtxSep,
-                                           grid, &symb_comm);
+                                           grid, &symb_comm);*/
 	      if (flinfo > 0)
 	          ABORT("ERROR in get perm_c parmetis.");
 	  } else {
