--- superlu_dist-4.1/make.inc	2015-08-05 15:56:21.000000000 -0700
+++ superlu_dist-4.1.patch/make.inc	2015-08-20 10:57:13.000000000 -0700
@@ -31,13 +31,14 @@
 #
 ############################################################################
 ## parmetis 4.x.x, 32-bit integer
- PARMETIS_DIR	:= ${HOME}/Edison/lib/parmetis-4.0.3
+# PARMETIS_DIR	:= ${HOME}/Edison/lib/parmetis-4.0.3
 ## parmetis 4.x.x, 64-bit integer
 # PARMETIS_DIR	:= ${HOME}/Edison/lib/parmetis-4.0.3_64
 
-METISLIB := -L${PARMETIS_DIR}/build/Linux-x86_64/libmetis -lmetis
-PARMETISLIB := -L${PARMETIS_DIR}/build/Linux-x86_64/libparmetis -lparmetis
-I_PARMETIS := -I${PARMETIS_DIR}/include -I${PARMETIS_DIR}/metis/include
+METISLIB := -L${METIS_LIB} -lmetis
+I_METIS := -I${METIS_inc}
+#PARMETISLIB := -L${PARMETIS_DIR}/build/Linux-x86_64/libparmetis -lparmetis
+#I_PARMETIS := -I${PARMETIS_DIR}/include -I${PARMETIS_DIR}/metis/include
 ############################################################################
 
 # Define the required Fortran libraries, if you use C compiler to link
@@ -57,19 +58,19 @@
 
 ############################################################################
 # C compiler setup
-CC           	= cc
+CC           	= mpicc
 # CFLAGS should be set to be the C flags that include optimization
-CFLAGS          = -fast -m64 -std=c99 -Wall -openmp \
-		$(I_PARMETIS) -DDEBUGlevel=0 -DPRNTlevel=0 -DPROFlevel=0 \
+CFLAGS          = -m64 -std=c99 -Wall -openmp \
+		$(I_METIS) -fPIC -DDEBUGlevel=0 -DPRNTlevel=0 -DPROFlevel=0 \
 # uncomment the following to use 64-bit integer
 # CFLAGS 		+= -D_LONGINT
 
 # NOOPTS should be set to be the C flags that turn off any optimization
-NOOPTS		= -O0
+NOOPTS		= -Os -fPIC
 ############################################################################
 # FORTRAN compiler setup
-FORTRAN         = ftn
-F90FLAGS	= -fast #-Mipa=fast,safe
+FORTRAN         = mpif90
+F90FLAGS	= 
 # uncomment the following to use 64-bit integer
 # F90FLAGS	+= -i8
 ############################################################################
