--- SuperLU_DIST_5.1.0/SRC/Makefile	2016-05-15 22:16:44.000000000 -0700
+++ SuperLU_DIST_5.1.0.patch/SRC/Makefile	2016-10-07 11:01:47.000000000 -0700
@@ -32,7 +32,7 @@
 ALLAUX 	= sp_ienv.o etree.o sp_colorder.o get_perm_c.o \
 	  mmd.o comm.o memory.o util.o superlu_grid.o \
 	  pxerr_dist.o superlu_timer.o symbfact.o \
-	  psymbfact.o psymbfact_util.o get_perm_c_parmetis.o mc64ad_dist.o \
+	  psymbfact.o psymbfact_util.o mc64ad_dist.o \
 	  static_schedule.o xerr_dist.o smach_dist.o dmach_dist.o
 
 ifeq "${ACC}" "GPU"
