--- etc/slurmctld.service.in.orig	2014-11-13 10:57:07.334929035 -0800
+++ etc/slurmctld.service.in	2014-11-13 10:57:18.575054329 -0800
@@ -5,9 +5,9 @@
 
 [Service]
 Type=forking
-EnvironmentFile=/etc/default/slurmctld
+###EnvironmentFile=/etc/default/slurmctld
 ExecStart=@sbindir@/slurmctld $SLURMCTLD_OPTIONS
-PIDFile=/var/run/slurm/slurmctld.pid
+PIDFile=/var/run/slurmctld.pid
 
 [Install]
 WantedBy=multi-user.target
