--- etc/init.d.slurm.in	2014-07-16 15:48:18.000000000 -0700
+++ etc/init.d.slurm.in	2014-11-10 17:51:55.682652184 -0800
@@ -5,12 +5,6 @@
 #              manages exclusive access to a set of compute \
 #              resources and distributes work to those resources.
 #
-# processname: @sbindir@/slurmd
-# pidfile: /var/run/slurmd.pid
-#
-# processname: @sbindir@/slurmctld
-# pidfile: /var/run/slurmctld.pid
-#
 # config: /etc/sysconfig/slurm
 #
 ### BEGIN INIT INFO
@@ -19,8 +13,8 @@
 # Required-Stop:     $remote_fs $syslog $network munge
 # Should-Start:      $named
 # Should-Stop:       $named
-# Default-Start:     2 3 4 5
-# Default-Stop:      0 1 6
+# Default-Start:     3 5
+# Default-Stop:      0 1 2 6
 # Short-Description: slurm daemon management
 # Description:       Start slurm to provide resource management
 ### END INIT INFO
@@ -93,7 +87,7 @@
     shift
     echo -n "starting $prog: "
     unset HOME MAIL USER USERNAME
-    $STARTPROC $SBINDIR/$prog $*
+    $STARTPROC -p $prog.pid $SBINDIR/$prog $*
     rc_status -v
     echo
     touch /var/lock/subsys/slurm
