--- unixccompiler.py    2018-02-20 11:19:24.339194396 -0600
+++ unixccompiler.py~   2018-02-20 12:23:32.927930210 -0600
@@ -80,16 +80,6 @@
     if sys.platform == "cygwin":
         exe_extension = ".exe"
 
-    def _fix_lib_args(self, libraries, library_dirs, runtime_library_dirs):
-        """Remove standard library path from rpath"""
-        libraries, library_dirs, runtime_library_dirs = super(
-            self.__class__, self)._fix_lib_args(libraries, library_dirs,
-            runtime_library_dirs)
-        libdir = sysconfig.get_config_var('LIBDIR')
-        if runtime_library_dirs and (libdir in runtime_library_dirs):
-            runtime_library_dirs.remove(libdir)
-        return libraries, library_dirs, runtime_library_dirs
-
     def preprocess(self, source, output_file=None, macros=None,
                    include_dirs=None, extra_preargs=None, extra_postargs=None):
         fixed_args = self._fix_compile_args(None, macros, include_dirs)
