--- warewulf-cluster-3.7/libexec/wwinit/50-ntpd.init	2017-02-22 09:32:01.000000000 -0800
+++ warewulf-cluster-3.7.patch/libexec/wwinit/50-ntpd.init	2017-02-22 13:38:12.000000000 -0800
@@ -75,9 +75,11 @@
 
 # Specify the key identifiers which are trusted.
 #trustedkey 4 8 42
+trustedkey 1
 
 # Specify the key identifier to use with the ntpdc utility.
 #requestkey 8
+requestkey 1
 
 # Specify the key identifier to use with the ntpq utility.
 #controlkey 8
