--- a/messageDir/en.lua	2017-11-16 15:23:41.000000000 -0800
+++ b/messageDir/en.lua	2018-01-16 10:00:36.000000000 -0800
@@ -111,7 +111,6 @@
 
   $  module swap %{oldName} %{fullName}
 
-Please submit a consulting ticket if you require additional assistance.
 ]==],
      e_Illegal_Load       = [==[The following module(s) are illegal: %{module_list}
 Lmod does not support modulefiles that start with two or more underscores
