From a9f56694c2b9e2d6003607d975a0a8312dc6dbc2 Mon Sep 17 00:00:00 2001
From: Paul Osmialowski <pawel.osmialowski@arm.com>
Date: Sat, 27 May 2017 06:23:58 +0100
Subject: [PATCH 2/2] build: Flang has the same interface as PGI

Signed-off-by: Paul Osmialowski <pawel.osmialowski@arm.com>
---
 common_x86_64.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/common_x86_64.h b/common_x86_64.h
index 11937b41..bee88d3c 100644
--- a/common_x86_64.h
+++ b/common_x86_64.h
@@ -245,6 +245,10 @@ static __inline int blas_quickdivide(unsigned int x, unsigned int y){
 #define RETURN_BY_STACK
 #endif
 
+#ifdef F_INTERFACE_FLANG
+#define RETURN_BY_STACK
+#endif
+
 #ifdef F_INTERFACE_PGI
 #define RETURN_BY_STACK
 #endif
-- 
2.13.0

