--- a/3rd_party/Makefile.am	2024-03-18 07:37:20.581000000 +0000
+++ b/3rd_party/Makefile.am	2024-03-18 07:34:07.021000000 +0000
@@ -9,7 +9,8 @@
 IPXE_VERSION = 09e8a15
 IPXE_SOURCE = $(top_srcdir)/3rd_party/GPL/ipxe-$(IPXE_VERSION).tar.xz
 IPXE_DIR = ipxe-$(IPXE_VERSION)
-
+# https://github.com/ipxe/ipxe/pull/1036
+IPXE_PATCH1 = GPL/1036.patch
 
 prep:
 	@ if [ ! -d "_work/$(IPXE_DIR)" ]; then \
@@ -18,6 +19,7 @@
 		tar -xJf $(IPXE_SOURCE) -C _work/ ;\
 		echo '#define DOWNLOAD_PROTO_HTTPS' >> "_work/$(IPXE_DIR)/src/config/local/general.h" ;\
 		echo '#define NTP_CMD' >> "_work/$(IPXE_DIR)/src/config/local/general.h" ;\
+		( cd _work/$(IPXE_DIR) && pwd && patch -p1 -i ../../$(IPXE_PATCH1) ) \
 	fi
 
 bin-i386-pcbios/undionly.kpxe: prep
