--- a/configure.ac	2018-07-11 13:56:05.000000000 -0700
+++ b/configure.ac	2018-07-11 15:07:55.000000000 -0700
@@ -37,6 +37,9 @@
 AC_MSG_CHECKING([for Debian based system])
 AM_CONDITIONAL(ISDEB, [test -f /etc/debian_version])
 
+AC_MSG_CHECKING([for SUSE based system])
+AM_CONDITIONAL(ISSUSE, [test -f /etc/SuSE-release])
+
 AC_PATH_PROG(PERLBIN, perl)
 if test -z "$PERLBIN"; then
    AC_MSG_ERROR([perl not found])
--- a/etc/Makefile.am	2018-07-11 13:56:05.000000000 -0700
+++ b/etc/Makefile.am	2018-07-11 20:16:02.000000000 -0700
@@ -4,7 +4,7 @@
 bash_completiondir = $(sysconfdir)/bash_completion.d/
 
 # Check for Debian
-if ISDEB
+if ISDEB || ISSUSE
 httpdir = $(sysconfdir)/apache2/conf.d/
 else
 # Assume RHEL Layout
