--- warewulf-provision-3.7/configure	2017-02-22 13:14:14.000000000 -0800
+++ warewulf-provision-3.7.patch/configure	2017-02-22 13:14:58.000000000 -0800
@@ -2423,7 +2423,10 @@
 
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for Debian based system" >&5
 $as_echo_n "checking for Debian based system... " >&6; }
- if test -f /etc/debian_version; then
+if test -f /etc/debian_version; then
+  ISDEB_TRUE=
+  ISDEB_FALSE='#'
+elif test -f /etc/SuSE-release; then
   ISDEB_TRUE=
   ISDEB_FALSE='#'
 else
