diff --git a/lib/Warewulf/Bootstrap.pm b/lib/Warewulf/Bootstrap.pm
index ff2ad21..daf794a 100644
--- a/lib/Warewulf/Bootstrap.pm
+++ b/lib/Warewulf/Bootstrap.pm
@@ -15,7 +15,7 @@ use Warewulf::Logger;
 use Warewulf::DataStore;
 use Warewulf::Util;
 use File::Basename;
-use File::Path;
+use File::Path qw(make_path);
 use Digest::MD5 qw(md5_hex);
 use POSIX qw(uname);
 
@@ -211,6 +211,9 @@ bootstrap_export()
 
             if (! -d $dirname) {
                 mkpath($dirname);
+                make_path($dirname, {
+                    chmod => 0755,
+                });
             }
         }
 
@@ -350,7 +353,9 @@ build_local_bootstrap()
             }
 
             mkpath($tmpdir);
-            mkpath($bootstrapdir);
+            make_path($bootstrapdir, {
+                chmod => 0755,
+            });
             chdir($tmpdir);
 
             &dprint("Opening gunzip/cpio pipe\n");
@@ -381,10 +386,13 @@ build_local_bootstrap()
             system("cd $tmpdir/initramfs; find . | cpio -o --quiet -H newc -F $bootstrapdir/initfs");
             &nprint("Compressing the initramfs\n");
             system("gzip -f -9 $bootstrapdir/initfs");
+            chmod(0644, "$bootstrapdir/initfs.gz");
             &nprint("Locating the kernel object\n");
             system("cp $tmpdir/kernel $bootstrapdir/kernel");
+            chmod(0644, "$bootstrapdir/kernel");
             system("rm -rf $tmpdir");
             open(COOKIE, "> $bootstrapdir/cookie");
+            chmod(0644, "$bootstrapdir/cookie");
             print COOKIE $self->checksum();
             close COOKIE;
             &nprint("Bootstrap image '$bootstrap_name' is ready\n");
diff --git a/lib/Warewulf/Provision/Pxe.pm b/lib/Warewulf/Provision/Pxe.pm
index c9f6a7b..bd3e0ac 100644
--- a/lib/Warewulf/Provision/Pxe.pm
+++ b/lib/Warewulf/Provision/Pxe.pm
@@ -18,7 +18,7 @@ use Warewulf::Network;
 use Warewulf::DataStore;
 use Warewulf::Provision::Tftp;
 use File::Basename;
-use File::Path;
+use File::Path qw(make_path);
 use POSIX qw(uname);
 
 our @ISA = ('Warewulf::Object');
@@ -94,8 +94,11 @@ setup()
                 if (-f "$datadir/warewulf/ipxe/$f") {
                     &iprint("Copying $f to the tftp root\n");
                     my $dirname = dirname("$tftpdir/warewulf/ipxe/$f");
-                    mkpath($dirname);
+                    make_path($dirname, {
+                        chmod => 0755,
+                    });
                     system("cp $datadir/warewulf/ipxe/$f $tftpdir/warewulf/ipxe/$f");
+                    chmod(0644, "$tftpdir/warewulf/ipxe/$f");
                 } elsif ($arch eq "x86_64") {
                     &eprint("Could not locate Warewulf's internal $datadir/warewulf/ipxe/$f! Things might be broken!\n");
                 }
@@ -106,8 +109,11 @@ setup()
                 if (-f "$datadir/warewulf/ipxe/$f") {
                     &iprint("Copying $f to the tftp root\n");
                     my $dirname = dirname("$tftpdir/warewulf/ipxe/$f");
-                    mkpath($dirname);
+                    make_path($dirname, {
+                        chmod => 0755,
+                    });
                     system("cp $datadir/warewulf/ipxe/$f $tftpdir/warewulf/ipxe/$f");
+                    chmod(0644, "$tftpdir/warewulf/ipxe/$f");
                 } elsif ($arch eq "aarch64") {
                     &eprint("Could not locate Warewulf's internal $datadir/warewulf/ipxe/$f! Things might be broken!\n");
                 }
@@ -154,7 +160,9 @@ update()
 
     if (! -d "$statedir/warewulf/ipxe/cfg") {
         &iprint("Creating ipxe configuration directory: $statedir/warewulf/ipxe/cfg");
-        mkpath("$statedir/warewulf/ipxe/cfg");
+        make_path("$statedir/warewulf/ipxe/cfg", {
+            chmod => 0755,
+        });
     }
 
     foreach my $nodeobj (@nodeobjs) {
@@ -322,6 +330,7 @@ update()
                     if (! close IPXE) {
                         &eprint("Could not write iPXE configuration file: $!\n");
                     }
+		    chmod(0644, "$statedir/warewulf/ipxe/cfg/$config");
                 } else {
                     &eprint("Node: $nodename-$devname: Bad characters in hwaddr: '$hwaddr'\n");
                 }
@@ -372,6 +381,9 @@ delete()
                 if (-f "$statedir/warewulf/ipxe/cfg/$config") {
                     unlink("$statedir/warewulf/ipxe/cfg/$config");
                 }
+                if (! chmod 0644, "$statedir/warewulf/ipxe/cfg/$config") {
+                    &eprint("Could not chmod Pxelinux configuration file: $!\n");
+                }
             } else {
                 &eprint("Bad characters in hwaddr: $hwaddr\n");
             }
