--- a/initramfs/Makefile.am	2019-06-17 14:01:40.000000000 -0700
+++ b/initramfs/Makefile.am	2019-06-25 13:47:36.842849489 -0700
@@ -24,7 +24,7 @@
 PARTED_VERSION = 3.2
 PARTED_SOURCE = $(top_srcdir)/3rd_party/GPL/parted-$(PARTED_VERSION).tar.xz
 PARTED_DIR = parted-$(PARTED_VERSION)
-PARTED_CONFIGARGS = --prefix=/usr --without-readline --disable-static --disable-nls --disable-dynamic-loading --disable-debug
+PARTED_CONFIGARGS = --prefix=/usr --libdir=/usr/lib64 --without-readline --disable-static --disable-nls --disable-dynamic-loading --disable-debug
 
 e2fsprogs:
 	@ if [ -n "@local_e2fsprogs_path@" -a -f "@local_e2fsprogs_path@" ]; then \
