diff -ruN warewulf-provision-3.6.old/initramfs/Makefile.am warewulf-provision-3.6/initramfs/Makefile.am
--- warewulf-provision-3.6.old/initramfs/Makefile.am	2014-05-21 19:13:01.000000000 +0100
+++ warewulf-provision-3.6/initramfs/Makefile.am	2016-09-17 19:31:43.503452761 +0100
@@ -23,7 +23,7 @@
 	fi
 	@ if [ ! -f "_work/$(E2FSPROGS_DIR)/misc/mke2fs" ]; then \
 		echo "Building e2fsprogs" ;\
-		(cd _work/$(E2FSPROGS_DIR)/; ./configure $(E2FSPROGS_CONFIGARGS)) ;\
+		(cp /usr/lib/rpm/config.guess _work/$(E2FSPROGS_DIR)/config ; cd _work/$(E2FSPROGS_DIR)/; ./configure $(E2FSPROGS_CONFIGARGS)) ;\
 		make -C _work/$(E2FSPROGS_DIR);\
 	fi
 
diff -ruN warewulf-provision-3.6.old/initramfs/Makefile.in warewulf-provision-3.6/initramfs/Makefile.in
--- warewulf-provision-3.6.old/initramfs/Makefile.in	2014-05-28 19:04:59.000000000 +0100
+++ warewulf-provision-3.6/initramfs/Makefile.in	2016-09-17 19:34:03.898261345 +0100
@@ -527,7 +527,7 @@
 	fi
 	@ if [ ! -f "_work/$(E2FSPROGS_DIR)/misc/mke2fs" ]; then \
 		echo "Building e2fsprogs" ;\
-		(cd _work/$(E2FSPROGS_DIR)/; ./configure $(E2FSPROGS_CONFIGARGS)) ;\
+		(cp /usr/lib/rpm/config.guess _work/$(E2FSPROGS_DIR)/config; cd _work/$(E2FSPROGS_DIR)/; ./configure $(E2FSPROGS_CONFIGARGS)) ;\
 		make -C _work/$(E2FSPROGS_DIR);\
 	fi
 
