--- warewulf-provision-3.7/initramfs/capabilities/setup-filesystems/80-mkbootable	2017-02-22 09:32:02.000000000 -0800
+++ warewulf-provision-3.7.patch/initramfs/capabilities/setup-filesystems/80-mkbootable	2017-02-22 13:12:08.000000000 -0800
@@ -7,6 +7,11 @@
 # required approvals from the U.S. Dept. of Energy).  All rights reserved.
 #
 
+# SLES requires /proc in the chroot
+if [ ! -f $NEWROOT/proc/self/mountinfo ]; then
+    mount --bind /proc $NEWROOT/proc
+    PROCMOUNT=1
+fi
 
 # Install a bootloader if $WWBOOTLOADER is set and the root device is known
 if [ -n "$WWBOOTLOADER" -a -f "/tmp/rootdev" ]; then
@@ -20,6 +25,9 @@
             if [ -x "$NEWROOT/sbin/dracut" ]; then
                 chroot $NEWROOT /sbin/dracut --force '' $KERNELVERSION
                 INITRD=`cd $NEWROOT; find boot/initr*-${KERNELVERSION}.img* 2>/dev/null | tail -n 1`
+            elif [ -x "$NEWROOT/usr/bin/dracut" ]; then
+                chroot $NEWROOT /usr/bin/dracut --force '' $KERNELVERSION
+                INITRD=`cd $NEWROOT; find boot/initr*-${KERNELVERSION}.img* 2>/dev/null | tail -n 1`
             elif [ -x "$NEWROOT/sbin/mkinitrd" ]; then
                 INITRD=`cd $NEWROOT; find boot/initr*-${KERNELVERSION}.img* 2>/dev/null | tail -n 1`
                 if [ -n "$INITRD" ]; then
@@ -162,5 +170,10 @@
     exit 1
 fi
 
+# Clean up SLES /proc in the chroot
+if [ $PROCMOUNT -eq 1 ]; then
+    umount $NEWROOT/proc
+fi
+
 # vim: filetype=sh:syntax=sh:expandtab:ts=4:sw=4:
 
