--- a/provision/initramfs/Makefile.in.orig	2017-03-10 10:18:39.101375078 -0500
+++ b/provision/initramfs/Makefile.in	2017-03-10 10:19:33.921378704 -0500
@@ -282,6 +282,14 @@
 MAINTAINERCLEANFILES = Makefile.in
 all: all-recursive
 
+ARCH := $(shell uname -m)
+
+ifeq ($(ARCH),ppc64le)
+	LOADER :=ld64
+else
+	LOADER :=ld-linux
+endif
+
 .SUFFIXES:
 $(srcdir)/Makefile.in:  $(srcdir)/Makefile.am  $(am__configure_deps)
 	@for dep in $?; do \
--- warewulf3-3.7pre/provision/initramfs/Makefile.in.orig	2017-03-13 11:43:02.946811424 -0400
+++ warewulf3-3.7pre/provision/initramfs/Makefile.in	2017-03-13 11:48:46.216827808 -0400
@@ -642,6 +642,12 @@
 	fi
 	@ if [ ! -f "_work/$(LIBARCHIVE_DIR)/" ]; then \
 		echo "Building libarchive" ;\
+		if [ -f "/usr/lib/rpm/config.guess" ]; then \
+			cp /usr/lib/rpm/config.guess _work/$(LIBARCHIVE_DIR)/build/autoconf;\
+		fi; \
+		if [ -f "/usr/lib/rpm/config.sub" ]; then \
+			cp /usr/lib/rpm/config.sub _work/$(LIBARCHIVE_DIR)/build/autoconf;\
+		fi; \
 		(cd _work/$(LIBARCHIVE_DIR)/; ./configure $(LIBARCHIVE_CONFIGARGS)) ;\
 		$(MAKE) -C _work/$(LIBARCHIVE_DIR);\
 	fi
@@ -683,8 +683,8 @@
 	ln -s mkfs.ext4 rootfs/sbin/mkfs.ext3
 	cp -a _work/$(LIBARCHIVE_DIR)/bsdtar rootfs/bin/bsdtar
 	$(MAKE) -C _work/$(PARTED_DIR)/ DESTDIR=`pwd`/rootfs install
-	cp -L --parents /lib*/ld-linux* rootfs/
-	find rootfs -type f -perm -o+x -print | grep -v ld-linux | xargs ldd | grep "=>" | awk '{print $$3}' | grep "^/" | sort | uniq | while read i; do cp -L --parents $$i rootfs/ && chmod 755 rootfs/$$i; done
+	cp -L --parents /lib*/$(LOADER)* rootfs/
+	find rootfs -type f -perm -o+x -print | grep -v $(LOADER) | xargs ldd | grep "=>" | awk '{print $$3}' | grep "^/" | sort | uniq | while read i; do cp -L --parents $$i rootfs/ && chmod 755 rootfs/$$i; done
 	rm -f rootfs/linuxrc rootfs/lib64/*.la rootfs/lib/*.la rootfs/usr/lib64/*.la rootfs/usr/lib/*.la
 	rm -rf rootfs/usr/share rootfs/usr/include rootfs/usr/lib/pkgconfig
 	find -type d \! -perm -u=w -exec chmod u+w {} \;
