diff --git i/lib/Warewulf/Provision/Pxe.pm w/lib/Warewulf/Provision/Pxe.pm
index fd20408..d5c59a3 100644
--- i/lib/Warewulf/Provision/Pxe.pm
+++ w/lib/Warewulf/Provision/Pxe.pm
@@ -360,7 +360,9 @@ update()
                         } else {
                             print IPXE "kernel \${base}/$arch/$bootstrapid/kernel ro initrd=initfs.gz wwhostname=$hostname wwtransport=$transport ";
                         }
-                        print IPXE join(" ", @kargs) . " ";
+                        if ($kargs[0]) { 
+                            print IPXE join(" ", @kargs) . " ";
+                        }
                         if ($console) {
                             print IPXE "console=tty0 console=$console ";
                         }
