--- warewulf3-3.7pre.ohpc1.3/lib/Warewulf/Provision/Pxelinux.pm	2017-03-22 08:41:20.000000000 -0700
+++ warewulf3-3.7pre.ohpc1.3.patch/lib/Warewulf/Provision/Pxelinux.pm	2017-08-04 11:14:31.000000000 -0700
@@ -89,8 +89,11 @@
             if (! -f "$tftpdir/warewulf/$f") {
                 if (-f "$datadir/warewulf/$f") {
                     &iprint("Copying $f to the tftp root\n");
-                    mkpath("$tftpdir/warewulf/");
+                    mkpath("$tftpdir/warewulf/", {
+                        mode => 0755,
+                    });
                     system("cp $datadir/warewulf/$f $tftpdir/warewulf/$f");
+                    chmod 0644, "$tftpdir/warewulf/$f";
                 } else {
                     &eprint("Could not locate Warewulf's internal $f! Things might be broken!\n");
                 }
@@ -138,7 +141,9 @@
 
     if (! -d "$tftproot/warewulf/pxelinux.cfg") {
         &iprint("Creating pxelinux configuration directory: $tftproot/warewulf/pxelinux.cfg");
-        mkpath("$tftproot/warewulf/pxelinux.cfg");
+        mkpath("$tftproot/warewulf/pxelinux.cfg", {
+            mode => 0755,
+        });
     }
 
     foreach my $nodeobj (@nodeobjs) {
@@ -279,6 +284,9 @@
                 if (! close PXELINUX) {
                     &eprint("Could not write Pxelinux configuration file: $!\n");
                 }
+                if (! chmod 0644, "$tftproot/warewulf/pxelinux.cfg/$config") {
+                    &eprint("Could not chmod Pxelinux configuration file: $!\n");
+                }
             } else {
                 &eprint("Node: $nodename-$devname: Bad characters in hwaddr: '$hwaddr'\n");
             }
--- warewulf3-3.7pre.ohpc1.3/lib/Warewulf/Bootstrap.pm	2017-03-22 08:41:20.000000000 -0700
+++ warewulf3-3.7pre.ohpc1.3.patch/lib/Warewulf/Bootstrap.pm	2017-08-07 15:47:34.000000000 -0700
@@ -186,7 +186,9 @@
             my $dirname = dirname($file);
 
             if (! -d $dirname) {
-                mkpath($dirname);
+                mkpath($dirname, {
+                    mode => 0755,
+                });
             }
         }
 
@@ -318,7 +320,9 @@
             }
 
             mkpath($tmpdir);
-            mkpath($bootstrapdir);
+            mkpath($bootstrapdir, {
+                mode => 0755,
+            });
             chdir($tmpdir);
 
             &dprint("Opening gunzip/cpio pipe\n");
@@ -348,12 +352,15 @@
             system("cd $tmpdir/initramfs; find . | cpio -o --quiet -H newc -F $bootstrapdir/initfs");
             &nprint("Compressing the initramfs\n");
             system("gzip -f -9 $bootstrapdir/initfs");
+            chmod 0644, "$bootstrapdir/initfs.gz";
             &nprint("Locating the kernel object\n");
             system("cp $tmpdir/kernel $bootstrapdir/kernel");
+            chmod 0644, "$bootstrapdir/kernel";
             system("rm -rf $tmpdir");
             open(COOKIE, "> $bootstrapdir/cookie");
             print COOKIE $self->checksum();
             close COOKIE;
+            chmod 0644, "$bootstrapdir/cookie";
             &nprint("Bootstrap image '$bootstrap_name' is ready\n");
         } else {
             &dprint("Bootstrap ID is invalid\n");
