--- warewulf-provision-3.6/lib/Warewulf/Provision/Dhcp/Isc.pm	2014-05-21 11:13:01.000000000 -0700
+++ warewulf-provision-3.6.patch/lib/Warewulf/Provision/Dhcp/Isc.pm	2015-06-03 09:00:17.000000000 -0700
@@ -229,8 +229,9 @@
             &eprint("No DB ID associated with this node object object: $hostname/$nodename:$n\n");
             next;
         }
-        $nodename =~ s/\./_/g;
         &dprint("Evaluating node: $nodename (object ID: $db_id)\n");
+        $dhcpd_contents .= "   # Evaluating Warewulf node: $nodename (DB ID:$db_id)\n";
+        $nodename =~ s/\./_/g;
         my @bootservers = $n->get("bootserver");
         if (! @bootservers or scalar(grep { $_ eq $ipaddr} @bootservers)) {
             my $clustername = $n->cluster();
@@ -269,32 +270,38 @@
                 my $node_testnetwork = $netobj->calc_network($node_ipaddr, $node_netmask);
 
                 if (! $hwaddr) {
-                    &dprint("Skipping DHCP config for $nodename-$devname (no defined HWADDR)\n");
+                    &iprint("Skipping DHCP config for $nodename-$devname (no defined HWADDR)\n");
+                    $dhcpd_contents .= "   # Skipping $nodename-$devname: No defined HWADDR\n";
                     next;
                 }
 
                 if (! $node_ipaddr) {
-                    &dprint("Skipping DHCP config for $nodename-$devname (no defined IPADDR)\n");
+                    &iprint("Skipping DHCP config for $nodename-$devname (no defined IPADDR)\n");
+                    $dhcpd_contents .= "   # Skipping $nodename-$devname: No defined IPADDR\n";
                     next;
                 }
 
                 if ($node_testnetwork ne $network) {
                     &iprint("Skipping DHCP config for $nodename-$devname (on a different network)\n");
+                    $dhcpd_contents .= "   # Skipping $nodename-$devname: Not on boot network ($node_testnetwork)\n";
                     next;
                 }
 
                 if (exists($seen{"NODESTRING"}) and exists($seen{"NODESTRING"}{"$nodename-$devname"})) {
                     my $redundant_node = $seen{"NODESTRING"}{"$nodename-$devname"};
+                    $dhcpd_contents .= "   # Skipping $nodename-$devname: duplicate nodename-netdev\n";
                     &iprint("Skipping DHCP redundant entry for $nodename-$devname (already seen in $redundant_node)\n");
                     next;
                 }
                 if (exists($seen{"HWADDR"}) and exists($seen{"HWADDR"}{"$hwaddr"})) {
                     my $redundant_node = $seen{"HWADDR"}{"$hwaddr"};
+                    $dhcpd_contents .= "   # Skipping $nodename-$devname: duplicate HWADDR ($hwaddr)\n";
                     &iprint("Skipping DHCP config for $nodename-$devname (HWADDR already seen in $redundant_node)\n");
                     next;
                 }
                 if (exists($seen{"IPADDR"}) and exists($seen{"IPADDR"}{"$node_ipaddr"})) {
                     my $redundant_node = $seen{"IPADDR"}{"$node_ipaddr"};
+                    $dhcpd_contents .= "   # Skipping $nodename-$devname: duplicate IPADDR ($node_ipaddr)\n";
                     &iprint("Skipping DHCP config for $nodename-$devname (IPADDR $node_ipaddr already seen in $redundant_node)\n");
                     next;
                 }
@@ -302,7 +309,7 @@
                 if ($nodename and $node_ipaddr and $hwaddr) {
                     &dprint("Adding a host entry for: $nodename-$devname\n");
 
-                    $dhcpd_contents .= "   # Node entry for Warewulf data store ID: $db_id\n";
+                    $dhcpd_contents .= "   # Adding host entry for $nodename-$devname\n";
                     $dhcpd_contents .= "   host $nodename-$devname {\n";
                     $dhcpd_contents .= "      option host-name $hostname;\n";
                     if ($pxelinux_file) {
@@ -328,7 +335,8 @@
                     $seen{"IPADDR"}{"$node_ipaddr"} = "$nodename-$devname";
 
                 } else {
-                    &dprint("Skipping node '$nodename-$devname' due to insufficient information\n");
+                    $dhcpd_contents .= "   # Skipping $nodename-$devname: insufficient configuration\n";
+                    &dprint("Skipping node $nodename-$devname: insufficient information\n");
                 }
             }
         }
@@ -357,13 +365,23 @@
         if (! $digest1 or $digest1 ne $digest2) {
             &dprint("Restarting DHCPD service\n");
             if (! $system->service("dhcpd", "restart")) {
-                &eprint($system->output() ."\n");
+                my $output = $system->output();
+                if ( $output ) {
+                    &eprint("$output\n");
+                } else {
+                    &eprint("There was an error restarting the DHCPD server\n");
+                }
             }
         } else {
             &dprint("Not restarting DHCPD service\n");
         }
         if (!$system->chkconfig("dhcpd", "on")) {
-            &eprint($system->output() ."\n");
+            my $output = $system->output();
+            if ( $output ) {
+                &eprint("$output\n");
+            } else {
+                &eprint("There was an error with chkconfig and the DHCPD server\n");
+            }
         }
     } else {
         &iprint("Not updating DHCP configuration: files are current\n");
