--- software/EasyBuild/2.2.0/lib/python2.7/site-packages/easybuild_framework-2.2.0-py2.7.egg/easybuild/tools/systemtools.py	2015-07-28 14:01:06.000000000 -0700
+++ software.patch/EasyBuild/2.2.0/lib/python2.7/site-packages/easybuild_framework-2.2.0-py2.7.egg/easybuild/tools/systemtools.py	2015-08-12 08:07:22.000000000 -0700
@@ -335,6 +335,9 @@
                     ('2.6.32', '_SP1'),
                     ('3.0', '_SP2'),
                 ],
+                "12": [
+                    ('3.12.28', ''),
+                ],
             }
 
             # append suitable suffix to system version
