--- numpy-1.11.1/numpy/distutils/cpuinfo.py	2016-06-18 09:15:37.000000000 -0700
+++ numpy-1.11.1.patch/numpy/distutils/cpuinfo.py	2016-10-18 10:39:57.000000000 -0700
@@ -97,7 +97,8 @@
         return nbits
 
     def _is_32bit(self):
-        return self.__get_nbits() == '32'
+        #return self.__get_nbits() == '32'
+        return self.__get_nbits() == '64'
 
     def _is_64bit(self):
         return self.__get_nbits() == '64'
