--- numpy-1.10.2/numpy/distutils/command/install.py	2015-08-11 10:03:43.000000000 -0700
+++ numpy-1.10.2.patch/numpy/distutils/command/install.py	2016-02-17 14:40:48.314582100 -0800
@@ -69,7 +69,7 @@
             need_rewrite = False
             for l in f:
                 l = l.rstrip()
-                if ' ' in l:
+                if ' ' in l and '%dir ' not in l:
                     need_rewrite = True
                     l = '"%s"' % (l)
                 lines.append(l)
--- numpy-1.10.2/numpy/distutils/command/autodist.py	2015-10-12 07:29:01.000000000 -0700
+++ numpy-1.10.2.patch/numpy/distutils/command/autodist.py	2016-02-17 14:40:48.320083200 -0800
@@ -49,7 +49,7 @@
     """Return True if the C compiler is GCC 4.x."""
     cmd._check_compiler()
     body = """
-int
+void
 main()
 {
 #if (! defined __GNUC__) || (__GNUC__ < 4)
