--- numpy-1.9.2/numpy/distutils/intelccompiler.py	2014-10-26 14:36:14.000000000 -0700
+++ numpy-1.9.2.patch/numpy/distutils/intelccompiler.py	2015-06-22 08:12:45.000000000 -0700
@@ -36,7 +36,7 @@
     cc_args = "-fPIC"
     def __init__ (self, verbose=0, dry_run=0, force=0):
         UnixCCompiler.__init__ (self, verbose, dry_run, force)
-        self.cc_exe = 'icc -m64 -fPIC'
+        self.cc_exe = 'icc -O3 -g -fPIC -fp-model strict -fomit-frame-pointer -openmp -xhost'
         compiler = self.cc_exe
         self.set_executables(compiler=compiler,
                              compiler_so=compiler,
