--- a/numpy/distutils/fcompiler/intel.py	2017-09-24 17:47:22.000000000 -0700
+++ b/numpy/distutils/fcompiler/intel.py	2017-12-08 14:36:37.000000000 -0800
@@ -2,6 +2,7 @@
 from __future__ import division, absolute_import, print_function
 
 import sys
+from os import environ
 
 from numpy.distutils.ccompiler import simple_version_match
 from numpy.distutils.fcompiler import FCompiler, dummy_fortran_file
@@ -61,7 +62,7 @@
         return ['-fp-model strict -O1 -{}'.format(mpopt)]
 
     def get_flags_arch(self):
-        return []
+        return environ.get('ARCH_FLAGS', '-xSSE4.2 -axCORE-AVX2,CORE-AVX512').strip().split()
 
     def get_flags_linker_so(self):
         opt = FCompiler.get_flags_linker_so(self)
@@ -170,10 +171,10 @@
         return []
 
     def get_flags_debug(self):
-        return ['/4Yb', '/d2']
+        return ["/4Yb", "/d2"]
 
     def get_flags_opt(self):
-        return ['/O1']  # Scipy test failures with /O2
+        return ["/fp:strict /Qopenmp /O1"]  # Scipy test failures with /O2
 
     def get_flags_arch(self):
         return ["/arch:IA32", "/QaxSSE3"]
@@ -209,7 +210,7 @@
     version_match = simple_version_match(start=r'Intel\(R\).*?64,')
 
     def get_flags_arch(self):
-        return ['']
+        return environ.get('ARCH_FLAGS', '/QxSSE4.2 /QaxCORE-AVX2,CORE-AVX512').strip().split()
 
 
 if __name__ == '__main__':
