--- numpy-1.9.1/numpy/distutils/command/install.py	2014-10-26 07:36:14.000000000 -0700
+++ numpy-1.9.1.patch/numpy/distutils/command/install.py	2014-11-17 14:59:39.327433700 -0800
@@ -69,7 +69,7 @@
             need_rewrite = False
             for l in f:
                 l = l.rstrip()
-                if ' ' in l:
+                if ' ' in l and '%dir ' not in l:
                     need_rewrite = True
                     l = '"%s"' % (l)
                 lines.append(l)
--- numpy-1.9.1/numpy/distutils/command/autodist.py	2014-10-26 07:36:14.000000000 -0700
+++ numpy-1.9.1.patch/numpy/distutils/command/autodist.py	2014-11-17 14:59:39.337935800 -0800
@@ -32,7 +32,7 @@
     """Return True if the C compiler is GCC 4.x."""
     cmd._check_compiler()
     body = """
-int
+void
 main()
 {
 #if (! defined __GNUC__) || (__GNUC__ < 4)
