--- a/source/adios2/toolkit/transport/file/FileFStream.cpp
+++ b/source/adios2/toolkit/transport/file/FileFStream.cpp
@@ -15,13 +15,7 @@
 /// \endcond

-#if __cplusplus >= 201703L
+#if __cplusplus >= 201703L && __has_include(<filesystem>)
 #define ADIOS2_USE_STD_FILESYSTEM
-
-#if defined(__GNUC__) && (__GNUC__ < 8) && !defined(__llvm__) && !defined(__INTEL_COMPILER)
-// GCC version 7 does not support std::filesystem from C++17 even though it
-// reports to support C++17.
-#undef ADIOS2_USE_STD_FILESYSTEM
-#endif
 #endif

 #ifdef ADIOS2_USE_STD_FILESYSTEM
