--- a/source/adios2/toolkit/transport/file/FileFStream.cpp.orig
+++ b/source/adios2/toolkit/transport/file/FileFStream.cpp
@@ -15,8 +15,8 @@
 #if __cplusplus >= 201703L
 #define ADIOS2_USE_STD_FILESYSTEM

-#if defined(__GNUC__) && (__GNUC__ < 8) && !defined(__llvm__) && !defined(__INTEL_COMPILER)
-// GCC version 7 does not support std::filesystem from C++17 even though it
+#if defined(__GNUC__) && (__GNUC__ < 8) && !defined(__llvm__)
+// GCC version 7 (and compilers using its libstdc++) does not support std::filesystem from C++17 even though it
 // reports to support C++17.
 #undef ADIOS2_USE_STD_FILESYSTEM
 #endif
