--- petsc-3.5.2/config/BuildSystem/config/setCompilers.py	2014-09-08 20:52:47.000000000 -0700
+++ petsc-3.5.2.patch/config/BuildSystem/config/setCompilers.py	2014-11-21 09:40:51.360866600 -0800
@@ -1330,7 +1330,7 @@
       self.pushLanguage(language)
       # test '-R' before '-rpath' as sun compilers [c,fortran] don't give proper errors with wrong options.
       if not Configure.isDarwin():
-        testFlags = ['-Wl,-rpath,', '-R','-rpath ' , '-Wl,-R,']
+        testFlags = []
       else:
         testFlags = []
       # test '-R' before '-Wl,-rpath' for SUN compilers [as cc on linux accepts -Wl,-rpath, but  f90 & CC do not.
