--- a/test/chtest/Build	2018-07-12 09:44:05.000000000 -0700
+++ b/test/chtest/Build	2018-07-20 10:20:14.000000000 -0700
@@ -16,10 +16,11 @@
 
 set -ex
 
-SRCDIR=$1
-TARBALL=$2
+ARCH=$1
+SRCDIR=$2
+TARBALL=$3
 TARBALL_UNCOMPRESSED=${TARBALL%.*}
-WORKDIR=$3
+WORKDIR=$4
 MIRROR=http://dl-cdn.alpinelinux.org/alpine/v3.6
 APK_TOOLS=apk-tools-static-2.7.6-r0.apk
 IMG="$WORKDIR/img"
@@ -37,7 +38,7 @@
 ## Bootstrap base Alpine Linux.
 
 # Download statically linked apk.
-wget $MIRROR/main/x86_64/$APK_TOOLS
+wget $MIRROR/main/$ARCH/$APK_TOOLS
 
 # Bootstrap directories.
 mkdir img
