--- a/test/chtest/Build	2018-09-04 12:15:54.000000000 -0700
+++ b/test/chtest/Build	2018-10-24 20:48:27.000000000 -0700
@@ -16,12 +16,13 @@
 
 set -ex
 
-SRCDIR=$1
-TARBALL=$2
+ARCH=$1
+SRCDIR=$2
+TARBALL=$3
 TARBALL_UNCOMPRESSED=${TARBALL%.*}
-WORKDIR=$3
+WORKDIR=$4
 MIRROR=http://dl-cdn.alpinelinux.org/alpine/v3.6
-APK_TOOLS=apk-tools-static-2.7.5-r0.apk
+APK_TOOLS=apk-tools-static-2.7.6-r0.apk
 IMG="$WORKDIR/img"
 
 cd $WORKDIR
@@ -37,7 +38,7 @@
 ## Bootstrap base Alpine Linux.
 
 # Download statically linked apk.
-wget $MIRROR/main/x86_64/$APK_TOOLS
+wget $MIRROR/main/$ARCH/$APK_TOOLS
 
 # Bootstrap directories.
 mkdir img
