diff --git a/overlays/host/rootfs/etc/dnsmasq.d/ww4-hosts.conf.ww b/overlays/host/rootfs/etc/dnsmasq.d/ww4-hosts.conf.ww
index 93295daf..5b6715ae 100644
--- a/overlays/host/rootfs/etc/dnsmasq.d/ww4-hosts.conf.ww
+++ b/overlays/host/rootfs/etc/dnsmasq.d/ww4-hosts.conf.ww
@@ -19,7 +19,7 @@ dhcp-boot=tag:x86PC,"warewulf/shim.efi"
 dhcp-boot=tag:x86PC,"/warewulf/{{ index $.Tftp.IpxeBinaries "00:07" }}"
 {{- end }}
 {{- with (index $.Tftp.IpxeBinaries "00:0B" ) }}
-dhcp-boot=tag:aarch64,"/warewulf/{{ index $.Tftp.IpxeBinaries "00:0B" }}"
+dhcp-boot=tag:aarch64,"/warewulf/{{ index $.Tftp.IpxeBinaries "00:0B" | basename }}"
 {{- end }}
 {{- end }}
 # iPXE binary will get the following configuration file
