From cc6b75a4f6aa2d758df9b68255fbe8ff6579f633 Mon Sep 17 00:00:00 2001
From: Christian Goll <cgoll@suse.de>
Date: Wed, 22 Jun 2022 12:17:04 +0200
Subject: [PATCH] bash completion for ssh command

---
 internal/app/wwctl/ssh/root.go | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/internal/app/wwctl/ssh/root.go b/internal/app/wwctl/ssh/root.go
index 81233e23..be6be9c9 100644
--- a/internal/app/wwctl/ssh/root.go
+++ b/internal/app/wwctl/ssh/root.go
@@ -1,6 +1,7 @@
 package ssh
 
 import (
+	"github.com/hpcng/warewulf/internal/pkg/node"
 	"github.com/spf13/cobra"
 )
 
@@ -12,6 +13,19 @@ var (
 		Long:                  "Easily ssh into nodes in parallel to run non-interactive commands\n",
 		RunE:                  CobraRunE,
 		Args:                  cobra.MinimumNArgs(2),
+		ValidArgsFunction: func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+			if len(args) != 0 {
+				return nil, cobra.ShellCompDirectiveNoFileComp
+			}
+
+			nodeDB, _ := node.New()
+			nodes, _ := nodeDB.FindAllNodes()
+			var node_names []string
+			for _, node := range nodes {
+				node_names = append(node_names, node.Id.Get())
+			}
+			return node_names, cobra.ShellCompDirectiveNoFileComp
+		},
 	}
 	DryRun  bool
 	FanOut  int
