--- config.orig 2014-05-28 08:47:22.780045984 -0700
+++ config      2014-05-28 08:54:37.600041219 -0700
@@ -1,13 +1,13 @@
 # Installation directories

 # Default installation prefix
-PREFIX=/usr/local
+#PREFIX=/usr/local

 # System's libraries directory (where binary libraries are installed)
-LUA_LIBDIR= $(PREFIX)/lib/lua/5.1
+LUA_LIBDIR= @@LIBDIR@@/lua/5.2

 # Lua includes directory
-LUA_INC= $(PREFIX)/include
+LUA_INC= @@INCLUDEDIR@@

 # OS dependent
 LIB_OPTION= -shared #for Linux
@@ -16,9 +16,10 @@
 LIBNAME= $T.so.$V

 # Compilation directives
-WARN= -O2 -Wall -fPIC -W -Waggregate-return -Wcast-align -Wmissing-prototypes -Wnested-externs -Wshadow -Wwrite-strings -pedantic
+OPTFLAGS = -O2
+WARN= -Wall -fPIC -W -Waggregate-return -Wcast-align -Wmissing-prototypes -Wnested-externs -Wshadow -Wwrite-strings -pedantic
 INCS= -I$(LUA_INC)
-CFLAGS= $(WARN) $(INCS)
+CFLAGS= $(WARN) $(INCS) $(OPTFLAGS)
 CC= gcc

 # $Id: config,v 1.21 2007/10/27 22:42:32 carregal Exp $
