--- /opt/fsp/pub/compiler/intel/compilers_and_libraries_2016.0.056/linux/ipp/bin/ippvars.sh	2015-06-02 14:38:57.128319818 -0400
+++ /opt/fsp/pub/compiler/intel/compilers_and_libraries_2016.0.056/linux/ipp/bin/ippvars.sh	2015-06-02 15:27:04.368317445 -0400
@@ -7,7 +7,7 @@
 # whole or in part without explicit written authorization from the Company.
 #
 
-export IPPROOT=<INSTALLDIR>/ipp
+export IPPROOT=/opt/fsp/pub/compiler/intel/compilers_and_libraries_2016.0.056/linux/ipp
 export CPATH=$IPPROOT/include:$CPATH
 
 if [[ "$1" != "ia32" && "$1" != "intel64" && "$1" != "ia32_intel64" ]]; then
