--- src/etc/munge.service.in	2013-08-27 11:35:31.000000000 -0700
+++ src/etc/munge.service.in	2014-11-10 11:37:55.104178894 -0800
@@ -12,6 +12,11 @@
 User=munge
 Group=munge
 Restart=on-abort
+PermissionsStartOnly=true
+ExecStartPre=-/usr/bin/mkdir -m 0755 -p /var/log/munge
+ExecStartPre=-/usr/bin/chown -R munge:munge /var/log/munge
+ExecStartPre=-/usr/bin/mkdir -m 0755 -p /var/run/munge
+ExecStartPre=-/usr/bin/chown -R munge:munge /var/run/munge
 
 [Install]
 WantedBy=multi-user.target
