--- a/configure	2018-05-02 07:46:24.000000000 -0700
+++ b/configure	2018-07-18 10:13:43.000000000 -0700
@@ -433,7 +433,7 @@
                   MD=Makefile.defs.linux-gnu
                   GFORTRAN=`which gfortran 2> /dev/null`
                   case ${COMPVER} in
-                  0.*|1.*|2.*|3.*|4.0.*|4.1.*) ;;
+                  0.*|1.*|2.*|3.*|4.0.*|4.1.*|5.*|6.*|7.*|8.*) ;;
                   *)
                       MD=Makefile.defs.linux-gomp
                       GFORTRAN="yes"
