diff --git i/TESTING/ddrive.c w/TESTING/ddrive.c
index 9bb18db..30037f4 100644
--- i/TESTING/ddrive.c
+++ w/TESTING/ddrive.c
@@ -29,6 +29,7 @@ at the top-level directory.
  */
 #include <getopt.h>
 #include <string.h>
+#include <unistd.h>
 #include "slu_ddefs.h"
 #include "MATGEN/matgen.h"
 
diff --git i/TESTING/sdrive.c w/TESTING/sdrive.c
index e2623d8..90bc782 100644
--- i/TESTING/sdrive.c
+++ w/TESTING/sdrive.c
@@ -29,6 +29,7 @@ at the top-level directory.
  */
 #include <getopt.h>
 #include <string.h>
+#include <unistd.h>
 #include "slu_sdefs.h"
 #include "MATGEN/matgen.h"

