diff -ruN SuperLU_5.2.1.old/make.inc SuperLU_5.2.1/make.inc
--- SuperLU_5.2.1.old/make.inc	2016-05-22 17:08:28.000000000 +0100
+++ SuperLU_5.2.1/make.inc	2016-09-29 21:04:09.000000000 +0100
@@ -18,14 +18,14 @@
 #
 #  The name of the libraries to be created/linked to
 #
-SuperLUroot	= /home/xiaoye/Dropbox/Codes/SuperLU/SuperLU_5.2.1/build
+SuperLUroot	= ../
 SUPERLULIB   	= $(SuperLUroot)/SRC/libsuperlu.a
 
 TMGLIB       	= libtmglib.a
 
-# BLASDEF 	= -DUSE_VENDOR_BLAS
-BLASLIB		= /home/xiaoye/Dropbox/Codes/SuperLU/SuperLU_5.2.1/build/CBLAS/libblas.a
-LIBS		= $(SUPERLULIB) /home/xiaoye/Dropbox/Codes/SuperLU/SuperLU_5.2.1/build/CBLAS/libblas.a
+BLASDEF 	= -DUSE_VENDOR_BLAS
+BLASLIB		= -lopenblas
+LIBS		= $(SUPERLULIB) $(BLASLIB)
 
 #
 #  The archiver and the flag(s) to use when building archive (library)
@@ -35,10 +35,11 @@
 ARCHFLAGS    = cr
 RANLIB       = /usr/bin/ranlib
 
-CC           = /usr/bin/cc
-CFLAGS 	     = -O3 -DNDEBUG -DPRNTlevel=0 -DAdd_  
-NOOPTS       = -O0
-FORTRAN	     = /usr/bin/gfortran
+#CC           = gcc
+CFLAGS       = $(RPM_OPT_FLAGS) -fPIC -DPIC
+NOOPTS       = $(RPM_OPT_FLAGS) -Os -fPIC -DPIC
+#FORTRAN	     = gfortran #g77
+FFLAGS       = $(RPM_OPT_FLAGS) -fPIC
 
 LOADER       = $(CC)
 LOADOPTS     =
@@ -46,4 +47,4 @@
 #
 # The directory in which Matlab is installed
 #
-MATLAB	     = /Applications/MATLAB_R2015b.app
+#MATLAB	     = /Applications/MATLAB_R2015b.app
