--- superlu-4.3/make.inc	2011-12-12 14:34:18.000000000 -0800
+++ superlu-4.3.patch/make.inc	2015-04-27 14:11:07.000000000 -0700
@@ -21,14 +21,15 @@
 #
 #  The name of the libraries to be created/linked to
 #
-SuperLUroot	= $(HOME)/Codes/SuperLU/SuperLU_4.3
+SuperLUroot	= ../
 SUPERLULIB   	= $(SuperLUroot)/lib/libsuperlu_4.3.a
+SUPERLUSLIB     = -L$(SuperLUroot)/lib -lsuperlu
 TMGLIB       	= libtmglib.a
 
 ## BLASLIB   	= $(SuperLUroot)/lib/libblas.a
 
 BLASDEF 	= -DUSE_VENDOR_BLAS
-BLASLIB 	= -L$(HOME)/lib/GotoBLAS -lgoto
+BLASLIB 	= -L$(MKLROOT)/intel64/lib -lmkl_rt
 
 ## ATLAS BLAS causes single-precision to fail
 #BLASLIB   	= -L/usr/lib/atlas -lblas
@@ -45,11 +46,11 @@
 ARCHFLAGS    = cr
 RANLIB       = ranlib
 
-CC           = gcc
-CFLAGS       = -DPRNTlevel=0 -O3
-NOOPTS       = 
-FORTRAN	     = g77
-FFLAGS       = -O2
+#CC           = gcc
+CFLAGS       = $(RPM_OPT_FLAGS) -fPIC
+NOOPTS       = $(RPM_OPT_FLAGS) -Os -fPIC
+#FORTRAN	     = g77
+FFLAGS       = $(CFLAGS)
 LOADER       = $(CC)
 LOADOPTS     =
 
@@ -61,5 +62,5 @@
 #
 # The directory in which Matlab is installed
 #
-MATLAB	     = /usr/sww/matlab
+#MATLAB	     = /usr/sww/matlab
 
