--- src/make_ict.orig	2015-03-11 16:55:20.000000000 -0400
+++ src/make_ict	2015-03-11 16:55:45.000000000 -0400
@@ -1,6 +1,6 @@
 LIB_PATH    =
 LIBS        =
-CC          = mpiicc 
+CC          = mpicc 
 ifeq (,$(shell which ${CC}))
 $(error ${CC} is not defined through the PATH environment variable setting. Please try sourcing an Intel(R) Cluster Tools script file such as "mpivars.[c]sh" or "ictvars.[c]sh")
 endif
