--- warewulf-vnfs-3.6/bin/wwvnfs	2014-05-21 11:13:14.000000000 -0700
+++ warewulf-vnfs-3.6.patch/bin/wwvnfs	2014-10-22 10:34:46.454781400 -0700
@@ -24,6 +24,7 @@
     require "sys/types.ph";
     require "syscall.ph";
 }
+use Encode;
 
 my $sysconfdir = Warewulf::ACVars::get('SYSCONFDIR');
 my $config = Warewulf::Config->new("vnfs.conf");
@@ -303,7 +304,7 @@
 sub
 vnfs_create_file($)
 {
-    my $path = $File::Find::name;
+    my $path = decode_utf8($File::Find::name);
 
     if ($path) {
         if ($path =~ /^\Q$opt_chroot\E([[:print:]]*)$/) {
@@ -328,7 +329,7 @@
         my ($dev, $ino, $mode, $nlink, $uid, $gid, $rdev, $size, $atime, $mtime) = lstat("$opt_chroot/$path");
 
         if (S_ISLNK($mode)) {
-            my $destination = readlink("$opt_chroot/$path");
+            my $destination = decode_utf8(readlink("$opt_chroot/$path"));
             &dprint("Creating symbolic link: $tmpdir/$path -> $destination\n");
             if ($destination =~ /^([[:print:]]+)$/) {
                 $destination = $1;
