diff -aruN SuperLU_4.3.orig/SRC/Makefile SuperLU_4.3/SRC/Makefile
--- SuperLU_4.3.orig/SRC/Makefile	2011-10-27 21:49:47.000000000 +0200
+++ SuperLU_4.3/SRC/Makefile	2013-01-08 21:59:19.495919213 +0100
@@ -51,7 +51,7 @@
 	  sp_coletree.o sp_preorder.o sp_ienv.o relax_snode.o \
 	  heap_relax_snode.o colamd.o \
 	  ilu_relax_snode.o ilu_heap_relax_snode.o mark_relax.o \
-	  mc64ad.o qselect.o
+	  qselect.o
 
 SLUSRC = \
 	sgssv.o sgssvx.o \
@@ -63,7 +63,7 @@
 	sreadhb.o sreadrb.o sreadtriple.o \
 	scolumn_dfs.o scolumn_bmod.o spivotL.o spruneL.o \
 	smemory.o sutil.o smyblas2.o \
-	sgsisx.o sgsitrf.o sldperm.o \
+	sgsitrf.o \
 	ilu_sdrop_row.o ilu_ssnode_dfs.o \
 	ilu_scolumn_dfs.o ilu_spanel_dfs.o ilu_scopy_to_ucol.o \
 	ilu_spivotL.o sdiagonal.o
@@ -77,7 +77,7 @@
 	dreadhb.o dreadrb.o dreadtriple.o \
 	dcolumn_dfs.o dcolumn_bmod.o dpivotL.o dpruneL.o \
 	dmemory.o dutil.o dmyblas2.o \
-	dgsisx.o dgsitrf.o dldperm.o \
+	dgsitrf.o \
 	ilu_ddrop_row.o ilu_dsnode_dfs.o \
 	ilu_dcolumn_dfs.o ilu_dpanel_dfs.o ilu_dcopy_to_ucol.o \
 	ilu_dpivotL.o ddiagonal.o 
@@ -92,7 +92,7 @@
 	creadhb.o creadrb.o creadtriple.o \
 	ccolumn_dfs.o ccolumn_bmod.o cpivotL.o cpruneL.o \
 	cmemory.o cutil.o cmyblas2.o \
-	cgsisx.o cgsitrf.o cldperm.o \
+	cgsitrf.o \
 	ilu_cdrop_row.o ilu_csnode_dfs.o \
 	ilu_ccolumn_dfs.o ilu_cpanel_dfs.o ilu_ccopy_to_ucol.o \
 	ilu_cpivotL.o cdiagonal.o
@@ -106,7 +106,7 @@
 	zreadhb.o zreadrb.o zreadtriple.o \
 	zcolumn_dfs.o zcolumn_bmod.o zpivotL.o zpruneL.o \
 	zmemory.o zutil.o zmyblas2.o \
-	zgsisx.o zgsitrf.o zldperm.o \
+	zgsitrf.o \
 	ilu_zdrop_row.o ilu_zsnode_dfs.o \
 	ilu_zcolumn_dfs.o ilu_zpanel_dfs.o ilu_zcopy_to_ucol.o \
 	ilu_zpivotL.o zdiagonal.o
