diff -aruN SuperLU_4.3.orig/make.inc SuperLU_4.3/make.inc
--- SuperLU_4.3.orig/make.inc	2011-10-27 21:49:47.000000000 +0200
+++ SuperLU_4.3/make.inc	2013-01-08 21:28:11.755985160 +0100
@@ -21,14 +21,15 @@
 #
 #  The name of the libraries to be created/linked to
 #
-SuperLUroot	= $(HOME)/Codes/SuperLU/SuperLU_4.2
-SUPERLULIB   	= $(SuperLUroot)/lib/libsuperlu_4.2.a
+SuperLUroot	= ../
+SUPERLULIB   	= $(SuperLUroot)/lib/libsuperlu_4.3.a
+SUPERLUSLIB   	= -L../lib -lsuperlu
 TMGLIB       	= libtmglib.a
 
 ## BLASLIB   	= $(SuperLUroot)/lib/libblas.a
 
 BLASDEF 	= -DUSE_VENDOR_BLAS
-BLASLIB 	= -L$(HOME)/lib/GotoBLAS -lgoto
+BLASLIB 	= -lblas
 
 ## ATLAS BLAS causes single-precision to fail
 #BLASLIB   	= -L/usr/lib/atlas -lblas
@@ -46,12 +47,19 @@
 RANLIB       = ranlib
 
 CC           = gcc
-CFLAGS       = -DPRNTlevel=0 -O3
-NOOPTS       = 
-FORTRAN	     = g77
-FFLAGS       = -O2
-LOADER       = $(CC)
-LOADOPTS     =
+CFLAGS       = $(RPM_OPT_FLAGS) -fPIC -fschedule-insns2 \
+               -funroll-loops -fstrict-aliasing -frerun-loop-opt \
+	       -ftree-vectorize -funit-at-a-time
+ifeq ($(MACH),i386)
+ CFLAGS     += -mtune=pentium-m -momit-leaf-frame-pointer # -ffast-math -msse2 -mfpmath=sse
+else
+ CFLAGS     += -O3 -finline-limit=720 --param max-inline-insns-auto=160  
+endif
+NOOPTS       = $(RPM_OPT_FLAGS) -Os -fPIC
+FORTRAN	     = gfortran
+FFLAGS       = $(CFLAGS)
+LOADER       = gcc
+LOADOPTS     = 
 
 #
 #  C preprocessor defs for compilation for the Fortran interface
@@ -61,5 +69,5 @@
 #
 # The directory in which Matlab is installed
 #
-MATLAB	     = /usr/sww/matlab
+#MATLAB	     = /usr/sww/matlab
 
diff -aruN SuperLU_4.3.orig/TESTING/Makefile SuperLU_4.3/TESTING/Makefile
--- SuperLU_4.3.orig/TESTING/Makefile	2011-10-27 21:49:47.000000000 +0200
+++ SuperLU_4.3/TESTING/Makefile	2013-01-08 21:12:51.444017656 +0100
@@ -54,7 +54,7 @@
 
 ./stest: $(SLINTST) $(ALINTST) $(SUPERLULIB) $(TMGLIB)
 	$(LOADER) $(LOADOPTS) $(SLINTST) $(ALINTST) \
-        $(TMGLIB) $(SUPERLULIB) $(BLASLIB) -lm -o $@
+        $(TMGLIB) $(SUPERLUSLIB) $(BLASLIB) -lm -o $@
 
 stest.out: stest stest.csh
 	@echo Testing SINGLE PRECISION linear equation routines 
@@ -64,7 +64,7 @@
 
 ./dtest: $(DLINTST) $(ALINTST) $(SUPERLULIB) $(TMGLIB)
 	$(LOADER) $(LOADOPTS) $(DLINTST) $(ALINTST) \
-        $(TMGLIB) $(SUPERLULIB) $(BLASLIB) -lm -o $@
+        $(TMGLIB) $(SUPERLUSLIB) $(BLASLIB) -lm -o $@
 
 dtest.out: dtest dtest.csh
 	@echo Testing DOUBLE PRECISION linear equation routines 
@@ -74,7 +74,7 @@
 
 ./ctest: $(CLINTST) $(ALINTST) $(SUPERLULIB) $(TMGLIB)
 	$(LOADER) $(LOADOPTS) $(CLINTST) $(ALINTST) \
-	$(TMGLIB) $(SUPERLULIB) $(BLASLIB) -lm -o $@
+	$(TMGLIB) $(SUPERLUSLIB) $(BLASLIB) -lm -o $@
 
 ctest.out: ctest ctest.csh
 	@echo Testing SINGLE COMPLEX linear equation routines 
@@ -84,7 +84,7 @@
 
 ./ztest: $(ZLINTST) $(ALINTST) $(SUPERLULIB) $(TMGLIB)
 	$(LOADER) $(LOADOPTS) $(ZLINTST) $(ALINTST) \
-	$(TMGLIB) $(SUPERLULIB) $(BLASLIB) -lm -o $@
+	$(TMGLIB) $(SUPERLUSLIB) $(BLASLIB) -lm -o $@
 
 ztest.out: ztest ztest.csh
 	@echo Testing DOUBLE COMPLEX linear equation routines 
