--- a/compile-macros.sh	2017-12-07 11:46:04.000000000 -0800
+++ b/compile-macros.sh	2017-12-08 09:18:49.000000000 -0800
@@ -1,10 +1,10 @@
 #!/bin/bash
 
-FLAVORS="python2 python3 pypy3"
+FLAVORS="python2 python34 pypy3"
 
 # order of BUILDSET is important, it is copied to order of %pythons,
 # and that determines the last installed binary
-BUILDSET="python2 python3"
+BUILDSET="python2 python34"
 
 
 ### flavor-specific: generate from flavor.in
--- a/functions.lua	2017-12-07 11:46:04.000000000 -0800
+++ b/functions.lua	2017-12-08 09:44:39.000000000 -0800
@@ -27,7 +27,7 @@
     python = "py",
     -- ??
     python2 = "py2",
-    python3 = "py3",
+    python34 = "py3",
     pypy = "pypy",
 }
 
--- a/macros/010-common-defs	2017-12-07 11:46:04.000000000 -0800
+++ b/macros/010-common-defs	2017-12-08 09:42:48.000000000 -0800
@@ -1,5 +1,5 @@
 %system_python python2
-%python_for_executables python3
+%python_for_executables python34
 
 ##### common functionality #####
 
