--- a/macros.lua	2017-12-07 11:46:04.000000000 -0800
+++ b/macros.lua	2017-12-08 13:18:52.000000000 -0800
@@ -14,7 +14,7 @@
     -- ...but check that it is actually in the buildset
     if insert_last_python then table.insert(pythons, last_python) end
 
-    modname = rpm.expand("%name")
+    modname = string.gsub(rpm.expand("%name"), "-ohpc", "")
     local spec_name_prefix = "python"
     -- modname from name
     local name = modname
