--- a/functions.lua	2017-12-07 11:46:04.000000000 -0800
+++ b/functions.lua	2017-12-11 09:22:39.000000000 -0800
@@ -59,7 +59,7 @@
     if flavor == "python2" and old_python2 then
         flavor = "python"
     end
-    local name = flavor .. "-" .. modname
+    local name = flavor .. "-" .. modname .. "%PROJ_DELIM"
     if subpkg and subpkg ~= "" then
         name = name .. "-" .. subpkg
     end
