--- slurm-17.02.6/slurm.spec	2017-07-05 13:53:56.000000000 -0500
+++ slurm.spec	2017-08-08 16:32:43.000000000 -0500
@@ -1,3 +1,20 @@
+#----------------------------------------------------------------------------bh-
+# This RPM .spec file is part of the OpenHPC project.
+#
+# It may have been modified from the default version supplied by the underlying
+# release package (if available) in order to apply patches, perform customized
+# build/install configurations, and supply additional files to support
+# desired integration conventions.
+#
+#----------------------------------------------------------------------------eh-
+
+%include %{_sourcedir}/OHPC_macros
+%global _with_mysql  1
+
+%define pname slurm
+
+# $Id$
+#
 # Note that this package is not relocatable
 
 #
@@ -77,19 +94,31 @@
 %slurm_with_opt sgijob
 %endif
 
-Name:    slurm
+%define ver_exp 17-02-06-1
+
+Name:    %{pname}%{PROJ_DELIM}
 Version: 17.02.6
 Release: 1%{?dist}
 
 Summary: Slurm Workload Manager
 
 License: GPL
-Group: System Environment/Base
+Group: %{PROJ_NAME}/rms
 Source: slurm-17.02.6.tar.bz2
-BuildRoot: %{_tmppath}/%{name}-%{version}-%{release}
+Source:    https://github.com/SchedMD/slurm/archive/%{pname}-%{ver_exp}.tar.gz
+Source1:   OHPC_macros
 URL: https://slurm.schedmd.com/
 
-Requires: slurm-plugins
+# 8/15/14 karl.w.schulz@intel.com - include prereq
+%if 0%{?sles_version} || 0%{?suse_version}
+PreReq: %{insserv_prereq} %{fillup_prereq}
+%if 0%{?suse_version} < 1315
+BuildRequires: klogd sysconfig
+%endif
+%endif
+
+Requires: %{pname}-plugins%{PROJ_DELIM}
+#!BuildIgnore: post-build-checks
 
 %ifos linux
 BuildRequires: python
@@ -119,25 +148,31 @@
 %endif
 
 %if %{slurm_with openssl}
+%if 0%{?suse_version}
+BuildRequires: libopenssl-devel openssl
+%else
 BuildRequires: openssl-devel >= 0.9.6 openssl >= 0.9.6
 %endif
+%endif
 
-%define use_mysql_devel %(perl -e '`rpm -q mariadb-devel`; print $?;')
-
-%if %{slurm_with mysql}
-%if %{use_mysql_devel}
-BuildRequires: mysql-devel >= 5.0.0
+#%if %{slurm_with mysql}
+%if 0%{?suse_version}
+%if 0%{?suse_version} >= 1200
+BuildRequires: libmysqlclient-devel
+%else
+BuildRequires: libmysql5
+%endif
 %else
+%if 0%{?rhel_version} >= 700 || 0%{?centos_version} >= 700
 BuildRequires: mariadb-devel >= 5.0.0
+%else
+BuildRequires: mysql-devel >= 5.0.0
 %endif
 %endif
+#%endif
 
 %if %{slurm_with cray_alps}
-%if %{use_mysql_devel}
 BuildRequires: mysql-devel
-%else
-BuildRequires: mariadb-devel
-%endif
 %endif
 
 %if %{slurm_with cray}
@@ -167,6 +202,19 @@
 
 BuildRequires: perl(ExtUtils::MakeMaker)
 
+%if 0%{?suse_version} >= 1230
+Requires(pre):  shadow
+%else
+%if 0%{?suse_version}
+Requires(pre):  pwdutils
+%else
+Requires(pre):  shadow-utils
+%endif
+%endif
+
+#needed to enable jobcomp_elasticsearch plugin
+BuildRequires: curl-devel
+
 %description
 Slurm is an open source, fault-tolerant, and highly
 scalable cluster management and job scheduling system for Linux clusters.
@@ -223,41 +271,47 @@
 %define _perlarchlibdir %{_prefix}%{_perlarchlib}
 %define _php_extdir %(php-config --extension-dir 2>/dev/null || echo %{_libdir}/php5)
 
-%package perlapi
+%package -n %{pname}-perlapi%{PROJ_DELIM}
 Summary: Perl API to Slurm
-Group: Development/System
-Requires: slurm
-%description perlapi
+Group: %{PROJ_NAME}/rms
+Requires: %{pname}%{PROJ_DELIM}
+%description -n %{pname}-perlapi%{PROJ_DELIM}
 Perl API package for Slurm.  This package includes the perl API to provide a
 helpful interface to Slurm through Perl
 
-%package devel
+%package -n %{pname}-devel%{PROJ_DELIM}
 Summary: Development package for Slurm
-Group: Development/System
-Requires: slurm
-%description devel
+Group: %{PROJ_NAME}/rms
+Requires: %{pname}%{PROJ_DELIM}
+%if 0%{?suse_version}
+BuildRequires:  pkg-config
+%else
+BuildRequires:  pkgconfig
+%endif
+
+%description -n %{pname}-devel%{PROJ_DELIM}
 Development package for Slurm.  This package includes the header files
 and static libraries for the Slurm API
 
 %if %{slurm_with auth_none}
-%package auth-none
+%package -n %{pname}-auth-none%{PROJ_DELIM}
 Summary: Slurm auth NULL implementation (no authentication)
-Group: System Environment/Base
-Requires: slurm
-%description auth-none
+Group: %{PROJ_NAME}/rms
+Requires: %{pname}%{PROJ_DELIM}
+%description -n %{pname}-auth-none%{PROJ_DELIM}
 Slurm NULL authentication module
 %endif
 
 # This is named munge instead of auth-munge since there are 2 plugins in the
 # package.  auth-munge and crypto-munge
 %if %{slurm_with munge}
-%package munge
+%package -n %{pname}-munge%{PROJ_DELIM}
 Summary: Slurm authentication and crypto implementation using Munge
-Group: System Environment/Base
-Requires: slurm munge
-BuildRequires: munge-devel munge-libs
+Group: %{PROJ_NAME}/rms
+Requires: %{pname}%{PROJ_DELIM} munge%{PROJ_DELIM}
+BuildRequires: munge-devel%{PROJ_DELIM} munge-libs%{PROJ_DELIM}
 Obsoletes: slurm-auth-munge
-%description munge
+%description -n %{pname}-munge%{PROJ_DELIM}
 Slurm authentication and crypto implementation using Munge. Used to
 authenticate user originating an RPC, digitally sign and/or encrypt messages
 %endif
@@ -265,54 +319,61 @@
 %if %{slurm_with bluegene}
 %package bluegene
 Summary: Slurm interfaces to IBM Blue Gene system
-Group: System Environment/Base
-Requires: slurm
+Group: %{PROJ_NAME}/rms
+Requires: %{pname}%{PROJ_DELIM}
 %description bluegene
 Slurm plugin interfaces to IBM Blue Gene system
 %endif
 
-%package slurmdbd
+%package -n %{pname}-slurmdbd%{PROJ_DELIM}
 Summary: Slurm database daemon
-Group: System Environment/Base
-Requires: slurm-plugins slurm-sql
-%description slurmdbd
+Group: %{PROJ_NAME}/rms
+Requires: %{pname}-plugins%{PROJ_DELIM} %{pname}-sql%{PROJ_DELIM}
+%description -n %{pname}-slurmdbd%{PROJ_DELIM}
 Slurm database daemon. Used to accept and process database RPCs and upload
 database changes to slurmctld daemons on each cluster
 
-%package sql
+%package -n %{pname}-sql%{PROJ_DELIM}
 Summary: Slurm SQL support
-Group: System Environment/Base
-%description sql
+Group: %{PROJ_NAME}/rms
+%description -n %{pname}-sql%{PROJ_DELIM}
 Slurm SQL support. Contains interfaces to MySQL.
 
-%package plugins
+%package -n %{pname}-plugins%{PROJ_DELIM}
+Summary: Slurm plugins (loadable shared objects)
+Group: %{PROJ_NAME}/rms
+Requires: munge-libs%{PROJ_DELIM}
+%description -n %{pname}-plugins%{PROJ_DELIM}
+
+%package -n %{pname}-plugins%{PROJ_DELIM}
 Summary: Slurm plugins (loadable shared objects)
-Group: System Environment/Base
-%description plugins
+Group: %{PROJ_NAME}/rms
+Requires: munge-libs%{PROJ_DELIM}
+%description -n %{pname}-plugins%{PROJ_DELIM}
 Slurm plugins (loadable shared objects) supporting a wide variety of
 architectures and behaviors. These basically provide the building blocks
 with which Slurm can be configured. Note that some system specific plugins
 are in other packages
 
-%package torque
-Summary: Torque/PBS wrappers for transitition from Torque/PBS to Slurm
-Group: Development/System
-Requires: slurm-perlapi
-%description torque
+%package -n %{pname}-torque%{PROJ_DELIM}
+Summary: Torque/PBS wrappers for transition from Torque/PBS to Slurm
+Group: %{PROJ_NAME}/rms
+Requires: %{pname}-perlapi%{PROJ_DELIM}
+%description -n %{pname}-torque%{PROJ_DELIM}
 Torque wrapper scripts used for helping migrate from Torque/PBS to Slurm
 
-%package openlava
+%package -n %{pname}-openlava%{PROJ_DELIM}
 Summary: openlava/LSF wrappers for transitition from OpenLava/LSF to Slurm
-Group: Development/System
-Requires: slurm-perlapi
-%description openlava
+Group: %{PROJ_NAME}/rms
+Requires: %{pname}-slurm-perlapi%{PROJ_DELIM}
+%description -n %{pname}-openlava%{PROJ_DELIM}
 OpenLava wrapper scripts used for helping migrate from OpenLava/LSF to Slurm
 
 %if %{slurm_with percs}
 %package percs
 Summary: Slurm plugins to run on an IBM PERCS system
-Group: System Environment/Base
-Requires: slurm nrt
+Group: %{PROJ_NAME}/rms
+Requires: %{pname}%{PROJ_DELIM} nrt
 BuildRequires: nrt
 %description percs
 Slurm plugins to run on an IBM PERCS system, POE interface and NRT switch plugin
@@ -321,8 +382,8 @@
 %if %{slurm_with sgijob}
 %package proctrack-sgi-job
 Summary: Slurm process tracking plugin for SGI job containers
-Group: System Environment/Base
-Requires: slurm
+Group: %{PROJ_NAME}/rms
+Requires: %{pname}%{PROJ_DELIM}
 BuildRequires: job
 %description proctrack-sgi-job
 Slurm process tracking plugin for SGI job containers
@@ -330,22 +391,22 @@
 %endif
 
 %if %{slurm_with lua}
-%package lua
+%package -n %{pname}-lua%{PROJ_DELIM}
 Summary: Slurm lua bindings
-Group: System Environment/Base
-Requires: slurm lua
+Group: %{PROJ_NAME}/rms
+Requires: %{pname}%{PROJ_DELIM} lua
 BuildRequires: lua-devel
-%description lua
+%description -n %{pname}-lua%{PROJ_DELIM}
 Slurm lua bindings
 Includes the Slurm proctrack/lua and job_submit/lua plugin
 %endif
 
-%package contribs
+%package -n %{pname}-contribs%{PROJ_DELIM}
 Summary: Perl tool to print Slurm job state information
-Group: Development/System
-Requires: slurm
-Obsoletes: slurm-sjobexit slurm-sjstat slurm-seff
-%description contribs
+Group: %{PROJ_NAME}/rms
+Requires: %{pname}-slurm%{PROJ_DELIM}
+Obsoletes: slurm-sjobexit%{PROJ_DELIM} slurm-sjstat%{PROJ_DELIM} slurm-seff%{PROJ_DELIM}
+%description -n %{pname}-contribs%{PROJ_DELIM}
 seff is a mail program used directly by the Slurm daemons. On completion of a
 job, wait for it's accounting information to be available and include that
 information in the email body.
@@ -358,13 +419,13 @@
 utilities will provide more information and greater depth of understanding.
 
 %if %{slurm_with pam}
-%package pam_slurm
+%package -n %{pname}-pam_slurm%{PROJ_DELIM}
 Summary: PAM module for restricting access to compute nodes via Slurm
-Group: System Environment/Base
-Requires: slurm
+Group: %{PROJ_NAME}/rms
+Requires: %{pname}%{PROJ_DELIM} %{pname}-devel%{PROJ_DELIM}
 BuildRequires: pam-devel
 Obsoletes: pam_slurm
-%description pam_slurm
+%description -n %{pname}-pam_slurm%{PROJ_DELIM}
 This module restricts access to compute nodes in a cluster where Slurm is in
 use.  Access is granted to root, any user with an Slurm-launched job currently
 running on the node, or any user who has allocated resources on the node
@@ -374,7 +435,7 @@
 #############################################################################
 
 %prep
-%setup -n slurm-17.02.6
+%setup -n slurm-slurm-%{ver_exp}
 
 %build
 %configure \
@@ -403,19 +464,6 @@
 %__make %{?_smp_mflags}
 
 %install
-
-
-# Strip out some dependencies
-
-cat > find-requires.sh <<'EOF'
-exec %{__find_requires} "$@" | egrep -v '^libpmix.so|libevent'
-EOF
-chmod +x find-requires.sh
-%global _use_internal_dependency_generator 0
-%global __find_requires %{_builddir}/%{buildsubdir}/find-requires.sh
-
-
-rm -rf "$RPM_BUILD_ROOT"
 DESTDIR="$RPM_BUILD_ROOT" %__make install
 DESTDIR="$RPM_BUILD_ROOT" %__make install-contrib
 
@@ -450,6 +498,7 @@
    rm -f $RPM_BUILD_ROOT/%{_sbindir}/slurmconfgen.py
 %endif
 
+install -D -m644 etc/slurm.conf.example ${RPM_BUILD_ROOT}%{_sysconfdir}/slurm.conf.example
 install -D -m644 etc/cgroup.conf.example ${RPM_BUILD_ROOT}%{_sysconfdir}/cgroup.conf.example
 install -D -m644 etc/cgroup_allowed_devices_file.conf.example ${RPM_BUILD_ROOT}%{_sysconfdir}/cgroup_allowed_devices_file.conf.example
 install -D -m644 etc/layouts.d.power.conf.example ${RPM_BUILD_ROOT}%{_sysconfdir}/layouts.d/power.conf.example
@@ -460,6 +509,32 @@
 install -D -m644 etc/slurmdbd.conf.example ${RPM_BUILD_ROOT}%{_sysconfdir}/slurmdbd.conf.example
 install -D -m755 contribs/sjstat ${RPM_BUILD_ROOT}%{_bindir}/sjstat
 
+# 9/8/14 karl.w.schulz@intel.com - provide starting config file
+%if 0%{?OHPC_BUILD}
+head -n -2 $RPM_BUILD_ROOT/%{_sysconfdir}/slurm.conf.example | grep -v ReturnToService > $RPM_BUILD_ROOT/%{_sysconfdir}/slurm.conf
+echo "# OpenHPC default configuration" >> $RPM_BUILD_ROOT/%{_sysconfdir}/slurm.conf
+echo "PropagateResourceLimitsExcept=MEMLOCK" >> $RPM_BUILD_ROOT/%{_sysconfdir}/slurm.conf
+echo "SlurmdLogFile=/var/log/slurm.log" >> $RPM_BUILD_ROOT/%{_sysconfdir}/slurm.conf
+echo "SlurmctldLogFile=/var/log/slurmctld.log" >> $RPM_BUILD_ROOT/%{_sysconfdir}/slurm.conf
+echo "AccountingStorageType=accounting_storage/filetxt" >> $RPM_BUILD_ROOT/%{_sysconfdir}/slurm.conf
+echo "Epilog=/etc/slurm/slurm.epilog.clean" >> $RPM_BUILD_ROOT/%{_sysconfdir}/slurm.conf
+echo "NodeName=c[1-4] Sockets=2 CoresPerSocket=8 ThreadsPerCore=2 State=UNKNOWN" >> $RPM_BUILD_ROOT/%{_sysconfdir}/slurm.conf
+echo "PartitionName=normal Nodes=c[1-4] Default=YES MaxTime=24:00:00 State=UP" >> $RPM_BUILD_ROOT/%{_sysconfdir}/slurm.conf
+# 6/3/16 nirmalasrjn@gmail.com - Adding ReturnToService Directive to starting config file (note removal of variable during above creation)
+echo "ReturnToService=1" >> $RPM_BUILD_ROOT/%{_sysconfdir}/slurm.conf
+# 9/17/14 karl.w.schulz@intel.com - Add option to drop VM cache during epilog
+sed -i '/^# No other SLURM jobs,/i \\n# Drop clean caches (OpenHPC)\necho 3 > /proc/sys/vm/drop_caches\n\n#' $RPM_BUILD_ROOT/%{_sysconfdir}/slurm.epilog.clean
+
+%endif
+
+# Correct some file permissions
+test -f $RPM_BUILD_ROOT/%{_libdir}/libpmi.la	&&
+	chmod 644 $RPM_BUILD_ROOT/%{_libdir}/libpmi.la
+test -f $RPM_BUILD_ROOT/%{_libdir}/libslurm.la	&&
+	chmod 644 $RPM_BUILD_ROOT/%{_libdir}/libslurm.la
+test -f $RPM_BUILD_ROOT/%{_libdir}/libslurmdb.la &&
+	chmod 644 $RPM_BUILD_ROOT/%{_libdir}/libslurmdb.la
+
 # Delete unpackaged files:
 test -s $RPM_BUILD_ROOT/%{_perldir}/auto/Slurm/Slurm.bs         ||
 rm   -f $RPM_BUILD_ROOT/%{_perldir}/auto/Slurm/Slurm.bs
@@ -586,8 +661,8 @@
 Cflags: -I\${includedir}
 Libs: -L\${libdir} -lslurm
 Description: Slurm API
-Name: slurm
-Version: 17.02.6
+Name: %{pname}
+Version: %{version}
 EOF
 
 %if %{slurm_with bluegene}
@@ -671,9 +746,9 @@
 test -f $RPM_BUILD_ROOT/%{_libdir}/slurm/mpi_pmix_v2.so              &&
    echo %{_libdir}/slurm/mpi_pmix_v2.so              >> $LIST
 test -f $RPM_BUILD_ROOT/%{_libdir}/slurm/node_features_knl_cray.so   &&
-   echo %{_libdir}/slurm/node_features_knl_cray.so   >> $LIST
+    echo %{_libdir}/slurm/node_features_knl_cray.so   >> $LIST
 test -f $RPM_BUILD_ROOT/%{_libdir}/slurm/node_features_knl_generic.so &&
-   echo %{_libdir}/slurm/node_features_knl_generic.so   >> $LIST
+    echo %{_libdir}/slurm/node_features_knl_generic.so >> $LIST 
 test -f $RPM_BUILD_ROOT/%{_libdir}/slurm/power_cray.so               &&
    echo %{_libdir}/slurm/power_cray.so               >> $LIST
 test -f $RPM_BUILD_ROOT/%{_libdir}/slurm/select_bluegene.so          &&
@@ -716,10 +791,8 @@
     test -f $RPM_BUILD_ROOT/lib64/security/pam_slurm_adopt.so		&&
 	echo /lib64/security/pam_slurm_adopt.so		>>$LIST
 %endif
-#############################################################################
 
-%clean
-rm -rf $RPM_BUILD_ROOT
+mkdir -p $RPM_BUILD_ROOT/%{_docdir}
 #############################################################################
 
 %files -f slurm.files
@@ -762,6 +835,26 @@
 %config %{_sysconfdir}/slurm.conf.template
 %{_sbindir}/slurmconfgen.py
 %endif
+%config %{_sysconfdir}/slurm.conf.example
+
+%{OHPC_PUB}
+%doc AUTHORS BUILD.NOTES COPYING DISCLAIMER INSTALL LICENSE.OpenSSL NEWS README.rst RELEASE_NOTES
+
+# 9/8/14 karl.w.schulz@intel.com - provide starting config file
+%if 0%{?OHPC_BUILD}
+%config (noreplace) %{_sysconfdir}/slurm.conf
+%endif
+
+# 11/13/14 karl.w.schulz@intel.com - include systemd files 
+
+%if 0%{?suse_version} >= 1200 || 0%{?rhel_version} >= 700 || 0%{?centos_version} >= 700
+
+%config /usr/lib/systemd/system/slurmctld.service
+%config /usr/lib/systemd/system/slurmd.service
+%config /usr/lib/systemd/system/slurmdbd.service
+
+%endif
+
 %config %{_sysconfdir}/cgroup.conf.example
 %config %{_sysconfdir}/cgroup_allowed_devices_file.conf.example
 %config %{_sysconfdir}/layouts.d/power.conf.example
@@ -773,7 +866,7 @@
 %exclude %{_mandir}/man1/sjstat*
 #############################################################################
 
-%files devel
+%files -n %{pname}-devel%{PROJ_DELIM}
 %defattr(-,root,root)
 %dir %attr(0755,root,root)
 %dir %{_prefix}/include/slurm
@@ -785,14 +878,14 @@
 #############################################################################
 
 %if %{slurm_with auth_none}
-%files auth-none
+%files -n %{pname}-auth-none%{PROJ_DELIM}
 %defattr(-,root,root)
 %{_libdir}/slurm/auth_none.so
 %endif
 #############################################################################
 
 %if %{slurm_with munge}
-%files munge
+%files -n %{pname}-munge%{PROJ_DELIM}
 %defattr(-,root,root)
 %{_libdir}/slurm/auth_munge.so
 %{_libdir}/slurm/crypto_munge.so
@@ -810,7 +903,7 @@
 %endif
 #############################################################################
 
-%files perlapi
+%files -n %{pname}-perlapi%PROJ_DELIM
 %defattr(-,root,root)
 %{_perldir}/Slurm.pm
 %{_perldir}/Slurm/Bitstr.pm
@@ -825,7 +918,7 @@
 
 #############################################################################
 
-%files -f slurmdbd.files slurmdbd
+%files -f slurmdbd.files -n %{pname}-slurmdbd%{PROJ_DELIM}
 %defattr(-,root,root)
 %{_sbindir}/slurmdbd
 %{_mandir}/man5/slurmdbd.*
@@ -833,12 +926,12 @@
 %config %{_sysconfdir}/slurmdbd.conf.example
 #############################################################################
 
-%files -f sql.files sql
+%files -f sql.files -n %{pname}-sql%{PROJ_DELIM}
 %defattr(-,root,root)
 %dir %{_libdir}/slurm
 #############################################################################
 
-%files -f plugins.files plugins
+%files -f plugins.files -n %{pname}-plugins%{PROJ_DELIM}
 %defattr(-,root,root)
 %dir %{_libdir}/slurm
 %{_libdir}/slurm/accounting_storage_filetxt.so
@@ -849,7 +942,6 @@
 %{_libdir}/slurm/acct_gather_infiniband_none.so
 %{_libdir}/slurm/acct_gather_energy_none.so
 %{_libdir}/slurm/acct_gather_profile_none.so
-%{_libdir}/slurm/burst_buffer_generic.so
 %{_libdir}/slurm/checkpoint_none.so
 %{_libdir}/slurm/checkpoint_ompi.so
 %{_libdir}/slurm/core_spec_cray.so
@@ -915,7 +1007,7 @@
 %{_libdir}/slurm/topology_tree.so
 #############################################################################
 
-%files torque
+%files -n %{pname}-torque%{PROJ_DELIM}
 %defattr(-,root,root)
 %{_bindir}/pbsnodes
 %{_bindir}/qalter
@@ -931,7 +1023,7 @@
 %{_libdir}/slurm/spank_pbs.so
 #############################################################################
 
-%files openlava
+%files -n %{pname}-openlava%{PROJ_DELIM}
 %defattr(-,root,root)
 %{_bindir}/bjobs
 %{_bindir}/bkill
@@ -954,7 +1046,7 @@
 #############################################################################
 
 %if %{slurm_with lua}
-%files lua
+%files -n %{pname}-lua%{PROJ_DELIM}
 %defattr(-,root,root)
 %doc contribs/lua/proctrack.lua
 %{_libdir}/slurm/job_submit_lua.so
@@ -962,7 +1054,7 @@
 %endif
 #############################################################################
 
-%files contribs
+%files -n %{pname}-contribs%{PROJ_DELIM}
 %defattr(-,root,root)
 %{_bindir}/seff
 %{_bindir}/sjobexitmod
@@ -972,7 +1064,7 @@
 #############################################################################
 
 %if %{slurm_with pam}
-%files -f pam.files pam_slurm
+%files -f pam.files -n %{pname}-pam_slurm%{PROJ_DELIM}
 %defattr(-,root,root)
 %endif
 #############################################################################
@@ -989,15 +1081,34 @@
 #    fi
 #fi
 
+getent passwd slurm >/dev/null || \
+    /usr/sbin/useradd -U -c "SLURM resource manager" \
+    -s /sbin/nologin -r -d %{_sysconfdir} slurm
+exit 0
+
 %post
-if [ -x /sbin/ldconfig ]; then
-    /sbin/ldconfig %{_libdir}
-    if [ $1 = 1 ]; then
+
+# 8/15/14 karl.w.schulz@intel.com - use insserv macro for init.d on suse
+
+%if 0%{?suse_version}
+%{fillup_and_insserv -f}
+%else
+if [ $1 = 1 ]; then
 	if [ -x /etc/init.d/slurm ]; then
 	    [ -x /sbin/chkconfig ] && /sbin/chkconfig --add slurm
-        fi
     fi
 fi
+%endif
+
+# 3/31/17 karl.w.schulz@intel.com - fix perm for txt accounting file possibility
+if [ ! -f /var/log/slurm_jobacct.log ];then
+    touch /var/log/slurm_jobacct.log
+    chown slurm: /var/log/slurm_jobacct.log
+fi
+
+if [ -x /sbin/ldconfig ]; then
+    /sbin/ldconfig %{_libdir}
+fi
 
 %if %{slurm_with bluegene}
 %post bluegene
@@ -1016,8 +1127,8 @@
     fi
 fi
 
-%preun slurmdbd
-if [ "$1" -eq 0 ]; then
+%preun -n %{pname}-slurmdbd%{PROJ_DELIM}
+if [ "$1" = 0 ]; then
     if [ -x /etc/init.d/slurmdbd ]; then
 	[ -x /sbin/chkconfig ] && /sbin/chkconfig --del slurmdbd
 	if /etc/init.d/slurmdbd status | grep -q running; then
@@ -1040,7 +1151,7 @@
 %insserv_cleanup
 %endif
 
-%postun slurmdbd
+%postun -n %{pname}-slurmdbd%{PROJ_DELIM}
 if [ "$1" -gt 1 ]; then
     if [ -x /etc/init.d/slurmdbd ]; then
         /etc/init.d/slurm condrestart
