--- MUMPS_5.1.1/src/cana_aux_par.F	2017-03-20 07:35:05.000000000 -0700
+++ MUMPS_5.1.1.patch/src/cana_aux_par.F	2017-04-05 10:18:54.000000000 -0700
@@ -14,7 +14,7 @@
       USE CMUMPS_STRUC_DEF
       USE MUMPS_MEMORY_MOD
       USE MUMPS_ANA_ORD_WRAPPERS
-      INCLUDE 'mpif.h'
+C      INCLUDE 'mpif.h'
       PUBLIC CMUMPS_ANA_F_PAR
       INTERFACE CMUMPS_ANA_F_PAR
       MODULE PROCEDURE CMUMPS_ANA_F_PAR
@@ -833,6 +833,7 @@
       END SUBROUTINE CMUMPS_GET_SUBTREES
       SUBROUTINE CMUMPS_PARSYMFACT(id, ord, GPE, GNV, WORK)  
       IMPLICIT NONE
+      INCLUDE 'mpif.h'
       TYPE(CMUMPS_STRUC)   :: id
       TYPE(ORD_TYPE)       :: ord
       INTEGER, POINTER     :: GPE(:), GNV(:)
@@ -1835,6 +1836,7 @@
       END SUBROUTINE CMUMPS_BUILD_LOC_GRAPH
       SUBROUTINE CMUMPS_SEND_BUF(APNT, PROC, NPROCS, BUFSIZE, IPE, PE,
      &     LENG, RCVBUF, MSGCNT, SNDCNT, COMM)
+      USE MPI
       IMPLICIT NONE
       INTEGER                 :: NPROCS, PROC, COMM
       TYPE(ARRPNT)            :: APNT(:)
@@ -1981,6 +1983,7 @@
       END SUBROUTINE CMUMPS_BUILD_TREE
       SUBROUTINE CMUMPS_GRAPH_DIST(id, ord, FIRST,
      &     LAST, BASE, NPROCS, WORK, TYPE)
+      USE MPI
       IMPLICIT NONE
       TYPE(CMUMPS_STRUC)   :: id
       TYPE(ORD_TYPE)       :: ord
--- MUMPS_5.1.1/src/dana_aux_par.F	2017-03-20 07:35:03.000000000 -0700
+++ MUMPS_5.1.1.patch/src/dana_aux_par.F	2017-04-05 10:33:51.000000000 -0700
@@ -14,7 +14,7 @@
       USE DMUMPS_STRUC_DEF
       USE MUMPS_MEMORY_MOD
       USE MUMPS_ANA_ORD_WRAPPERS
-      INCLUDE 'mpif.h'
+C      INCLUDE 'mpif.h'
       PUBLIC DMUMPS_ANA_F_PAR
       INTERFACE DMUMPS_ANA_F_PAR
       MODULE PROCEDURE DMUMPS_ANA_F_PAR
@@ -833,6 +833,7 @@
       END SUBROUTINE DMUMPS_GET_SUBTREES
       SUBROUTINE DMUMPS_PARSYMFACT(id, ord, GPE, GNV, WORK)  
       IMPLICIT NONE
+      INCLUDE 'mpif.h'
       TYPE(DMUMPS_STRUC)   :: id
       TYPE(ORD_TYPE)       :: ord
       INTEGER, POINTER     :: GPE(:), GNV(:)
@@ -1348,6 +1349,7 @@
       SUBROUTINE DMUMPS_BUILD_DIST_GRAPH(id, FIRST, LAST, IPE,
      &     PE, WORK)
       IMPLICIT NONE
+      INCLUDE 'mpif.h'
       TYPE(DMUMPS_STRUC)      :: id
       INTEGER(8), POINTER     :: IPE(:)
       INTEGER, POINTER        :: FIRST(:), LAST(:), PE(:),
@@ -1540,6 +1542,7 @@
       SUBROUTINE DMUMPS_BUILD_LOC_GRAPH(id, ord, GSIZE, IPE, PE, LENG,
      &     I_HALO_MAP, top_graph, WORK)
       IMPLICIT NONE
+      INCLUDE 'mpif.h'
       TYPE(DMUMPS_STRUC)   :: id
       TYPE(ORD_TYPE)       :: ord
       TYPE(GRAPH_TYPE)     :: top_graph
@@ -1836,6 +1839,7 @@
       SUBROUTINE DMUMPS_SEND_BUF(APNT, PROC, NPROCS, BUFSIZE, IPE, PE,
      &     LENG, RCVBUF, MSGCNT, SNDCNT, COMM)
       IMPLICIT NONE
+      INCLUDE 'mpif.h'
       INTEGER                 :: NPROCS, PROC, COMM
       TYPE(ARRPNT)            :: APNT(:)
       INTEGER                 :: BUFSIZE
@@ -1981,6 +1985,7 @@
       END SUBROUTINE DMUMPS_BUILD_TREE
       SUBROUTINE DMUMPS_GRAPH_DIST(id, ord, FIRST,
      &     LAST, BASE, NPROCS, WORK, TYPE)
+      USE MPI
       IMPLICIT NONE
       TYPE(DMUMPS_STRUC)   :: id
       TYPE(ORD_TYPE)       :: ord
--- MUMPS_5.1.1/src/sana_aux_par.F	2017-03-20 07:35:01.000000000 -0700
+++ MUMPS_5.1.1.patch/src/sana_aux_par.F	2017-04-05 10:31:24.000000000 -0700
@@ -14,7 +14,7 @@
       USE SMUMPS_STRUC_DEF
       USE MUMPS_MEMORY_MOD
       USE MUMPS_ANA_ORD_WRAPPERS
-      INCLUDE 'mpif.h'
+C      INCLUDE 'mpif.h'
       PUBLIC SMUMPS_ANA_F_PAR
       INTERFACE SMUMPS_ANA_F_PAR
       MODULE PROCEDURE SMUMPS_ANA_F_PAR
@@ -833,6 +833,7 @@
       END SUBROUTINE SMUMPS_GET_SUBTREES
       SUBROUTINE SMUMPS_PARSYMFACT(id, ord, GPE, GNV, WORK)  
       IMPLICIT NONE
+      INCLUDE 'mpif.h'
       TYPE(SMUMPS_STRUC)   :: id
       TYPE(ORD_TYPE)       :: ord
       INTEGER, POINTER     :: GPE(:), GNV(:)
@@ -1982,6 +1983,7 @@
       SUBROUTINE SMUMPS_GRAPH_DIST(id, ord, FIRST,
      &     LAST, BASE, NPROCS, WORK, TYPE)
       IMPLICIT NONE
+      INCLUDE 'mpif.h'
       TYPE(SMUMPS_STRUC)   :: id
       TYPE(ORD_TYPE)       :: ord
       INTEGER              :: FIRST(:), LAST(:), BASE, NPROCS, TYPE
--- MUMPS_5.1.1/src/zana_aux_par.F	2017-03-20 07:35:07.000000000 -0700
+++ MUMPS_5.1.1.patch/src/zana_aux_par.F	2017-04-05 10:33:06.000000000 -0700
@@ -14,7 +14,7 @@
       USE ZMUMPS_STRUC_DEF
       USE MUMPS_MEMORY_MOD
       USE MUMPS_ANA_ORD_WRAPPERS
-      INCLUDE 'mpif.h'
+C      INCLUDE 'mpif.h'
       PUBLIC ZMUMPS_ANA_F_PAR
       INTERFACE ZMUMPS_ANA_F_PAR
       MODULE PROCEDURE ZMUMPS_ANA_F_PAR
@@ -703,6 +703,7 @@
       END FUNCTION ZMUMPS_CNT_KIDS
       SUBROUTINE ZMUMPS_GET_SUBTREES(ord, id)
       IMPLICIT NONE
+      INCLUDE 'mpif.h'
       TYPE(ORD_TYPE)     :: ord
       TYPE(ZMUMPS_STRUC) :: id
       INTEGER, ALLOCATABLE :: ALIST(:), AWEIGHTS(:), LIST(:), WORK(:)
@@ -1982,6 +1983,7 @@
       SUBROUTINE ZMUMPS_GRAPH_DIST(id, ord, FIRST,
      &     LAST, BASE, NPROCS, WORK, TYPE)
       IMPLICIT NONE
+      INCLUDE 'mpif.h'
       TYPE(ZMUMPS_STRUC)   :: id
       TYPE(ORD_TYPE)       :: ord
       INTEGER              :: FIRST(:), LAST(:), BASE, NPROCS, TYPE
