--- conman-conman-0.2.8/etc/conman.init.in	2016-11-22 13:31:04.000000000 -0800
+++ conman-conman-0.2.8.patch/etc/conman.init.in	2017-04-05 15:41:54.000000000 -0700
@@ -18,6 +18,8 @@
 # Default-Stop:
 # Short-Description:  Start/Stop the ConMan (serial console manager) daemon.
 # Description:        Start/Stop the ConMan (serial console manager) daemon.
+# Default-Start: 3 5
+# Default-Stop: 0 1 2 6
 ### END INIT INFO
 ###############################################################################
 
