--- conman-0.2.7/etc/conman.init.in	2009-04-02 18:21:52.000000000 -0700
+++ conman-0.2.7-patch/etc/conman.init.in	2014-08-05 15:46:00.624007900 -0700
@@ -18,6 +18,8 @@
 # Should-Stop:        $remote_fs $syslog $time
 # Short-Description:  Start/Stop the ConMan (serial console manager) daemon.
 # Description:        Start/Stop the ConMan (serial console manager) daemon.
+# Default-Start: 3 5
+# Default-Stop: 0 1 2 6
 ### END INIT INFO
 ###############################################################################
 
