diff --git a/m4/with_hwloc.m4 b/m4/with_hwloc.m4
index 37d0392..5294200 100644
--- a/m4/with_hwloc.m4
+++ b/m4/with_hwloc.m4
@@ -66,10 +66,10 @@ AC_DEFUN([PBS_AC_WITH_HWLOC],
       AC_MSG_ERROR([hwloc shared object library not found.])
     ),
     # Using developer installed hwloc
-    [test -r "${hwloc_dir}/lib64/libhwloc_embedded.a"],
-    [hwloc_lib="${hwloc_dir}/lib64/libhwloc_embedded.a"],
-    [test -r "${hwloc_dir}/lib/libhwloc_embedded.a"],
-    [hwloc_lib="${hwloc_dir}/lib/libhwloc_embedded.a"],
+    [test -r "${hwloc_dir}/lib/libhwloc.so"],
+    [hwloc_lib="-L${hwloc_dir}/lib -Wl,-rpath,${hwloc_dir}/lib -lhwloc"],
+    [test -r "${hwloc_dir}/lib64/libhwloc.so"],
+    [hwloc_lib="-L${hwloc_dir}/lib64 -Wl,-rpath,${hwloc_dir}/lib64 -lhwloc"],
     AC_MSG_ERROR([hwloc library not found.])
   )
   AC_MSG_RESULT([$hwloc_dir])
