--- a/etc/slurm.conf.example	2024-10-18 13:23:42.997000000 +0000
+++ b/etc/slurm.conf.example	2024-10-18 13:24:29.375000000 +0000
@@ -57,7 +57,7 @@
 StateSaveLocation=/var/spool/slurmctld
 SwitchType=switch/none
 #TaskEpilog=
-TaskPlugin=task/affinity
+#TaskPlugin=task/affinity
 #TaskProlog=
 #TopologyPlugin=topology/tree
 #TmpFS=/tmp
@@ -93,6 +93,9 @@
 #SchedulerTimeSlice=30
 SchedulerType=sched/backfill
 SelectType=select/cons_tres
+# This is added to silence the following warning:
+# slurmctld: select/cons_tres: select_p_node_init: select/cons_tres SelectTypeParameters not specified, using default value: CR_Core_Memory
+SelectTypeParameters=CR_Core_Memory
 #
 #
 # JOB PRIORITY
@@ -122,7 +125,7 @@
 #JobCompLoc=
 #JobCompPass=
 #JobCompPort=
-JobCompType=jobcomp/none
+#JobCompType=jobcomp/none
 #JobCompUser=
 #JobContainerType=
 JobAcctGatherFrequency=30
@@ -149,5 +152,20 @@
 #
 #
 # COMPUTE NODES
-NodeName=linux[1-32] CPUs=1 State=UNKNOWN
-PartitionName=debug Nodes=ALL Default=YES MaxTime=INFINITE State=UP
+#NodeName=linux[1-32] CPUs=1 State=UNKNOWN
+#PartitionName=debug Nodes=ALL Default=YES MaxTime=INFINITE State=UP
+
+
+
+# OpenHPC default configuration
+# Enable the task/affinity plugin to add the --cpu-bind option to srun for GEOPM
+TaskPlugin=task/affinity
+PropagateResourceLimitsExcept=MEMLOCK
+JobCompType=jobcomp/filetxt
+Epilog=/etc/slurm/slurm.epilog.clean
+NodeName=c[1-4] Sockets=2 CoresPerSocket=8 ThreadsPerCore=2 State=UNKNOWN
+PartitionName=normal Nodes=c[1-4] Default=YES MaxTime=24:00:00 State=UP Oversubscribe=EXCLUSIVE
+# Enable configless option
+SlurmctldParameters=enable_configless
+# Setup interactive jobs for salloc
+LaunchParameters=use_interactive_step
