--- src/common/x11_util.c	2019-03-07 22:09:17.000000000 +0100
+++ src/common/x11_util.c	2019-03-29 10:15:41.279221289 +0100
@@ -151,7 +151,7 @@
 	xauth_argv[1] = xstrdup("list");
 	xauth_argv[2] = xstrdup(getenv("DISPLAY"));
 
-	result = run_command("xauth", XAUTH_PATH, xauth_argv, 100, &status);
+	result = run_command("xauth", XAUTH_PATH, xauth_argv, 10000, &status);
 
 	debug2("%s: result from xauth: %s", __func__, result);
 
