Index: boost_1_56_0/tools/build/src/engine/function.c
===================================================================
--- boost_1_56_0.orig/tools/build/src/engine/function.c
+++ boost_1_56_0/tools/build/src/engine/function.c
@@ -230,7 +230,7 @@ STACK * stack_global()
 
 static void check_alignment( STACK * s )
 {
-    assert( (size_t)s->data % sizeof( LIST * ) == 0 );
+    assert( (size_t)s->data % __alignof__( LIST * ) == 0 );
 }
 
 void * stack_allocate( STACK * s, int size )
