--- a/src/tests/unit/teams/CMakeLists.txt	2019-04-10 16:06:09.000000000 -0500
+++ b/src/tests/unit/teams/CMakeLists.txt	2019-05-24 17:10:12.000000000 -0500
@@ -1,3 +1,3 @@
 caf_compile_executable(team_number team-number.f90)
 caf_compile_executable(teams_subset teams_subset.f90)
-caf_compile_executable(get_communicator get-communicator.F90)
+#caf_compile_executable(get_communicator get-communicator.F90)
