diff -Nupr scotch_6.0.4.orig/src/Makefile scotch_6.0.4/src/Makefile
--- scotch_6.0.4.orig/src/Makefile	2014-10-03 00:31:36.000000000 +0900
+++ scotch_6.0.4/src/Makefile	2017-01-30 16:27:51.127814499 +0900
@@ -56,14 +56,15 @@
 
 include Makefile.inc
 
+DESTDIR		?=
 prefix		?= /usr/local
-bindir		?= $(prefix)/bin
-includedir	?= $(prefix)/include
+bindir		?= $(DESTDIR)$(prefix)/bin
+includedir	?= $(DESTDIR)$(prefix)/include
 includestub     ?= include_stub
-includestubdir	?= $(prefix)/$(includestub)
-libdir		?= $(prefix)/lib
-datarootdir	?= $(prefix)/share
-mandir		?= $(datarootdir)/man
+includestubdir	?= $(DESTDIR)$(prefix)/$(includestub)
+libdir		?= $(DESTDIR)$(prefix)/lib
+datarootdir	?= $(DESTDIR)$(prefix)/share
+mandir		?= $(DESTDIR)$(datarootdir)/man
 
 ../bin				:
 					-$(MKDIR) ../bin

