--- a/bin/wwbootstrap	2018-07-12 10:18:54.000000000 -0700
+++ b/bin/wwbootstrap	2018-07-12 10:24:12.000000000 -0700
@@ -153,8 +153,8 @@
 
 mkpath("$tmpdir/initramfs");
 
-if (! -f "$opt_chroot/boot/vmlinuz-$opt_kversion") {
-    &eprint("Can't locate the boot kernel: ". $opt_chroot ."/boot/vmlinuz-$opt_kversion\n");
+if (! -f "$opt_chroot/boot/vmlinuz-$opt_kversion" && -f "$opt_chroot/boot/vmlinux-$opt_kversion.gz") {
+    &eprint("Can't locate the boot kernel\n");
     exit 1;
 }
 
@@ -321,7 +321,11 @@
 
 # Attempt to gunzip the kernel, aarch64 kernels are compressed and iPXE can't boot gzip compressed kernels.
 # Note, if the kernel isn't a gzip, IO::Uncompress::Gunzip makes a direct copy of the file.
-gunzip "$opt_chroot/boot/vmlinuz-$opt_kversion" => "$tmpdir/kernel" or die "gunzip of kernel failed: $GunzipError\n";
+if (-e "$opt_chroot/boot/vmlinuz-$opt_kversion") {
+    gunzip "$opt_chroot/boot/vmlinuz-$opt_kversion" => "$tmpdir/kernel" or die "gunzip of kernel failed: $GunzipError\n";
+} elsif (-e "$opt_chroot/boot/vmlinux-$opt_kversion.gz") {
+    gunzip "$opt_chroot/boot/vmlinux-$opt_kversion.gz" => "$tmpdir/kernel" or die "gunzip of kernel failed: $GunzipError\n";
+}
 
 &nprint("Building and compressing bootstrap\n");
 system("(cd $tmpdir; find . | cpio -o --quiet -H newc ) | gzip -c9 > $output");
