--- warewulf-vnfs-3.7/bin/wwvnfs	2017-02-22 09:32:02.000000000 -0800
+++ warewulf-vnfs-3.7.patch/bin/wwvnfs	2017-02-23 20:44:05.000000000 -0800
@@ -347,14 +347,13 @@
 sub
 vnfs_create_file($)
 {
-    my $path = decode_utf8($File::Find::name);
+    my $path = $File::Find::name;
 
     if ($path) {
         if ($path =~ /^\Q$opt_chroot\E([[:print:]]*)$/) {
             $path = $1;
         } else {
-            &eprint("Path '$path' did not pass untaint check, skipping.\n");
-            return;
+            $path = encode_utf8($1);
         }
     }
 
