--- vnfs/libexec/wwmkchroot/opensuse-15.3.tmpl
+++ vnfs/libexec/wwmkchroot/opensuse-15.3.tmpl
@@ -31,7 +31,7 @@ PKGLIST="systemd-sysvinit aaa_base bash dracut distribution-release coreutils \
     mingetty net-tools nfs-kernel-server pam rpcbind procps psmisc shadow \
     rsync sed rsyslog tcpd timezone util-linux tar less gzip kmod-compat \
     udev openssh dhcp-client pciutils vim strace cron cpupower cpio wget
-    zypper netcfg ncurses-utils"
+    zypper netcfg ncurses-utils which"
