--- b/libexec/wwmkchroot/include-rhel	2020-02-21 10:17:41.000000000 -0800
+++ a/libexec/wwmkchroot/include-rhel	2020-02-21 17:09:27.232376164 -0800
@@ -89,2 +89,7 @@
     echo "reposdir=0" >> $CHROOTDIR/$YUM_CONF
+   # 03/13/15 karl.w.schulz@intel.com - honor proxy setting if configured on local host
+   proxy_host=`grep "^proxy=" /etc/yum.conf`
+   if [ $? -eq 0 ];then
+       echo "$proxy_host" >> $CHROOTDIR/$YUM_CONF
+   fi
     if [ "$PKG_MGR" = "dnf" ]; then
