--- warewulf-vnfs-3.7/libexec/wwmkchroot/include-suse	2017-02-22 09:32:02.000000000 -0800
+++ warewulf-vnfs-3.7.patch/libexec/wwmkchroot/include-suse	2017-02-23 09:33:29.000000000 -0800
@@ -40,7 +40,7 @@
     mkdir -p $CHROOTDIR/etc/zypp
     mkdir -p $CHROOTDIR/dev
 
-    cp -rap /etc/zypp/zypper.conf /etc/zypp/repos.d $CHROOTDIR/etc
+    cp -rap /etc/zypp/zypper.conf /etc/zypp/repos.d $CHROOTDIR/etc/zypp/.
     sed -i -e "s/\$releasever/$VERSION/g" `find $CHROOTDIR/etc/zypp* -type f`
 
     ZYP_CONF_DIRNAME=`dirname $ZYP_CONF`
@@ -92,7 +92,7 @@
         ZYP_MIRROR=$INSTALLDIRS
     fi
 
-    PKGR_CMD="$ZYP_CMD install $PKGLIST"
+    PKGR_CMD="$ZYP_CMD install --auto-agree-with-licenses $PKGLIST"
     return 0
 }
 
