--- a/vnfs/libexec/wwmkchroot/opensuse-15.1.tmpl	2020-04-28 09:59:58.211811488 -0500
+++ b/vnfs/libexec/wwmkchroot/opensuse-15.1.tmpl	2020-04-28 10:00:09.739812062 -0500
@@ -30,4 +30,4 @@
     e2fsprogs ethtool filesystem findutils gawk grep iproute2 iputils \
     mingetty net-tools nfs-kernel-server pam rpcbind procps psmisc shadow \
     rsync sed rsyslog tcpd timezone util-linux tar less gzip kmod-compat \
-    udev openssh dhcp-client pciutils vim strace cron cpupower cpio wget zypper"
+    udev openssh dhcp-client pciutils vim strace cron cpupower cpio wget \
+    zypper which"