--- a/bin/wwbootstrap	2020-03-31 22:07:54.600000000 +0000
+++ b/bin/wwbootstrap	2020-03-31 22:08:04.740000000 +0000
@@ -153,13 +153,13 @@
 
 mkpath("$tmpdir/initramfs");
 
-if (! -f "$opt_chroot/boot/vmlinuz-$opt_kversion" && ! -f "$opt_chroot/boot/Image-$opt_kversion") {
+if (! -f "$opt_chroot/boot/vmlinuz-$opt_kversion" && ! -f "$opt_chroot/boot/Image-$opt_kversion" &&
+    ! -f "$opt_chroot/lib/modules/$opt_kversion/vmlinuz") {
     &eprint("Can't locate the boot kernel\n");
     exit 1;
 }
 
 
-
 if ($config->get("drivers")) {
     my @drivers = $config->get("drivers");
     my %mod_path;
@@ -325,6 +325,8 @@
     gunzip "$opt_chroot/boot/vmlinuz-$opt_kversion" => "$tmpdir/kernel" or die "gunzip of kernel failed: $GunzipError\n";
 } elsif (-e "$opt_chroot/boot/Image-$opt_kversion") {
     gunzip "$opt_chroot/boot/Image-$opt_kversion" => "$tmpdir/kernel" or die "gunzip of kernel failed: $GunzipError\n";
+} elsif (-e "$opt_chroot/lib/modules/$opt_kversion/vmlinuz") {
+    gunzip "$opt_chroot/lib/modules/$opt_kversion/vmlinuz" => "$tmpdir/kernel" or die "gunzip of kernel failed: $GunzipError\n";
 }
 
 &nprint("Building and compressing bootstrap\n");
