--- luafilesystem-1.6.2.orig/Makefile	2014-01-29 15:43:20.000000000 -0800
+++ luafilesystem-1.6.2/Makefile	2014-05-28 09:15:54.619039297 -0700
@@ -3,6 +3,7 @@
 T= lfs

 CONFIG= ./config
+DESTDIR=

 include $(CONFIG)

@@ -15,8 +16,7 @@
 	MACOSX_DEPLOYMENT_TARGET="10.3"; export MACOSX_DEPLOYMENT_TARGET; $(CC) $(CFLAGS) $(LIB_OPTION) -o src/lfs.so $(OBJS)

 install:
-	mkdir -p $(LUA_LIBDIR)
-	cp src/lfs.so $(LUA_LIBDIR)
-
+	mkdir -p $(DESTDIR)$(LUA_LIBDIR)
+	install -m0755 src/lfs.so $(DESTDIR)$(LUA_LIBDIR)/
 clean:
 	rm -f src/lfs.so $(OBJS)
