--- a/configure.ac	2018-05-24 05:03:22.000000000 -0700
+++ b/configure.ac	2020-03-23 00:30:02.353000000 -0700
@@ -334,7 +334,7 @@
         case "${host_os}" in
             *darwin*) ;; # icc -no-gcc fails to compile some system headers
             *) 
-	       AX_CHECK_COMPILER_FLAGS([-no-gcc], [CC="$CC -no-gcc"])
+	       AX_CHECK_COMPILER_FLAGS([-no-gcc], [CC="$CC -gcc-sys"])
                ;;
         esac
         ;;
