--- src/make_ict.orig	2014-12-29 17:23:47.572584271 -0500
+++ src/make_ict	2014-12-29 17:24:17.172841399 -0500
@@ -1,6 +1,6 @@
 LIB_PATH    =
 LIBS        =
-CC          = mpiicc
+CC          = mpicc
 ifeq (,$(shell which ${CC}))
 $(error ${CC} is not defined through the PATH environment variable setting. Please try sourcing an Intel(r) Cluster Tools script file such as "mpivars.[c]sh" or "ictvars.[c]sh")
 endif
