From 0c0bf12492b81b53bed71901e1d2a43d3cc4e7c0 Mon Sep 17 00:00:00 2001
From: Peter Lemenkov <lemenkov@gmail.com>
Date: Fri, 18 Jun 2010 14:11:33 +0400
Subject: [PATCH 5/8] Do not start by default

---
 init-script.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/init-script.in b/init-script.in
index 186c757..ed6c366 100644
--- a/init-script.in
+++ b/init-script.in
@@ -5,7 +5,7 @@
 # nrpe          This shell script takes care of starting and stopping
 #               nrpe.
 #
-# chkconfig: 2345 80 30
+# chkconfig: - 80 30
 # description: nrpe is a daemon for a remote nagios server, \
 #              running nagios plugins on this host.
 # processname: nrpe
-- 
1.8.3.1

