From a3528084ceb16047ccfec1b8a0a04dddec9535d3 Mon Sep 17 00:00:00 2001
From: Peter Lemenkov <lemenkov@gmail.com>
Date: Fri, 18 Jun 2010 13:45:05 +0400
Subject: [PATCH 1/8] Add reload target to the init-script

---
 init-script.in | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/init-script.in b/init-script.in
index 07b17c7..0bd7629 100644
--- a/init-script.in
+++ b/init-script.in
@@ -51,11 +51,16 @@ case "$1" in
 	$0 stop
 	$0 start
 	;;
+  reload)
+	echo -n "Reloading nrpe"
+	killproc nrpe -HUP
+	echo
+	;;
   status)
 	status nrpe
 	;;
   *)
-	echo "Usage: nrpe {start|stop|restart|status}"
+	echo "Usage: nrpe {start|stop|restart|reload|status}"
 	exit 1
 esac
 
-- 
1.8.3.1

