diff -ruN lustre-2.8.0.old/lustre/conf/Makefile.am lustre-2.8.0/lustre/conf/Makefile.am
--- lustre-2.8.0.old/lustre/conf/Makefile.am	2016-03-09 04:31:30.000000000 +0000
+++ lustre-2.8.0/lustre/conf/Makefile.am	2016-09-20 22:57:14.000000000 +0100
@@ -48,12 +48,12 @@
 udevrulesdir = $(sysconfdir)/udev/rules.d
 udevrules_DATA = 99-lustre.rules
 
-if RHEL
+# if RHEL
 if SERVER
 sysconfigdir = $(sysconfdir)/sysconfig
 sysconfig_DATA = lustre
 endif
-endif
+# endif
 
 modprobedir = $(sysconfdir)/modprobe.d
 modprobe_DATA = ko2iblnd.conf
diff -ruN lustre-2.8.0.old/lustre/conf/Makefile.in lustre-2.8.0/lustre/conf/Makefile.in
--- lustre-2.8.0.old/lustre/conf/Makefile.in	2016-03-09 04:31:41.000000000 +0000
+++ lustre-2.8.0/lustre/conf/Makefile.in	2016-09-20 22:56:39.000000000 +0100
@@ -331,8 +331,8 @@
 @UTILS_TRUE@noinst_DATA = top.ldif lustre2ldif.xsl
 udevrulesdir = $(sysconfdir)/udev/rules.d
 udevrules_DATA = 99-lustre.rules
-@RHEL_TRUE@@SERVER_TRUE@sysconfigdir = $(sysconfdir)/sysconfig
-@RHEL_TRUE@@SERVER_TRUE@sysconfig_DATA = lustre
+@SERVER_TRUE@sysconfigdir = $(sysconfdir)/sysconfig
+@SERVER_TRUE@sysconfig_DATA = lustre
 modprobedir = $(sysconfdir)/modprobe.d
 modprobe_DATA = ko2iblnd.conf
 sysconf_DATA = ldev.conf
diff -ruN lustre-2.8.0.old/lustre/scripts/Makefile.am lustre-2.8.0/lustre/scripts/Makefile.am
--- lustre-2.8.0.old/lustre/scripts/Makefile.am	2016-03-09 04:31:30.000000000 +0000
+++ lustre-2.8.0/lustre/scripts/Makefile.am	2016-09-20 22:52:35.000000000 +0100
@@ -41,13 +41,13 @@
 
 sbin_SCRIPTS = lustre_rmmod ko2iblnd-probe
 
-if RHEL
+# if RHEL
 initdir = $(sysconfdir)/init.d
 init_SCRIPTS  = lnet lsvcgss
 if SERVER
 init_SCRIPTS += lustre
 endif
-endif
+# endif
 
 if UTILS
 sbin_SCRIPTS += ldev lustre_routes_config lustre_routes_conversion
diff -ruN lustre-2.8.0.old/lustre/scripts/Makefile.in lustre-2.8.0/lustre/scripts/Makefile.in
--- lustre-2.8.0.old/lustre/scripts/Makefile.in	2016-03-09 04:31:42.000000000 +0000
+++ lustre-2.8.0/lustre/scripts/Makefile.in	2016-09-20 22:56:56.000000000 +0100
@@ -74,7 +74,7 @@
 build_triplet = @build@
 host_triplet = @host@
 target_triplet = @target@
-@RHEL_TRUE@@SERVER_TRUE@am__append_1 = lustre
+@SERVER_TRUE@am__append_1 = lustre
 @UTILS_TRUE@am__append_2 = ldev lustre_routes_config lustre_routes_conversion
 @SERVER_TRUE@@UTILS_TRUE@am__append_3 = $(genscripts) lc_mon lhbadm lc_servip
 @SERVER_TRUE@@UTILS_TRUE@am__append_4 = lustre_req_history
@@ -335,8 +335,8 @@
 genscripts = lc_modprobe lc_net lc_hb lc_cluman lc_md lc_lvm lustre_start
 sbin_SCRIPTS = lustre_rmmod ko2iblnd-probe $(am__append_2) \
 	$(am__append_3) $(am__append_5)
-@RHEL_TRUE@initdir = $(sysconfdir)/init.d
-@RHEL_TRUE@init_SCRIPTS = lnet lsvcgss $(am__append_1)
+initdir = $(sysconfdir)/init.d
+init_SCRIPTS = lnet lsvcgss $(am__append_1)
 @UTILS_TRUE@bin_SCRIPTS = lfs_migrate $(am__append_4)
 @SERVER_TRUE@@UTILS_TRUE@hadir = $(sysconfdir)/ha.d/resource.d
 @SERVER_TRUE@@UTILS_TRUE@ha_SCRIPTS = Lustre.ha_v2
diff -ruN lustre-2.8.0.old/lustre.spec lustre-2.8.0/lustre.spec
--- lustre-2.8.0.old/lustre.spec	2016-03-09 04:31:49.000000000 +0000
+++ lustre-2.8.0/lustre.spec	2016-09-20 22:50:36.000000000 +0100
@@ -122,11 +122,11 @@
 AutoReqProv: no
 %else
 # GSS requires this: BuildRequires: pkgconfig, libgssapi-devel >= 0.10
-%if %{_vendor}=="redhat" || %{_vendor}=="fedora"
+# %if %{_vendor}=="redhat" || %{_vendor}=="fedora"
 #suse don't support selinux
 BuildRequires: libselinux-devel
 Requires: libselinux
-%endif
+# %endif
 %endif
 
 %description
@@ -140,7 +140,7 @@
 AutoReqProv: no
 %else
 # for SLES11, we need nothing here
-%if %{_vendor}=="redhat" || %{_vendor}=="fedora"
+# %if %{_vendor}=="redhat" || %{_vendor}=="fedora"
 %if %{defined krequires}
 # for RHEL we need to require the specific kernel still since weak-modules
 # support on RH is, well, weak, to be punny about it
@@ -150,7 +150,7 @@
 Requires: libyaml
 BuildRequires: libyaml-devel
 %endif
-%endif
+# %endif
 %endif
 Group: Development/Kernel
 
@@ -369,7 +369,7 @@
 echo '%{_sysconfdir}/ha.d/resource.d/Lustre' >>lustre.files
 %endif
 
-%if %{_vendor}=="redhat"
+# %if %{_vendor}=="redhat"
 # The following scripts are Red Hat specific
 %if %{with servers}
 echo '%{_sysconfdir}/sysconfig/lustre' >>lustre.files
@@ -377,7 +377,7 @@
 %endif
 echo '%{_sysconfdir}/init.d/lnet' >>lustre.files
 echo '%{_sysconfdir}/init.d/lsvcgss' >>lustre.files
-%endif
+# %endif
 
 # Create the pristine source directory.
 cd $RPM_BUILD_DIR/lustre-%{version}
diff -ruN lustre-2.8.0.old/lustre.spec.in lustre-2.8.0/lustre.spec.in
--- lustre-2.8.0.old/lustre.spec.in	2016-03-09 04:31:30.000000000 +0000
+++ lustre-2.8.0/lustre.spec.in	2016-09-20 22:51:01.000000000 +0100
@@ -122,11 +122,11 @@
 AutoReqProv: no
 %else
 # GSS requires this: BuildRequires: pkgconfig, libgssapi-devel >= 0.10
-%if %{_vendor}=="redhat" || %{_vendor}=="fedora"
+# %if %{_vendor}=="redhat" || %{_vendor}=="fedora"
 #suse don't support selinux
 BuildRequires: libselinux-devel
 Requires: libselinux
-%endif
+# %endif
 %endif
 
 %description
@@ -140,7 +140,7 @@
 AutoReqProv: no
 %else
 # for SLES11, we need nothing here
-%if %{_vendor}=="redhat" || %{_vendor}=="fedora"
+# %if %{_vendor}=="redhat" || %{_vendor}=="fedora"
 %if %{defined krequires}
 # for RHEL we need to require the specific kernel still since weak-modules
 # support on RH is, well, weak, to be punny about it
@@ -150,7 +150,7 @@
 Requires: libyaml
 BuildRequires: libyaml-devel
 %endif
-%endif
+# %endif
 %endif
 Group: Development/Kernel
 
@@ -369,7 +369,7 @@
 echo '%{_sysconfdir}/ha.d/resource.d/Lustre' >>lustre.files
 %endif
 
-%if %{_vendor}=="redhat"
+# %if %{_vendor}=="redhat"
 # The following scripts are Red Hat specific
 %if %{with servers}
 echo '%{_sysconfdir}/sysconfig/lustre' >>lustre.files
@@ -377,7 +377,7 @@
 %endif
 echo '%{_sysconfdir}/init.d/lnet' >>lustre.files
 echo '%{_sysconfdir}/init.d/lsvcgss' >>lustre.files
-%endif
+# %endif
 
 # Create the pristine source directory.
 cd $RPM_BUILD_DIR/lustre-%{version}
