--- superlu_dist-4.0/SRC/Makefile	2014-09-20 07:55:32.000000000 -0700
+++ superlu_dist-4.0.patch/SRC/Makefile	2015-04-28 11:05:22.000000000 -0700
@@ -45,7 +45,7 @@
 #
 # Routines literally taken from SuperLU
 #
-DSLUSRC	= dlangs.o dgsequ.o dlaqgs.o dutil.o \
+DSLUSRC	= dgsequ.o dlangs.o dlaqgs.o dutil.o \
 	  dmemory.o dmyblas2.o dsp_blas2.o dsp_blas3.o
 ZSLUSRC	= dcomplex.o zlangs.o zgsequ.o zlaqgs.o zutil.o \
 	  zmemory.o zmyblas2.o zsp_blas2.o zsp_blas3.o
