diff -aruN SuperLU_DIST_3.2.orig/make.inc SuperLU_DIST_3.2/make.inc
--- SuperLU_DIST_3.2.orig/make.inc	2012-10-24 21:46:19.000000000 +0200
+++ SuperLU_DIST_3.2/make.inc	2012-12-30 19:06:59.442616750 +0100
@@ -23,23 +23,20 @@
 #  The name of the libraries to be created/linked to
 #
 DSuperLUroot 	= ${HOME}/Release_Codes/SuperLU_DIST-branch
-DSUPERLULIB   	= $(DSuperLUroot)/lib/libsuperlu_dist_3.0.a
+DSUPERLULIB   	= $(DSuperLUroot)/lib/libsuperlu_dist_3.2.a
+DSUPERLUSLIB   	= -L$(DSuperLUroot)/lib -lsuperlu_dist
 #
 BLASDEF	     	= -DUSE_VENDOR_BLAS
-BLASLIB	        =
+BLASLIB	        = -lmkl_rt
 #
-## parmetis 4.x.x, 32-bit integer
-METISLIB    	= -L${HOME}/Hopper/lib/parmetis-4.0.2/build/Linux-x86_64/libmetis -lmetis
-PARMETISLIB  	= -L${HOME}/Hopper/lib/parmetis-4.0.2/build/Linux-x86_64/libparmetis -lparmetis
-## parmetis 4.x.x, 64-bit integer
-# METISLIB    	= -L${HOME}/Hopper/lib/parmetis-4.0.2-64bit/build/Linux-x86_64/libmetis -lmetis
-# PARMETISLIB  	= -L${HOME}/Hopper/lib/parmetis-4.0.2-64bit/build/Linux-x86_64/libparmetis -lparmetis
+METISLIB    	= -L$(METIS_LIB) -lmetis
+#PARMETISLIB  	= 
 
 # Define the required Fortran libraries, if you use C compiler to link
-FLIBS	 	= -lpgf90 -lpgf90_rpm1
+FLIBS	 	= 
 
 # Define all the libraries
-LIBS	     	= $(DSUPERLULIB) $(BLASLIB) $(PARMETISLIB) $(METISLIB) $(FLIBS)
+LIBS	     	= $(DSUPERLUSLIB) $(BLASLIB) $(PARMETISLIB) $(METISLIB) $(FLIBS)
 
 #
 #  The archiver and the flag(s) to use when building archive (library)
@@ -51,21 +48,26 @@
 
 ############################################################################
 # C compiler setup
-CC           	= cc
+CC           	= mpicc
 # CFLAGS should be set to be the C flags that include optimization
-CFLAGS          = -fast -Mipa=fast,safe \
-		-DDEBUGlevel=0 -DPRNTlevel=1 -DPROFlevel=0 \
-#		-D_LONGINT
+CFLAGS       = $(RPM_OPT_FLAGS) -fPIC -fschedule-insns2 \
+               -funroll-loops -fstrict-aliasing -frerun-loop-opt \
+	       -ftree-vectorize -funit-at-a-time
+ifeq ($(MACH),i386)
+ CFLAGS     += -mtune=pentium-m -momit-leaf-frame-pointer # -ffast-math -msse2 -mfpmath=sse
+else
+ CFLAGS     += -O3 -finline-limit=720 --param max-inline-insns-auto=160  
+endif		-D_LONGINT
 #
 # NOOPTS should be set to be the C flags that turn off any optimization
-NOOPTS		= -O0
+NOOPTS		= $(RPM_OPT_FLAGS) -Os -fPIC
 ############################################################################
 # FORTRAN compiler setup
-FORTRAN         = ftn
-F90FLAGS	= -fast -Mipa=fast,safe
+FORTRAN         = mpif90
+F90FLAGS	=
 ############################################################################
 LOADER	        = $(CC)
-LOADOPTS	= -fast
+LOADOPTS	=
 ############################################################################
 #  C preprocessor defs for compilation (-DNoChange, -DAdd_, or -DUpCase)
 #
