diff -aruN SuperLU_DIST_3.2.orig/SRC/pdgstrf.c SuperLU_DIST_3.2/SRC/pdgstrf.c
--- SuperLU_DIST_3.2.orig/SRC/pdgstrf.c	2012-10-24 21:46:17.000000000 +0200
+++ SuperLU_DIST_3.2/SRC/pdgstrf.c	2012-12-08 22:56:02.996450019 +0100
@@ -2598,7 +2598,7 @@
 		    MPI_Isend(ublk_ptr, u_diag_cnt, MPI_DOUBLE, pr,
 			      tag, comm, U_diag_blk_send_req + pr);
 		}
-	    U_diag_blk_send_req[krow] = 1; /* flag outstanding Isend */
+	    U_diag_blk_send_req[krow] = (int *)(1); /* flag outstanding Isend */
 	}
 
     } else  { /* non-diagonal process */
diff -aruN SuperLU_DIST_3.2.orig/SRC/pzgstrf.c SuperLU_DIST_3.2/SRC/pzgstrf.c
--- SuperLU_DIST_3.2.orig/SRC/pzgstrf.c	2012-10-24 21:46:17.000000000 +0200
+++ SuperLU_DIST_3.2/SRC/pzgstrf.c	2012-12-08 23:00:21.160450018 +0100
@@ -2544,7 +2544,7 @@
 		    MPI_Isend(ublk_ptr, u_diag_cnt, SuperLU_MPI_DOUBLE_COMPLEX, pr,
 			      tag, comm, U_diag_blk_send_req + pr);
 		}
-	    U_diag_blk_send_req[krow] = 1; /* flag outstanding Isend */
+	    U_diag_blk_send_req[krow] = (int *)(1); /* flag outstanding Isend */
 	}
 
     } else  { /* non-diagonal process */
