From 657876dd7454bf526c70e0747c9d8a1a17828e99 Mon Sep 17 00:00:00 2001
From: Peter Lemenkov <lemenkov@gmail.com>
Date: Wed, 10 Mar 2010 12:08:35 +0300
Subject: [PATCH 02/10] Remove assignment of (not parsed) to jitter

---
 plugins-scripts/check_ntp.pl | 1 -
 1 file changed, 1 deletion(-)

diff --git a/plugins-scripts/check_ntp.pl b/plugins-scripts/check_ntp.pl
index 5c87e0a..0733a63 100755
--- a/plugins-scripts/check_ntp.pl
+++ b/plugins-scripts/check_ntp.pl
@@ -313,7 +313,6 @@ if ($have_ntpq) {
 				}
 			} else {
 				print "No match!\n" if $verbose;
-				$jitter = '(not parsed)';
 			}
 			
 		}
-- 
1.8.3.1

