diff --git b/plugins-scripts/check_log.sh a/plugins-scripts/check_log.sh
index ca5810a..7dba651 100755
--- b/plugins-scripts/check_log.sh
+++ a/plugins-scripts/check_log.sh
@@ -60,8 +60,8 @@
 # TV: removed PATH restriction. Need to think more about what this means overall
 #PATH=""
 
-GREP="/bin/egrep"
-DIFF="/bin/diff"
+GREP="/usr/bin/egrep"
+DIFF="/usr/bin/diff"
 TAIL="/bin/tail"
 CAT="/bin/cat"
 RM="/bin/rm"
