diff -ruN nagios/daemon-init.in nagios-modified/daemon-init.in
--- nagios/daemon-init.in	2010-10-23 10:59:18.000000000 +0100
+++ nagios-modified/daemon-init.in	2013-08-29 03:19:07.501259965 +0100
@@ -129,7 +129,8 @@
 		echo -n "Starting nagios:"
 		$NagiosBin -v $NagiosCfgFile > /dev/null 2>&1;
 		if [ $? -eq 0 ]; then
-			su - $NagiosUser -c "touch $NagiosVarDir/nagios.log $NagiosRetentionFile"
+                        touch $NagiosVarDir/nagios.log $NagiosRetentionFile
+                        chown $NagiosUser:$NagiosGroup $NagiosVarDir/nagios.log $NagiosRetentionFile
 			rm -f $NagiosCommandFile
 			touch $NagiosRunFile
 			chown $NagiosUser:$NagiosGroup $NagiosRunFile
