--- a/daemon-init.in	2017-08-24 14:43:48.000000000 -0700
+++ b/daemon-init.in	2017-09-29 09:25:55.000000000 -0700
@@ -210,6 +210,7 @@
 		su $NagiosUser -c "touch $NagiosVarDir/nagios.log $NagiosRetentionFile"
 		remove_commandfile
 		touch $NagiosRunFile
+		[ -x /sbin/restorecon ] && /sbin/restorecon $NagiosRunFile
 		$NagiosBin -d $NagiosCfgFile
 		if [ -d $NagiosLockDir ]; then touch $NagiosLockDir/$NagiosLockFile; fi
 
