--- a/startup/default-init.in	2018-06-25 14:25:22.000000000 -0700
+++ b/startup/default-init.in	2018-07-13 10:47:53.000000000 -0700
@@ -138,11 +138,14 @@
 
 printstatus_nagios ()
 {
-	if status_nagios; then
+	status_nagios
+	RETVAL=$?
+	if [ "0" -eq "$RETVAL" ]; then
 		echo "nagios (pid $NagiosPID) is running..."
 	else
 		echo "nagios is not running"
 	fi
+	return $RETVAL
 }
 
 killproc_nagios ()
@@ -236,6 +239,7 @@
 			exit 3
 		fi
 		printstatus_nagios
+		exit $?
 		;;
 
 	checkconfig)
