From 53ad344a9df8af8c3d59d0b4127201e6f1d19277 Mon Sep 17 00:00:00 2001
From: Peter Lemenkov <lemenkov@gmail.com>
Date: Thu, 23 Sep 2010 22:32:46 +0400
Subject: [PATCH 06/11] Do not start service by default

Signed-off-by: Peter Lemenkov <lemenkov@gmail.com>
---
 daemon-init.in |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/daemon-init.in b/daemon-init.in
index ecd072a..5e52637 100644
--- a/startup/default-init.in
+++ b/startup/default-init.in
@@ -1,6 +1,6 @@
 #!/bin/sh
 #
-# chkconfig: 345 99 01
+# chkconfig: - 99 01
 # description: Nagios network monitor
 # processname: nagios
 # File : nagios
-- 
1.7.3.5
