--- nagios-4.3.1/Makefile.in	2017-02-23 12:00:40.000000000 -0800
+++ nagios-4.3.1.patch/Makefile.in	2017-02-27 15:02:59.000000000 -0800
@@ -295,6 +295,7 @@
 	@echo ""
 
 install-webconf:
+	$(INSTALL) -m 775 $(INSTALL_OPTS) -d $(DESTDIR)$(HTTPD_CONF)
 	$(INSTALL) -m 644 sample-config/httpd.conf $(DESTDIR)$(HTTPD_CONF)/nagios.conf
 	if [ $(LN_HTTPD_SITES_ENABLED) -eq 1 ]; then \
 		$(LN_S) $(DESTDIR)$(HTTPD_CONF)/nagios.conf $(DESTDIR)/etc/apache2/sites-enabled/nagios.conf; \
