--- nagios-3.4.4/sample-config/template-object/templates.cfg.in	2007-10-03 22:43:14.000000000 +0100
+++ nagios-3.4.4-modified/sample-config/template-object/templates.cfg.in	2013-01-13 03:12:00.791398046 +0000
@@ -117,6 +117,7 @@
 	notification_interval	30		; Resend notifications every 30 minutes
 	notification_options	d,r		; Only send notifications for specific host states
 	contact_groups		admins		; Notifications get sent to the admins by default
+	statusmap_image		printer.png
 	register		0		; DONT REGISTER THIS - ITS JUST A TEMPLATE
 	}
 
@@ -134,6 +135,15 @@
 	notification_interval	30		; Resend notifications every 30 minutes
 	notification_options	d,r		; Only send notifications for specific host states
 	contact_groups		admins		; Notifications get sent to the admins by default
+	statusmap_image		switch.png
+	register		0		; DONT REGISTER THIS - ITS JUST A TEMPLATE
+	}
+
+# Define a template for routers that we can reuse
+define host{
+	name			generic-router	; The name of this host template
+	use			generic-switch	; Inherit default values from the generic-host template
+	statusmap_image		router.png
 	register		0		; DONT REGISTER THIS - ITS JUST A TEMPLATE
 	}
 
