--- nagios-4.3.1/daemon-init.in	2017-02-23 12:00:40.000000000 -0800
+++ nagios-4.3.1.patch/daemon-init.in	2017-02-27 14:58:21.000000000 -0800
@@ -136,11 +136,14 @@
 
 printstatus_nagios ()
 {
-	if status_nagios; then
+	status_nagios
+	RETVAL=$?
+	if [ "0" -eq "$RETVAL" ]; then
 		echo "nagios (pid $NagiosPID) is running..."
 	else
 		echo "nagios is not running"
 	fi
+	return $RETVAL
 }
 
 killproc_nagios ()
@@ -235,6 +239,7 @@
 	status)
 		pid_nagios
 		printstatus_nagios
+		exit $?
 		;;
 
 	checkconfig)
