--- nagios-4.3.1/daemon-init.in	2017-02-23 12:00:40.000000000 -0800
+++ nagios-4.3.1.patch/daemon-init.in	2017-02-27 14:54:05.000000000 -0800
@@ -195,6 +195,7 @@
 		rm -f $NagiosCommandFile
 		touch $NagiosRunFile
 		chown -h $NagiosUser:$NagiosGroup $NagiosRunFile $NagiosVarDir/nagios.log $NagiosRetentionFile
+        [ -x /sbin/restorecon ] && /sbin/restorecon $NagiosRunFile
 		$NagiosBin -d $NagiosCfgFile
 		if [ -d $NagiosLockDir ]; then touch $NagiosLockDir/$NagiosLockFile; fi
 
