--- nagios-4.3.1/sample-config/httpd.conf.in	2017-02-23 12:00:40.000000000 -0800
+++ nagios-4.3.1.patch/sample-config/httpd.conf.in	2017-02-27 15:11:56.000000000 -0800
@@ -5,9 +5,9 @@
 # configuration file.  Customize the paths, etc. as
 # needed to fit your system.
 
-ScriptAlias @cgiurl@ "@sbindir@"
+ScriptAlias @cgiurl@ "@cgidir@"
 
-<Directory "@sbindir@">
+<Directory "@cgidir@">
 #  SSLRequireSSL
    Options ExecCGI
    AllowOverride None
--- nagios-4.3.1/subst.in	2017-02-23 12:00:40.000000000 -0800
+++ nagios-4.3.1.patch/subst.in	2017-02-27 15:11:56.000000000 -0800
@@ -35,6 +35,7 @@
 		s|\@datarootdir\@|@datarootdir@|g;
 		s|\@sbindir\@|@sbindir@|g;
 		s|\@bindir\@|@bindir@|g;
+		s|\@cgidir\@|@libdir@/cgi-bin/|g;
 		s|\@htmurl\@|@htmurl@|g;
 		s|\@cgiurl\@|@cgiurl@|g;
 		s|\@MAIL_PROG\@|@MAIL_PROG@|g;
