--- nagios-4.3.1/sample-config/httpd.conf.in	2017-02-23 12:00:40.000000000 -0800
+++ nagios-4.3.1.patch/sample-config/httpd.conf.in	2017-02-28 13:38:02.000000000 -0800
@@ -18,8 +17,24 @@
 
          AuthName "Nagios Access"
          AuthType Basic
-         AuthUserFile @sysconfdir@/htpasswd.users
-         Require valid-user
+         AuthUserFile @sysconfdir@/passwd
+         <IfModule mod_authz_core.c>
+            # Apache 2.4
+            <RequireAll>
+               Require all granted
+               # Require local
+               Require valid-user
+            </RequireAll>
+         </IfModule>
+         <IfModule !mod_authz_core.c>
+            # Apache 2.2
+            Order allow,deny
+            Allow from all
+            #  Order deny,allow
+            #  Deny from all
+            #  Allow from 127.0.0.1
+            Require valid-user
+         </IfModule>
       </RequireAll>
    </IfVersion>
    <IfVersion < 2.3>
@@ -31,7 +46,7 @@
 
       AuthName "Nagios Access"
       AuthType Basic
-      AuthUserFile @sysconfdir@/htpasswd.users
+      AuthUserFile @sysconfdir@/passwd
       Require valid-user
    </IfVersion>
 </Directory>
@@ -41,16 +56,31 @@
 <Directory "@datadir@">
 #  SSLRequireSSL
    Options None
-   AllowOverride None
    <IfVersion >= 2.3>
       <RequireAll>
          Require all granted
-#        Require host 127.0.0.1
 
          AuthName "Nagios Access"
          AuthType Basic
-         AuthUserFile @sysconfdir@/htpasswd.users
-         Require valid-user
+         AuthUserFile @sysconfdir@/passwd
+
+          <IfModule mod_authz_core.c>
+            # Apache 2.4
+            <RequireAll>
+               Require all granted
+               # Require local
+               Require valid-user
+            </RequireAll>
+         </IfModule>
+         <IfModule !mod_authz_core.c>
+            # Apache 2.2
+            Order allow,deny
+            Allow from all
+            #  Order deny,allow
+            #  Deny from all
+            #  Allow from 127.0.0.1
+            Require valid-user
+         </IfModule>
       </RequireAll>
    </IfVersion>
    <IfVersion < 2.3>
@@ -62,7 +92,7 @@
 
       AuthName "Nagios Access"
       AuthType Basic
-      AuthUserFile @sysconfdir@/htpasswd.users
+      AuthUserFile @sysconfdir@/passwd
       Require valid-user
    </IfVersion>
 </Directory>
