--- nagios-4.3.1/Makefile.in	2017-02-23 12:00:40.000000000 -0800
+++ nagios-4.3.1.patch/Makefile.in	2017-02-27 15:05:49.000000000 -0800
@@ -376,7 +377,7 @@
 	doxygen doxy.conf
 
 
-fullinstall: install install-init install-commandmode install-webconf install-devel
+fullinstall: install install-init install-commandmode install-webconf install-devel install-config
 
 # Uninstall is too destructive if base install directory is /usr, etc.
 #uninstall:
