--- a/Makefile.in	2018-06-25 14:25:22.000000000 -0700
+++ b/Makefile.in	2018-07-13 10:50:33.000000000 -0700
@@ -439,7 +439,7 @@
 	doxygen doxy.conf
 
 
-fullinstall: install install-init install-commandmode install-webconf install-devel
+fullinstall: install install-init install-commandmode install-webconf install-devel install-config
 
 # Uninstall is too destructive if base install directory is /usr, etc.
 #uninstall:
