diff -ruN nagios/daemon-init.in nagios-modified/daemon-init.in
--- nagios/daemon-init.in	2013-08-29 03:19:07.501259965 +0100
+++ nagios-modified/daemon-init.in	2013-08-29 03:20:36.733259743 +0100
@@ -187,6 +187,7 @@
 		rm -f $NagiosCommandFile
 		touch $NagiosRunFile
 		chown $NagiosUser:$NagiosGroup $NagiosRunFile $NagiosVarDir/nagios.log $NagiosRetentionFile
+		[ -x /sbin/restorecon ] && /sbin/restorecon $NagiosRunFile
 		$NagiosBin -d $NagiosCfgFile
 		if [ -d $NagiosLockDir ]; then touch $NagiosLockDir/$NagiosLockFile; fi
 
