diff -ruN easybuild-easyblocks-2.5.0.old/easybuild/easyblocks/a/atlas.py easybuild-easyblocks-2.5.0/easybuild/easyblocks/a/atlas.py
--- easybuild-easyblocks-2.5.0.old/easybuild/easyblocks/a/atlas.py	2015-12-14 22:08:03.000000000 +0000
+++ easybuild-easyblocks-2.5.0/easybuild/easyblocks/a/atlas.py	2016-09-28 13:30:17.168871697 +0100
@@ -90,8 +90,8 @@
         # use cycle accurate timer for timings
         # see http://math-atlas.sourceforge.net/atlas_install/node23.html
         # this should work on Linux with both GCC and Intel compilers
-        cpu_freq = int(get_cpu_speed())
-        self.cfg.update('configopts', "-D c -DPentiumCPS=%s" % cpu_freq)
+        # use -DWALL for non-x86, see http://math-atlas.sourceforge.net/atlas_install/node25.html
+        self.cfg.update('configopts', "-D c -DWALL")
 
 
         # if LAPACK is found, instruct ATLAS to provide a full LAPACK library
