--- contrib/platform/intel/hillsboro/orcm-linux.conf	2014-09-30 08:30:24.000000000 -0700
+++ contrib/platform/intel/hillsboro/orcm-linux.conf	2015-01-25 20:18:19.000000000 -0800
@@ -82,5 +82,5 @@
 ## Setup shared memory
 btl_sm_free_list_max = 768
 
-sensor_ipmi_bmc_username = root
-sensor_ipmi_bmc_password = knc123
+sensor_ipmi_bmc_username = <your-bmc-username>
+sensor_ipmi_bmc_password = <your-bmc-password>
